{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ejercicios\n",
    "+ <a href=\"#Ejercicio_1\" style=\"color:#058B8C\">Ejercicio 1</a>\n",
    "+ <a href=\"#Ejercicio_2\" style=\"color:#058B8C\">Ejercicio 2</a>\n",
    "+ <a href=\"#Ejercicio_3\" style=\"color:#058B8C\">Ejercicio 3</a>\n",
    "+ <a href=\"#Ejercicio_4\" style=\"color:#058B8C\">Ejercicio 4</a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_1\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 1</h2>\n",
    "<p>Arreglar el código para que solo muestre las cominaciones únicas</p>\n",
    "\n",
    "\n",
    "<h3 id=\"sample-output\">Resultado emitido de muestra</h3>\n",
    "<div class=\"codehilite\"><pre>possible di-sulfide bond: Cys 155 - Cys 181 2.0\n",
    "possible di-sulfide bond: Cys 257 - Cys 338 2.03\n",
    "possible di-sulfide bond: Cys 515 - Cys 541 2.02\n",
    "possible di-sulfide bond: Cys 618 - Cys 699 2.03\n",
    "possible di-sulfide bond: Cys 881 - Cys 855 2.02\n",
    "\n",
    "</pre></div>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "import Bio.PDB.PDBParser\n",
    "import sys\n",
    "\n",
    "# Use QUIET=True to avoid lots of warnings...\n",
    "parser = Bio.PDB.PDBParser(QUIET=True)\n",
    "structure = parser.get_structure(\"1KCW\", \"1kcw.pdb\")\n",
    "model = structure[0]\n",
    "achain = model['A']\n",
    "\n",
    "cysresidues = []\n",
    "\n",
    "#Voy por cada residuo hasta encontrar CYS\n",
    "for residue in achain:\n",
    "    if residue.get_resname() == 'CYS':\n",
    "        cysresidues.append(residue)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "possible di-sulfide bond: Cys 155 - Cys 181 2.0\n",
      "possible di-sulfide bond: Cys 257 - Cys 338 2.03\n",
      "possible di-sulfide bond: Cys 515 - Cys 541 2.02\n",
      "possible di-sulfide bond: Cys 618 - Cys 699 2.03\n",
      "possible di-sulfide bond: Cys 855 - Cys 881 2.02\n"
     ]
    }
   ],
   "source": [
    "# Se crean 3 listas vacias(esto con el fin de no repetir resultados) para guardar los indices de c1 y c2 y también guardar su distancia.\n",
    "list_distance=[]\n",
    "list_c1index=[]\n",
    "list_c2index=[]\n",
    "for c1 in cysresidues:\n",
    "    c1index = c1.get_id()[1]\n",
    "    for c2 in cysresidues:\n",
    "        c2index = c2.get_id()[1]\n",
    "#El primer \"and\" se usa para evitar cys con el mismo indice y valor,\n",
    "# el 2° y 3° \"and\" se utilizan para que los datos que esten en una lista no se reflejen en la otra. Así tenemos 2 listas unicas,con valores unicos.\n",
    "        if ((c1['SG'] - c2['SG']) < 8.0) and ((c1index!=c2index)) and ((c2index in list_c1index)==False) and ((c1index in list_c2index)==False) :\n",
    "            list_c1index.append(c1index)\n",
    "            list_c2index.append(c2index)\n",
    "            list_distance.append(round(c1['SG'] - c2['SG'],2))\n",
    "\n",
    "#Se imprimen los datos obtenidos.\n",
    "for i in range (len(list_c1index)):\n",
    "    print(\"possible di-sulfide bond:\", end=\" \")\n",
    "    print(\"Cys\",list_c1index[i],\"-\", end=\" \")\n",
    "    print(\"Cys\",list_c2index[i], end=\" \")\n",
    "    print(list_distance[i])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_2\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 2</h2>\n",
    "<p> BS3 es un  agente entrecruzante  impermeable a la membrana plasmática. Sirve para el estudio de\n",
    "interacciones de proteínas de membrana plasmática con residuos de lisina\n",
    "accesibles desde el medio extracelular</p>\n",
    "<img alt=\"GenBank Format\"src=\"attachment:image.png\" width=\"600\">\n",
    "\n",
    "    \n",
    "+ Encuentren los pares de lisinas que podrían ser unidas utilizando BS3 \n",
    "\n",
    "La distancia suele ser menor a 12 Å\n",
    "\n",
    "</pre></div>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HEADER    HIV-1 ENVELOPE PROTEIN                  12-MAR-99   1CE0              \n",
      "TITLE     TRIMERIZATION SPECIFICITY IN HIV-1 GP41: ANALYSIS WITH A GCN4 LEUCINE \n",
      "TITLE    2 ZIPPER MODEL                                                         \n",
      "COMPND    MOL_ID: 1;                                                            \n",
      "COMPND   2 MOLECULE: PROTEIN (LEUCINE ZIPPER MODEL H38-P1);                     \n",
      "COMPND   3 CHAIN: A, B, C;                                                      \n",
      "COMPND   4 ENGINEERED: YES                                                      \n",
      "SOURCE    MOL_ID: 1;                                                            \n",
      "SOURCE   2 ORGANISM_SCIENTIFIC: HUMAN IMMUNODEFICIENCY VIRUS 1;                 \n",
      "SOURCE   3 ORGANISM_TAXID: 11676;                                               \n",
      "SOURCE   4 EXPRESSION_SYSTEM: ESCHERICHIA COLI;                                 \n",
      "SOURCE   5 EXPRESSION_SYSTEM_TAXID: 562;                                        \n",
      "SOURCE   6 EXPRESSION_SYSTEM_STRAIN: BL21(PLYS);                                \n",
      "SOURCE   7 EXPRESSION_SYSTEM_PLASMID: PH38-P1;                                  \n",
      "SOURCE   8 OTHER_DETAILS: SYNTHETIC GENE                                        \n",
      "KEYWDS    HIV-1 ENVELOPE PROTEIN, GP41, PROTEIN OLIGOMERIZATION, COILED COIL,   \n",
      "KEYWDS   2 LEUCINE ZIPPER                                                       \n",
      "EXPDTA    X-RAY DIFFRACTION                                                     \n",
      "AUTHOR    W.SHU,H.JI,M.LU                                                       \n",
      "REVDAT   5   06-NOV-19 1CE0    1       JRNL                                     \n",
      "REVDAT   4   24-FEB-09 1CE0    1       VERSN                                    \n",
      "REVDAT   3   26-SEP-01 1CE0    3       ATOM                                     \n",
      "REVDAT   2   04-MAY-99 1CE0    1       JRNL                                     \n",
      "REVDAT   1   19-MAR-99 1CE0    0                                                \n",
      "JRNL        AUTH   W.SHU,H.JI,M.LU                                              \n",
      "JRNL        TITL   TRIMERIZATION SPECIFICITY IN HIV-1 GP41: ANALYSIS WITH A     \n",
      "JRNL        TITL 2 GCN4 LEUCINE ZIPPER MODEL.                                   \n",
      "JRNL        REF    BIOCHEMISTRY                  V.  38  5378 1999              \n",
      "JRNL        REFN                   ISSN 0006-2960                               \n",
      "JRNL        PMID   10220324                                                     \n",
      "JRNL        DOI    10.1021/BI990199W                                            \n",
      "REMARK   1                                                                      \n",
      "REMARK   1 REFERENCE 1                                                          \n",
      "REMARK   1  AUTH   E.K.O'SHEA,J.D.KLEMM,P.S.KIM,T.ALBER                         \n",
      "REMARK   1  TITL   X-RAY STRUCTURE OF THE GCN4 LEUCINE ZIPPER, A TWO-STRANDED,  \n",
      "REMARK   1  TITL 2 PARALLEL COILED COIL.                                        \n",
      "REMARK   1  REF    SCIENCE                       V. 254   539 1991              \n",
      "REMARK   1  REFN                   ISSN 0036-8075                               \n",
      "REMARK   1  PMID   1948029                                                      \n",
      "REMARK   1  DOI    10.1126/SCIENCE.1948029                                      \n",
      "REMARK   1 REFERENCE 2                                                          \n",
      "REMARK   1  AUTH   K.TAN,J.LIU,J.WANG,S.SHEN,M.LU                               \n",
      "REMARK   1  TITL   ATOMIC STRUCTURE OF A THERMOSTABLE SUBDOMAIN OF HIV-1 GP41.  \n",
      "REMARK   1  REF    PROC.NATL.ACAD.SCI.USA        V.  94 12303 1997              \n",
      "REMARK   1  REFN                   ISSN 0027-8424                               \n",
      "REMARK   1  PMID   9356444                                                      \n",
      "REMARK   1  DOI    10.1073/PNAS.94.23.12303                                     \n",
      "REMARK   2                                                                      \n",
      "REMARK   2 RESOLUTION.    2.40 ANGSTROMS.                                       \n",
      "REMARK   3                                                                      \n",
      "REMARK   3 REFINEMENT.                                                          \n",
      "REMARK   3   PROGRAM     : X-PLOR 3.851                                         \n",
      "REMARK   3   AUTHORS     : BRUNGER                                              \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  DATA USED IN REFINEMENT.                                            \n",
      "REMARK   3   RESOLUTION RANGE HIGH (ANGSTROMS) : 2.40                           \n",
      "REMARK   3   RESOLUTION RANGE LOW  (ANGSTROMS) : 8.00                           \n",
      "REMARK   3   DATA CUTOFF            (SIGMA(F)) : 2.000                          \n",
      "REMARK   3   DATA CUTOFF HIGH         (ABS(F)) : NULL                           \n",
      "REMARK   3   DATA CUTOFF LOW          (ABS(F)) : NULL                           \n",
      "REMARK   3   COMPLETENESS (WORKING+TEST)   (%) : 95.8                           \n",
      "REMARK   3   NUMBER OF REFLECTIONS             : 6559                           \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  FIT TO DATA USED IN REFINEMENT.                                     \n",
      "REMARK   3   CROSS-VALIDATION METHOD          : THROUGHOUT                      \n",
      "REMARK   3   FREE R VALUE TEST SET SELECTION  : RANDOM                          \n",
      "REMARK   3   R VALUE            (WORKING SET) : 0.221                           \n",
      "REMARK   3   FREE R VALUE                     : 0.294                           \n",
      "REMARK   3   FREE R VALUE TEST SET SIZE   (%) : 7.000                           \n",
      "REMARK   3   FREE R VALUE TEST SET COUNT      : 500                             \n",
      "REMARK   3   ESTIMATED ERROR OF FREE R VALUE  : NULL                            \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  FIT IN THE HIGHEST RESOLUTION BIN.                                  \n",
      "REMARK   3   TOTAL NUMBER OF BINS USED           : NULL                         \n",
      "REMARK   3   BIN RESOLUTION RANGE HIGH       (A) : NULL                         \n",
      "REMARK   3   BIN RESOLUTION RANGE LOW        (A) : NULL                         \n",
      "REMARK   3   BIN COMPLETENESS (WORKING+TEST) (%) : NULL                         \n",
      "REMARK   3   REFLECTIONS IN BIN    (WORKING SET) : NULL                         \n",
      "REMARK   3   BIN R VALUE           (WORKING SET) : NULL                         \n",
      "REMARK   3   BIN FREE R VALUE                    : NULL                         \n",
      "REMARK   3   BIN FREE R VALUE TEST SET SIZE  (%) : NULL                         \n",
      "REMARK   3   BIN FREE R VALUE TEST SET COUNT     : NULL                         \n",
      "REMARK   3   ESTIMATED ERROR OF BIN FREE R VALUE : NULL                         \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  NUMBER OF NON-HYDROGEN ATOMS USED IN REFINEMENT.                    \n",
      "REMARK   3   PROTEIN ATOMS            : 892                                     \n",
      "REMARK   3   NUCLEIC ACID ATOMS       : 0                                       \n",
      "REMARK   3   HETEROGEN ATOMS          : 0                                       \n",
      "REMARK   3   SOLVENT ATOMS            : 78                                      \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  B VALUES.                                                           \n",
      "REMARK   3   FROM WILSON PLOT           (A**2) : NULL                           \n",
      "REMARK   3   MEAN B VALUE      (OVERALL, A**2) : 37.50                          \n",
      "REMARK   3   OVERALL ANISOTROPIC B VALUE.                                       \n",
      "REMARK   3    B11 (A**2) : NULL                                                 \n",
      "REMARK   3    B22 (A**2) : NULL                                                 \n",
      "REMARK   3    B33 (A**2) : NULL                                                 \n",
      "REMARK   3    B12 (A**2) : NULL                                                 \n",
      "REMARK   3    B13 (A**2) : NULL                                                 \n",
      "REMARK   3    B23 (A**2) : NULL                                                 \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  ESTIMATED COORDINATE ERROR.                                         \n",
      "REMARK   3   ESD FROM LUZZATI PLOT        (A) : NULL                            \n",
      "REMARK   3   ESD FROM SIGMAA              (A) : NULL                            \n",
      "REMARK   3   LOW RESOLUTION CUTOFF        (A) : NULL                            \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  CROSS-VALIDATED ESTIMATED COORDINATE ERROR.                         \n",
      "REMARK   3   ESD FROM C-V LUZZATI PLOT    (A) : NULL                            \n",
      "REMARK   3   ESD FROM C-V SIGMAA          (A) : NULL                            \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  RMS DEVIATIONS FROM IDEAL VALUES.                                   \n",
      "REMARK   3   BOND LENGTHS                 (A) : 0.006                           \n",
      "REMARK   3   BOND ANGLES            (DEGREES) : 1.303                           \n",
      "REMARK   3   DIHEDRAL ANGLES        (DEGREES) : 21.90                           \n",
      "REMARK   3   IMPROPER ANGLES        (DEGREES) : 0.516                           \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  ISOTROPIC THERMAL MODEL : NULL                                      \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  ISOTROPIC THERMAL FACTOR RESTRAINTS.    RMS    SIGMA                \n",
      "REMARK   3   MAIN-CHAIN BOND              (A**2) : NULL  ; NULL                 \n",
      "REMARK   3   MAIN-CHAIN ANGLE             (A**2) : NULL  ; NULL                 \n",
      "REMARK   3   SIDE-CHAIN BOND              (A**2) : NULL  ; NULL                 \n",
      "REMARK   3   SIDE-CHAIN ANGLE             (A**2) : NULL  ; NULL                 \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  NCS MODEL : NULL                                                    \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  NCS RESTRAINTS.                         RMS   SIGMA/WEIGHT          \n",
      "REMARK   3   GROUP  1  POSITIONAL            (A) : NULL  ; NULL                 \n",
      "REMARK   3   GROUP  1  B-FACTOR           (A**2) : NULL  ; NULL                 \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  PARAMETER FILE  1  : NULL                                           \n",
      "REMARK   3  TOPOLOGY FILE  1   : NULL                                           \n",
      "REMARK   3                                                                      \n",
      "REMARK   3  OTHER REFINEMENT REMARKS: NULL                                      \n",
      "REMARK   4                                                                      \n",
      "REMARK   4 1CE0 COMPLIES WITH FORMAT V. 3.30, 13-JUL-11                         \n",
      "REMARK 100                                                                      \n",
      "REMARK 100 THIS ENTRY HAS BEEN PROCESSED BY RCSB ON 15-MAR-99.                  \n",
      "REMARK 100 THE DEPOSITION ID IS D_1000000636.                                   \n",
      "REMARK 200                                                                      \n",
      "REMARK 200 EXPERIMENTAL DETAILS                                                 \n",
      "REMARK 200  EXPERIMENT TYPE                : X-RAY DIFFRACTION                  \n",
      "REMARK 200  DATE OF DATA COLLECTION        : 15-OCT-98                          \n",
      "REMARK 200  TEMPERATURE           (KELVIN) : 100                                \n",
      "REMARK 200  PH                             : 7.5                                \n",
      "REMARK 200  NUMBER OF CRYSTALS USED        : 1                                  \n",
      "REMARK 200                                                                      \n",
      "REMARK 200  SYNCHROTRON              (Y/N) : Y                                  \n",
      "REMARK 200  RADIATION SOURCE               : NSLS                               \n",
      "REMARK 200  BEAMLINE                       : X12B                               \n",
      "REMARK 200  X-RAY GENERATOR MODEL          : NULL                               \n",
      "REMARK 200  MONOCHROMATIC OR LAUE    (M/L) : M                                  \n",
      "REMARK 200  WAVELENGTH OR RANGE        (A) : 0.978                              \n",
      "REMARK 200  MONOCHROMATOR                  : DOUBLE-CRYSTAL, FIXED-EXIT SI      \n",
      "REMARK 200                                   -III MONOCHROMATOR                 \n",
      "REMARK 200  OPTICS                         : REFOCUSSING MIRROR, RH WORKING     \n",
      "REMARK 200                                   SURFACE                            \n",
      "REMARK 200                                                                      \n",
      "REMARK 200  DETECTOR TYPE                  : IMAGE PLATE                        \n",
      "REMARK 200  DETECTOR MANUFACTURER          : MARRESEARCH                        \n",
      "REMARK 200  INTENSITY-INTEGRATION SOFTWARE : DENZO                              \n",
      "REMARK 200  DATA SCALING SOFTWARE          : SCALEPACK                          \n",
      "REMARK 200                                                                      \n",
      "REMARK 200  NUMBER OF UNIQUE REFLECTIONS   : 6765                               \n",
      "REMARK 200  RESOLUTION RANGE HIGH      (A) : 2.400                              \n",
      "REMARK 200  RESOLUTION RANGE LOW       (A) : 30.000                             \n",
      "REMARK 200  REJECTION CRITERIA  (SIGMA(I)) : -3.000                             \n",
      "REMARK 200                                                                      \n",
      "REMARK 200 OVERALL.                                                             \n",
      "REMARK 200  COMPLETENESS FOR RANGE     (%) : 98.8                               \n",
      "REMARK 200  DATA REDUNDANCY                : 3.300                              \n",
      "REMARK 200  R MERGE                    (I) : 0.03900                            \n",
      "REMARK 200  R SYM                      (I) : 0.03900                            \n",
      "REMARK 200  <I/SIGMA(I)> FOR THE DATA SET  : 20.4000                            \n",
      "REMARK 200                                                                      \n",
      "REMARK 200 IN THE HIGHEST RESOLUTION SHELL.                                     \n",
      "REMARK 200  HIGHEST RESOLUTION SHELL, RANGE HIGH (A) : 2.40                     \n",
      "REMARK 200  HIGHEST RESOLUTION SHELL, RANGE LOW  (A) : 2.49                     \n",
      "REMARK 200  COMPLETENESS FOR SHELL     (%) : 99.8                               \n",
      "REMARK 200  DATA REDUNDANCY IN SHELL       : 3.30                               \n",
      "REMARK 200  R MERGE FOR SHELL          (I) : 0.09700                            \n",
      "REMARK 200  R SYM FOR SHELL            (I) : 0.09700                            \n",
      "REMARK 200  <I/SIGMA(I)> FOR SHELL         : 10.50                              \n",
      "REMARK 200                                                                      \n",
      "REMARK 200 DIFFRACTION PROTOCOL: SINGLE WAVELENGTH                              \n",
      "REMARK 200 METHOD USED TO DETERMINE THE STRUCTURE: MOLECULAR REPLACEMENT        \n",
      "REMARK 200 SOFTWARE USED: AMORE                                                 \n",
      "REMARK 200 STARTING MODEL: PDB ENTRY 1GCM                                       \n",
      "REMARK 200                                                                      \n",
      "REMARK 200 REMARK: NULL                                                         \n",
      "REMARK 280                                                                      \n",
      "REMARK 280 CRYSTAL                                                              \n",
      "REMARK 280 SOLVENT CONTENT, VS   (%): 69.70                                     \n",
      "REMARK 280 MATTHEWS COEFFICIENT, VM (ANGSTROMS**3/DA): 3.27                     \n",
      "REMARK 280                                                                      \n",
      "REMARK 280 CRYSTALLIZATION CONDITIONS: PH 7.5                                   \n",
      "REMARK 290                                                                      \n",
      "REMARK 290 CRYSTALLOGRAPHIC SYMMETRY                                            \n",
      "REMARK 290 SYMMETRY OPERATORS FOR SPACE GROUP: P 61                             \n",
      "REMARK 290                                                                      \n",
      "REMARK 290      SYMOP   SYMMETRY                                                \n",
      "REMARK 290     NNNMMM   OPERATOR                                                \n",
      "REMARK 290       1555   X,Y,Z                                                   \n",
      "REMARK 290       2555   -Y,X-Y,Z+1/3                                            \n",
      "REMARK 290       3555   -X+Y,-X,Z+2/3                                           \n",
      "REMARK 290       4555   -X,-Y,Z+1/2                                             \n",
      "REMARK 290       5555   Y,-X+Y,Z+5/6                                            \n",
      "REMARK 290       6555   X-Y,X,Z+1/6                                             \n",
      "REMARK 290                                                                      \n",
      "REMARK 290     WHERE NNN -> OPERATOR NUMBER                                     \n",
      "REMARK 290           MMM -> TRANSLATION VECTOR                                  \n",
      "REMARK 290                                                                      \n",
      "REMARK 290 CRYSTALLOGRAPHIC SYMMETRY TRANSFORMATIONS                            \n",
      "REMARK 290 THE FOLLOWING TRANSFORMATIONS OPERATE ON THE ATOM/HETATM             \n",
      "REMARK 290 RECORDS IN THIS ENTRY TO PRODUCE CRYSTALLOGRAPHICALLY                \n",
      "REMARK 290 RELATED MOLECULES.                                                   \n",
      "REMARK 290   SMTRY1   1  1.000000  0.000000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   1  0.000000  1.000000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   1  0.000000  0.000000  1.000000        0.00000            \n",
      "REMARK 290   SMTRY1   2 -0.500000 -0.866025  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   2  0.866025 -0.500000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   2  0.000000  0.000000  1.000000       39.75033            \n",
      "REMARK 290   SMTRY1   3 -0.500000  0.866025  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   3 -0.866025 -0.500000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   3  0.000000  0.000000  1.000000       79.50067            \n",
      "REMARK 290   SMTRY1   4 -1.000000  0.000000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   4  0.000000 -1.000000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   4  0.000000  0.000000  1.000000       59.62550            \n",
      "REMARK 290   SMTRY1   5  0.500000  0.866025  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   5 -0.866025  0.500000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   5  0.000000  0.000000  1.000000       99.37583            \n",
      "REMARK 290   SMTRY1   6  0.500000 -0.866025  0.000000        0.00000            \n",
      "REMARK 290   SMTRY2   6  0.866025  0.500000  0.000000        0.00000            \n",
      "REMARK 290   SMTRY3   6  0.000000  0.000000  1.000000       19.87517            \n",
      "REMARK 290                                                                      \n",
      "REMARK 290 REMARK: NULL                                                         \n",
      "REMARK 300                                                                      \n",
      "REMARK 300 BIOMOLECULE: 1                                                       \n",
      "REMARK 300 SEE REMARK 350 FOR THE AUTHOR PROVIDED AND/OR PROGRAM                \n",
      "REMARK 300 GENERATED ASSEMBLY INFORMATION FOR THE STRUCTURE IN                  \n",
      "REMARK 300 THIS ENTRY. THE REMARK MAY ALSO PROVIDE INFORMATION ON               \n",
      "REMARK 300 BURIED SURFACE AREA.                                                 \n",
      "REMARK 350                                                                      \n",
      "REMARK 350 COORDINATES FOR A COMPLETE MULTIMER REPRESENTING THE KNOWN           \n",
      "REMARK 350 BIOLOGICALLY SIGNIFICANT OLIGOMERIZATION STATE OF THE                \n",
      "REMARK 350 MOLECULE CAN BE GENERATED BY APPLYING BIOMT TRANSFORMATIONS          \n",
      "REMARK 350 GIVEN BELOW.  BOTH NON-CRYSTALLOGRAPHIC AND                          \n",
      "REMARK 350 CRYSTALLOGRAPHIC OPERATIONS ARE GIVEN.                               \n",
      "REMARK 350                                                                      \n",
      "REMARK 350 BIOMOLECULE: 1                                                       \n",
      "REMARK 350 AUTHOR DETERMINED BIOLOGICAL UNIT: TRIMERIC                          \n",
      "REMARK 350 SOFTWARE DETERMINED QUATERNARY STRUCTURE: TRIMERIC                   \n",
      "REMARK 350 SOFTWARE USED: PISA                                                  \n",
      "REMARK 350 TOTAL BURIED SURFACE AREA: 4390 ANGSTROM**2                          \n",
      "REMARK 350 SURFACE AREA OF THE COMPLEX: 6720 ANGSTROM**2                        \n",
      "REMARK 350 CHANGE IN SOLVENT FREE ENERGY: -32.0 KCAL/MOL                        \n",
      "REMARK 350 APPLY THE FOLLOWING TO CHAINS: A, B, C                               \n",
      "REMARK 350   BIOMT1   1  1.000000  0.000000  0.000000        0.00000            \n",
      "REMARK 350   BIOMT2   1  0.000000  1.000000  0.000000        0.00000            \n",
      "REMARK 350   BIOMT3   1  0.000000  0.000000  1.000000        0.00000            \n",
      "REMARK 465                                                                      \n",
      "REMARK 465 MISSING RESIDUES                                                     \n",
      "REMARK 465 THE FOLLOWING RESIDUES WERE NOT LOCATED IN THE                       \n",
      "REMARK 465 EXPERIMENT. (M=MODEL NUMBER; RES=RESIDUE NAME; C=CHAIN               \n",
      "REMARK 465 IDENTIFIER; SSSEQ=SEQUENCE NUMBER; I=INSERTION CODE.)                \n",
      "REMARK 465                                                                      \n",
      "REMARK 465   M RES C SSSEQI                                                     \n",
      "REMARK 465     GLU A    36                                                      \n",
      "REMARK 465     ARG A    37                                                      \n",
      "REMARK 465     ARG B    37                                                      \n",
      "REMARK 465     GLU C    36                                                      \n",
      "REMARK 465     ARG C    37                                                      \n",
      "REMARK 470                                                                      \n",
      "REMARK 470 MISSING ATOM                                                         \n",
      "REMARK 470 THE FOLLOWING RESIDUES HAVE MISSING ATOMS (M=MODEL NUMBER;           \n",
      "REMARK 470 RES=RESIDUE NAME; C=CHAIN IDENTIFIER; SSEQ=SEQUENCE NUMBER;          \n",
      "REMARK 470 I=INSERTION CODE):                                                   \n",
      "REMARK 470   M RES CSSEQI  ATOMS                                                \n",
      "REMARK 470     GLN B   4    CG   CD   OE1  NE2                                  \n",
      "REMARK 470     GLN C   4    CG   CD   OE1  NE2                                  \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \n",
      "REMARK 500 SUBTOPIC: COVALENT BOND ANGLES                                       \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 THE STEREOCHEMICAL PARAMETERS OF THE FOLLOWING RESIDUES              \n",
      "REMARK 500 HAVE VALUES WHICH DEVIATE FROM EXPECTED VALUES BY MORE               \n",
      "REMARK 500 THAN 6*RMSD (M=MODEL NUMBER; RES=RESIDUE NAME; C=CHAIN               \n",
      "REMARK 500 IDENTIFIER; SSEQ=SEQUENCE NUMBER; I=INSERTION CODE).                 \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 STANDARD TABLE:                                                      \n",
      "REMARK 500 FORMAT: (10X,I3,1X,A3,1X,A1,I4,A1,3(1X,A4,2X),12X,F5.1)              \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 EXPECTED VALUES PROTEIN: ENGH AND HUBER, 1999                        \n",
      "REMARK 500 EXPECTED VALUES NUCLEIC ACID: CLOWNEY ET AL 1996                     \n",
      "REMARK 500                                                                      \n",
      "REMARK 500  M RES CSSEQI ATM1   ATM2   ATM3                                     \n",
      "REMARK 500    LEU A  33   N   -  CA  -  C   ANGL. DEV. = -22.1 DEGREES          \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 REMARK: NULL                                                         \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \n",
      "REMARK 500 SUBTOPIC: TORSION ANGLES                                             \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 TORSION ANGLES OUTSIDE THE EXPECTED RAMACHANDRAN REGIONS:            \n",
      "REMARK 500 (M=MODEL NUMBER; RES=RESIDUE NAME; C=CHAIN IDENTIFIER;               \n",
      "REMARK 500 SSEQ=SEQUENCE NUMBER; I=INSERTION CODE).                             \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 STANDARD TABLE:                                                      \n",
      "REMARK 500 FORMAT:(10X,I3,1X,A3,1X,A1,I4,A1,4X,F7.2,3X,F7.2)                    \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 EXPECTED VALUES: GJ KLEYWEGT AND TA JONES (1996). PHI/PSI-           \n",
      "REMARK 500 CHOLOGY: RAMACHANDRAN REVISITED. STRUCTURE 4, 1395 - 1400            \n",
      "REMARK 500                                                                      \n",
      "REMARK 500  M RES CSSEQI        PSI       PHI                                   \n",
      "REMARK 500    LEU A  33       17.91   -162.87                                   \n",
      "REMARK 500    LEU A  34       51.60   -172.48                                   \n",
      "REMARK 500                                                                      \n",
      "REMARK 500 REMARK: NULL                                                         \n",
      "DBREF  1CE0 A    1    37  UNP    Q7SIH0   Q7SIH0_9HIV1     1     37             \n",
      "DBREF  1CE0 B    1    37  UNP    Q7SIH0   Q7SIH0_9HIV1     1     37             \n",
      "DBREF  1CE0 C    1    37  UNP    Q7SIH0   Q7SIH0_9HIV1     1     37             \n",
      "SEQRES   1 A   37  ARG LEU LEU GLN ARG ILE LYS GLN GLN GLU ASP LYS LEU          \n",
      "SEQRES   2 A   37  GLU GLU THR LEU SER LYS ILE TYR HIS LEU GLU ASN GLU          \n",
      "SEQRES   3 A   37  ILE ALA ARG VAL LYS LYS LEU LEU GLY GLU ARG                  \n",
      "SEQRES   1 B   37  ARG LEU LEU GLN ARG ILE LYS GLN GLN GLU ASP LYS LEU          \n",
      "SEQRES   2 B   37  GLU GLU THR LEU SER LYS ILE TYR HIS LEU GLU ASN GLU          \n",
      "SEQRES   3 B   37  ILE ALA ARG VAL LYS LYS LEU LEU GLY GLU ARG                  \n",
      "SEQRES   1 C   37  ARG LEU LEU GLN ARG ILE LYS GLN GLN GLU ASP LYS LEU          \n",
      "SEQRES   2 C   37  GLU GLU THR LEU SER LYS ILE TYR HIS LEU GLU ASN GLU          \n",
      "SEQRES   3 C   37  ILE ALA ARG VAL LYS LYS LEU LEU GLY GLU ARG                  \n",
      "FORMUL   4  HOH   *78(H2 O)                                                     \n",
      "HELIX    1   1 LEU A    2  VAL A   30  1                                  29    \n",
      "HELIX    2   2 LEU B    2  LEU B   34  1                                  33    \n",
      "HELIX    3   3 LEU C    2  LEU C   33  1                                  32    \n",
      "CRYST1   50.785   50.785  119.251  90.00  90.00 120.00 P 61         18          \n",
      "ORIGX1      1.000000  0.000000  0.000000        0.00000                         \n",
      "ORIGX2      0.000000  1.000000  0.000000        0.00000                         \n",
      "ORIGX3      0.000000  0.000000  1.000000        0.00000                         \n",
      "SCALE1      0.019691  0.011368  0.000000        0.00000                         \n",
      "SCALE2      0.000000  0.022737  0.000000        0.00000                         \n",
      "SCALE3      0.000000  0.000000  0.008386        0.00000                         \n",
      "ATOM      1  N   ARG A   1      15.069  17.029  -3.952  1.00 62.22           N  \n",
      "ATOM      2  CA  ARG A   1      16.249  17.648  -4.616  1.00 60.42           C  \n",
      "ATOM      3  C   ARG A   1      15.834  18.613  -5.717  1.00 58.92           C  \n",
      "ATOM      4  O   ARG A   1      15.586  19.791  -5.469  1.00 63.04           O  \n",
      "ATOM      5  CB  ARG A   1      17.150  16.565  -5.219  1.00 65.04           C  \n",
      "ATOM      6  CG  ARG A   1      18.629  16.708  -4.879  1.00 68.12           C  \n",
      "ATOM      7  CD  ARG A   1      19.513  16.463  -6.093  1.00 71.02           C  \n",
      "ATOM      8  NE  ARG A   1      19.662  15.040  -6.395  1.00 78.86           N  \n",
      "ATOM      9  CZ  ARG A   1      20.667  14.522  -7.097  1.00 83.30           C  \n",
      "ATOM     10  NH1 ARG A   1      21.623  15.307  -7.576  1.00 85.82           N  \n",
      "ATOM     11  NH2 ARG A   1      20.714  13.216  -7.322  1.00 83.17           N  \n",
      "ATOM     12  N   LEU A   2      15.759  18.088  -6.932  1.00 52.93           N  \n",
      "ATOM     13  CA  LEU A   2      15.413  18.860  -8.121  1.00 54.88           C  \n",
      "ATOM     14  C   LEU A   2      14.487  20.068  -7.989  1.00 54.78           C  \n",
      "ATOM     15  O   LEU A   2      14.804  21.144  -8.501  1.00 55.38           O  \n",
      "ATOM     16  CB  LEU A   2      14.869  17.930  -9.194  1.00 49.35           C  \n",
      "ATOM     17  CG  LEU A   2      15.732  17.981 -10.451  1.00 51.62           C  \n",
      "ATOM     18  CD1 LEU A   2      17.171  17.676 -10.063  1.00 50.86           C  \n",
      "ATOM     19  CD2 LEU A   2      15.222  17.004 -11.495  1.00 54.61           C  \n",
      "ATOM     20  N   LEU A   3      13.349  19.907  -7.325  1.00 52.54           N  \n",
      "ATOM     21  CA  LEU A   3      12.407  21.012  -7.174  1.00 49.30           C  \n",
      "ATOM     22  C   LEU A   3      12.920  22.030  -6.167  1.00 47.18           C  \n",
      "ATOM     23  O   LEU A   3      12.781  23.245  -6.356  1.00 43.66           O  \n",
      "ATOM     24  CB  LEU A   3      11.048  20.493  -6.718  1.00 49.33           C  \n",
      "ATOM     25  CG  LEU A   3       9.853  20.665  -7.656  1.00 54.61           C  \n",
      "ATOM     26  CD1 LEU A   3      10.230  21.482  -8.889  1.00 53.92           C  \n",
      "ATOM     27  CD2 LEU A   3       9.347  19.287  -8.035  1.00 56.35           C  \n",
      "ATOM     28  N   GLN A   4      13.498  21.517  -5.086  1.00 44.21           N  \n",
      "ATOM     29  CA  GLN A   4      14.048  22.354  -4.037  1.00 45.16           C  \n",
      "ATOM     30  C   GLN A   4      15.349  23.009  -4.526  1.00 44.78           C  \n",
      "ATOM     31  O   GLN A   4      15.790  24.021  -3.981  1.00 42.78           O  \n",
      "ATOM     32  CB  GLN A   4      14.277  21.511  -2.772  1.00 45.40           C  \n",
      "ATOM     33  CG  GLN A   4      15.579  21.765  -2.056  1.00 55.58           C  \n",
      "ATOM     34  CD  GLN A   4      15.917  20.657  -1.092  1.00 67.14           C  \n",
      "ATOM     35  OE1 GLN A   4      15.519  20.690   0.071  1.00 68.45           O  \n",
      "ATOM     36  NE2 GLN A   4      16.648  19.657  -1.573  1.00 72.71           N  \n",
      "ATOM     37  N   ARG A   5      15.948  22.444  -5.573  1.00 42.65           N  \n",
      "ATOM     38  CA  ARG A   5      17.187  22.986  -6.118  1.00 39.42           C  \n",
      "ATOM     39  C   ARG A   5      16.927  24.035  -7.187  1.00 37.43           C  \n",
      "ATOM     40  O   ARG A   5      17.747  24.932  -7.390  1.00 37.42           O  \n",
      "ATOM     41  CB  ARG A   5      18.066  21.870  -6.686  1.00 39.81           C  \n",
      "ATOM     42  CG  ARG A   5      19.104  21.350  -5.704  1.00 36.95           C  \n",
      "ATOM     43  CD  ARG A   5      19.729  20.047  -6.197  1.00 39.57           C  \n",
      "ATOM     44  NE  ARG A   5      20.870  20.274  -7.084  1.00 38.44           N  \n",
      "ATOM     45  CZ  ARG A   5      21.414  19.347  -7.874  1.00 37.99           C  \n",
      "ATOM     46  NH1 ARG A   5      20.924  18.117  -7.905  1.00 41.73           N  \n",
      "ATOM     47  NH2 ARG A   5      22.453  19.653  -8.642  1.00 43.35           N  \n",
      "ATOM     48  N   ILE A   6      15.789  23.914  -7.877  1.00 33.84           N  \n",
      "ATOM     49  CA  ILE A   6      15.422  24.870  -8.913  1.00 36.80           C  \n",
      "ATOM     50  C   ILE A   6      14.894  26.127  -8.232  1.00 37.28           C  \n",
      "ATOM     51  O   ILE A   6      15.119  27.240  -8.705  1.00 37.13           O  \n",
      "ATOM     52  CB  ILE A   6      14.339  24.309  -9.893  1.00 35.93           C  \n",
      "ATOM     53  CG1 ILE A   6      14.148  25.286 -11.056  1.00 42.48           C  \n",
      "ATOM     54  CG2 ILE A   6      13.015  24.110  -9.187  1.00 40.62           C  \n",
      "ATOM     55  CD1 ILE A   6      13.419  24.711 -12.230  1.00 47.66           C  \n",
      "ATOM     56  N   LYS A   7      14.208  25.946  -7.108  1.00 36.64           N  \n",
      "ATOM     57  CA  LYS A   7      13.669  27.067  -6.359  1.00 37.84           C  \n",
      "ATOM     58  C   LYS A   7      14.828  27.890  -5.802  1.00 35.27           C  \n",
      "ATOM     59  O   LYS A   7      14.851  29.113  -5.920  1.00 31.00           O  \n",
      "ATOM     60  CB  LYS A   7      12.790  26.566  -5.212  1.00 42.71           C  \n",
      "ATOM     61  CG  LYS A   7      11.711  27.546  -4.778  1.00 44.76           C  \n",
      "ATOM     62  CD  LYS A   7      12.314  28.858  -4.315  1.00 48.22           C  \n",
      "ATOM     63  CE  LYS A   7      11.330  29.653  -3.484  1.00 49.54           C  \n",
      "ATOM     64  NZ  LYS A   7      11.517  29.419  -2.026  1.00 55.17           N  \n",
      "ATOM     65  N   GLN A   8      15.785  27.194  -5.202  1.00 31.72           N  \n",
      "ATOM     66  CA  GLN A   8      16.964  27.804  -4.611  1.00 32.12           C  \n",
      "ATOM     67  C   GLN A   8      17.795  28.545  -5.673  1.00 33.13           C  \n",
      "ATOM     68  O   GLN A   8      18.418  29.574  -5.389  1.00 30.63           O  \n",
      "ATOM     69  CB  GLN A   8      17.791  26.707  -3.935  1.00 35.70           C  \n",
      "ATOM     70  CG  GLN A   8      19.287  26.944  -3.905  1.00 50.40           C  \n",
      "ATOM     71  CD  GLN A   8      20.033  25.837  -3.188  1.00 60.12           C  \n",
      "ATOM     72  OE1 GLN A   8      20.335  24.796  -3.775  1.00 64.96           O  \n",
      "ATOM     73  NE2 GLN A   8      20.333  26.055  -1.911  1.00 61.91           N  \n",
      "ATOM     74  N   GLN A   9      17.783  28.023  -6.900  1.00 31.08           N  \n",
      "ATOM     75  CA  GLN A   9      18.523  28.622  -8.003  1.00 30.03           C  \n",
      "ATOM     76  C   GLN A   9      17.792  29.866  -8.494  1.00 32.54           C  \n",
      "ATOM     77  O   GLN A   9      18.417  30.864  -8.863  1.00 32.34           O  \n",
      "ATOM     78  CB  GLN A   9      18.674  27.609  -9.147  1.00 26.92           C  \n",
      "ATOM     79  CG  GLN A   9      19.420  28.131 -10.376  1.00 36.40           C  \n",
      "ATOM     80  CD  GLN A   9      20.937  28.126 -10.201  1.00 40.61           C  \n",
      "ATOM     81  OE1 GLN A   9      21.453  27.810  -9.121  1.00 40.09           O  \n",
      "ATOM     82  NE2 GLN A   9      21.657  28.477 -11.265  1.00 35.42           N  \n",
      "ATOM     83  N   GLU A  10      16.463  29.799  -8.491  1.00 30.51           N  \n",
      "ATOM     84  CA  GLU A  10      15.625  30.912  -8.918  1.00 30.43           C  \n",
      "ATOM     85  C   GLU A  10      15.765  32.088  -7.950  1.00 25.72           C  \n",
      "ATOM     86  O   GLU A  10      15.792  33.246  -8.370  1.00 23.93           O  \n",
      "ATOM     87  CB  GLU A  10      14.162  30.467  -9.001  1.00 31.37           C  \n",
      "ATOM     88  CG  GLU A  10      13.882  29.495 -10.143  1.00 40.47           C  \n",
      "ATOM     89  CD  GLU A  10      12.466  28.918 -10.132  1.00 41.74           C  \n",
      "ATOM     90  OE1 GLU A  10      11.826  28.882  -9.053  1.00 36.66           O  \n",
      "ATOM     91  OE2 GLU A  10      11.996  28.499 -11.212  1.00 35.37           O  \n",
      "ATOM     92  N   ASP A  11      15.850  31.785  -6.659  1.00 25.09           N  \n",
      "ATOM     93  CA  ASP A  11      16.005  32.813  -5.638  1.00 28.86           C  \n",
      "ATOM     94  C   ASP A  11      17.383  33.435  -5.813  1.00 30.36           C  \n",
      "ATOM     95  O   ASP A  11      17.534  34.655  -5.856  1.00 27.20           O  \n",
      "ATOM     96  CB  ASP A  11      15.920  32.209  -4.233  1.00 28.81           C  \n",
      "ATOM     97  CG  ASP A  11      14.529  31.711  -3.902  1.00 35.84           C  \n",
      "ATOM     98  OD1 ASP A  11      13.571  32.075  -4.621  1.00 34.15           O  \n",
      "ATOM     99  OD2 ASP A  11      14.406  30.950  -2.917  1.00 37.63           O  \n",
      "ATOM    100  N   LYS A  12      18.382  32.561  -5.924  1.00 29.25           N  \n",
      "ATOM    101  CA  LYS A  12      19.757  32.972  -6.096  1.00 28.56           C  \n",
      "ATOM    102  C   LYS A  12      19.916  33.900  -7.288  1.00 27.38           C  \n",
      "ATOM    103  O   LYS A  12      20.570  34.938  -7.189  1.00 27.15           O  \n",
      "ATOM    104  CB  LYS A  12      20.649  31.743  -6.282  1.00 34.37           C  \n",
      "ATOM    105  CG  LYS A  12      21.869  31.741  -5.385  1.00 45.31           C  \n",
      "ATOM    106  CD  LYS A  12      23.142  31.505  -6.178  1.00 51.48           C  \n",
      "ATOM    107  CE  LYS A  12      24.114  30.627  -5.391  1.00 58.13           C  \n",
      "ATOM    108  NZ  LYS A  12      24.678  31.324  -4.189  1.00 63.09           N  \n",
      "ATOM    109  N   LEU A  13      19.316  33.530  -8.416  1.00 23.72           N  \n",
      "ATOM    110  CA  LEU A  13      19.404  34.338  -9.626  1.00 24.06           C  \n",
      "ATOM    111  C   LEU A  13      18.693  35.678  -9.487  1.00 27.16           C  \n",
      "ATOM    112  O   LEU A  13      19.077  36.661 -10.122  1.00 29.62           O  \n",
      "ATOM    113  CB  LEU A  13      18.832  33.583 -10.829  1.00 14.29           C  \n",
      "ATOM    114  CG  LEU A  13      19.066  34.243 -12.192  1.00 25.46           C  \n",
      "ATOM    115  CD1 LEU A  13      20.527  34.666 -12.333  1.00 22.49           C  \n",
      "ATOM    116  CD2 LEU A  13      18.687  33.271 -13.298  1.00 26.74           C  \n",
      "ATOM    117  N   GLU A  14      17.647  35.718  -8.668  1.00 29.10           N  \n",
      "ATOM    118  CA  GLU A  14      16.920  36.959  -8.463  1.00 31.25           C  \n",
      "ATOM    119  C   GLU A  14      17.770  37.910  -7.619  1.00 29.51           C  \n",
      "ATOM    120  O   GLU A  14      17.743  39.129  -7.811  1.00 30.87           O  \n",
      "ATOM    121  CB  GLU A  14      15.588  36.707  -7.758  1.00 34.82           C  \n",
      "ATOM    122  CG  GLU A  14      14.531  37.740  -8.114  1.00 47.64           C  \n",
      "ATOM    123  CD  GLU A  14      14.391  37.920  -9.619  1.00 55.73           C  \n",
      "ATOM    124  OE1 GLU A  14      14.374  36.893 -10.326  1.00 59.70           O  \n",
      "ATOM    125  OE2 GLU A  14      14.302  39.077 -10.091  1.00 62.70           O  \n",
      "ATOM    126  N   GLU A  15      18.517  37.339  -6.680  1.00 24.11           N  \n",
      "ATOM    127  CA  GLU A  15      19.401  38.101  -5.807  1.00 27.06           C  \n",
      "ATOM    128  C   GLU A  15      20.564  38.630  -6.623  1.00 24.47           C  \n",
      "ATOM    129  O   GLU A  15      21.054  39.734  -6.387  1.00 25.14           O  \n",
      "ATOM    130  CB  GLU A  15      19.956  37.206  -4.709  1.00 32.87           C  \n",
      "ATOM    131  CG  GLU A  15      19.163  37.221  -3.437  1.00 50.73           C  \n",
      "ATOM    132  CD  GLU A  15      19.401  35.974  -2.622  1.00 59.45           C  \n",
      "ATOM    133  OE1 GLU A  15      20.557  35.764  -2.191  1.00 62.80           O  \n",
      "ATOM    134  OE2 GLU A  15      18.438  35.205  -2.419  1.00 53.83           O  \n",
      "ATOM    135  N   THR A  16      21.011  37.814  -7.576  1.00 19.94           N  \n",
      "ATOM    136  CA  THR A  16      22.125  38.155  -8.444  1.00 20.97           C  \n",
      "ATOM    137  C   THR A  16      21.743  39.321  -9.344  1.00 22.53           C  \n",
      "ATOM    138  O   THR A  16      22.504  40.275  -9.516  1.00 21.48           O  \n",
      "ATOM    139  CB  THR A  16      22.533  36.938  -9.314  1.00 17.66           C  \n",
      "ATOM    140  OG1 THR A  16      23.143  35.939  -8.489  1.00 20.20           O  \n",
      "ATOM    141  CG2 THR A  16      23.516  37.354 -10.399  1.00 11.68           C  \n",
      "ATOM    142  N   LEU A  17      20.544  39.244  -9.906  1.00 23.19           N  \n",
      "ATOM    143  CA  LEU A  17      20.037  40.282 -10.787  1.00 25.02           C  \n",
      "ATOM    144  C   LEU A  17      19.948  41.622 -10.053  1.00 25.82           C  \n",
      "ATOM    145  O   LEU A  17      20.154  42.688 -10.648  1.00 29.03           O  \n",
      "ATOM    146  CB  LEU A  17      18.664  39.871 -11.332  1.00 20.16           C  \n",
      "ATOM    147  CG  LEU A  17      18.719  38.950 -12.550  1.00 25.54           C  \n",
      "ATOM    148  CD1 LEU A  17      17.300  38.596 -12.950  1.00 27.03           C  \n",
      "ATOM    149  CD2 LEU A  17      19.460  39.632 -13.696  1.00 26.52           C  \n",
      "ATOM    150  N   SER A  18      19.637  41.562  -8.761  1.00 23.39           N  \n",
      "ATOM    151  CA  SER A  18      19.543  42.759  -7.937  1.00 24.70           C  \n",
      "ATOM    152  C   SER A  18      20.935  43.380  -7.828  1.00 24.60           C  \n",
      "ATOM    153  O   SER A  18      21.090  44.601  -7.872  1.00 23.92           O  \n",
      "ATOM    154  CB  SER A  18      19.029  42.399  -6.543  1.00 25.30           C  \n",
      "ATOM    155  OG  SER A  18      19.422  43.367  -5.584  1.00 42.17           O  \n",
      "ATOM    156  N   LYS A  19      21.944  42.522  -7.691  1.00 21.92           N  \n",
      "ATOM    157  CA  LYS A  19      23.325  42.962  -7.586  1.00 17.48           C  \n",
      "ATOM    158  C   LYS A  19      23.720  43.656  -8.880  1.00 18.81           C  \n",
      "ATOM    159  O   LYS A  19      24.365  44.706  -8.865  1.00 18.81           O  \n",
      "ATOM    160  CB  LYS A  19      24.253  41.762  -7.333  1.00 21.05           C  \n",
      "ATOM    161  CG  LYS A  19      24.243  41.266  -5.895  1.00 19.44           C  \n",
      "ATOM    162  CD  LYS A  19      25.466  40.394  -5.594  1.00 23.55           C  \n",
      "ATOM    163  CE  LYS A  19      25.315  39.646  -4.261  1.00 14.49           C  \n",
      "ATOM    164  NZ  LYS A  19      25.501  40.517  -3.060  1.00 15.20           N  \n",
      "ATOM    165  N   ILE A  20      23.341  43.055 -10.004  1.00 16.55           N  \n",
      "ATOM    166  CA  ILE A  20      23.644  43.604 -11.324  1.00 20.41           C  \n",
      "ATOM    167  C   ILE A  20      23.019  44.985 -11.478  1.00 27.19           C  \n",
      "ATOM    168  O   ILE A  20      23.630  45.911 -12.010  1.00 28.18           O  \n",
      "ATOM    169  CB  ILE A  20      23.080  42.700 -12.438  1.00 19.31           C  \n",
      "ATOM    170  CG1 ILE A  20      23.795  41.349 -12.424  1.00 20.20           C  \n",
      "ATOM    171  CG2 ILE A  20      23.192  43.391 -13.796  1.00 17.33           C  \n",
      "ATOM    172  CD1 ILE A  20      25.281  41.440 -12.624  1.00 29.61           C  \n",
      "ATOM    173  N   TYR A  21      21.787  45.106 -11.006  1.00 26.54           N  \n",
      "ATOM    174  CA  TYR A  21      21.044  46.347 -11.072  1.00 26.34           C  \n",
      "ATOM    175  C   TYR A  21      21.785  47.478 -10.348  1.00 22.42           C  \n",
      "ATOM    176  O   TYR A  21      21.964  48.570 -10.892  1.00 21.12           O  \n",
      "ATOM    177  CB  TYR A  21      19.662  46.120 -10.457  1.00 28.26           C  \n",
      "ATOM    178  CG  TYR A  21      18.924  47.380 -10.110  1.00 30.33           C  \n",
      "ATOM    179  CD1 TYR A  21      18.389  48.193 -11.109  1.00 33.79           C  \n",
      "ATOM    180  CD2 TYR A  21      18.756  47.764  -8.780  1.00 28.20           C  \n",
      "ATOM    181  CE1 TYR A  21      17.706  49.364 -10.795  1.00 39.88           C  \n",
      "ATOM    182  CE2 TYR A  21      18.075  48.935  -8.451  1.00 33.33           C  \n",
      "ATOM    183  CZ  TYR A  21      17.550  49.728  -9.464  1.00 38.30           C  \n",
      "ATOM    184  OH  TYR A  21      16.874  50.884  -9.152  1.00 45.75           O  \n",
      "ATOM    185  N   HIS A  22      22.215  47.199  -9.119  1.00 16.28           N  \n",
      "ATOM    186  CA  HIS A  22      22.937  48.161  -8.299  1.00 20.63           C  \n",
      "ATOM    187  C   HIS A  22      24.275  48.500  -8.936  1.00 22.44           C  \n",
      "ATOM    188  O   HIS A  22      24.765  49.621  -8.816  1.00 26.74           O  \n",
      "ATOM    189  CB  HIS A  22      23.161  47.594  -6.894  1.00 13.38           C  \n",
      "ATOM    190  CG  HIS A  22      21.891  47.408  -6.116  1.00 28.12           C  \n",
      "ATOM    191  ND1 HIS A  22      20.964  48.409  -5.959  1.00 27.88           N  \n",
      "ATOM    192  CD2 HIS A  22      21.412  46.334  -5.440  1.00 26.81           C  \n",
      "ATOM    193  CE1 HIS A  22      19.961  47.969  -5.213  1.00 30.88           C  \n",
      "ATOM    194  NE2 HIS A  22      20.211  46.714  -4.884  1.00 32.89           N  \n",
      "ATOM    195  N   LEU A  23      24.858  47.516  -9.615  1.00 21.42           N  \n",
      "ATOM    196  CA  LEU A  23      26.135  47.680 -10.288  1.00 23.52           C  \n",
      "ATOM    197  C   LEU A  23      25.945  48.643 -11.452  1.00 23.09           C  \n",
      "ATOM    198  O   LEU A  23      26.726  49.576 -11.637  1.00 22.96           O  \n",
      "ATOM    199  CB  LEU A  23      26.637  46.321 -10.799  1.00 22.76           C  \n",
      "ATOM    200  CG  LEU A  23      28.115  45.932 -10.683  1.00 20.31           C  \n",
      "ATOM    201  CD1 LEU A  23      28.760  46.554  -9.450  1.00 15.62           C  \n",
      "ATOM    202  CD2 LEU A  23      28.205  44.413 -10.623  1.00 30.09           C  \n",
      "ATOM    203  N   GLU A  24      24.900  48.415 -12.240  1.00 20.61           N  \n",
      "ATOM    204  CA  GLU A  24      24.602  49.270 -13.379  1.00 25.60           C  \n",
      "ATOM    205  C   GLU A  24      24.386  50.709 -12.920  1.00 27.42           C  \n",
      "ATOM    206  O   GLU A  24      24.783  51.661 -13.595  1.00 27.92           O  \n",
      "ATOM    207  CB  GLU A  24      23.350  48.771 -14.097  1.00 22.74           C  \n",
      "ATOM    208  CG  GLU A  24      23.621  47.670 -15.114  1.00 35.65           C  \n",
      "ATOM    209  CD  GLU A  24      22.355  46.930 -15.537  1.00 39.38           C  \n",
      "ATOM    210  OE1 GLU A  24      21.305  47.135 -14.884  1.00 34.55           O  \n",
      "ATOM    211  OE2 GLU A  24      22.414  46.154 -16.523  1.00 40.99           O  \n",
      "ATOM    212  N   ASN A  25      23.754  50.855 -11.763  1.00 23.70           N  \n",
      "ATOM    213  CA  ASN A  25      23.470  52.165 -11.200  1.00 28.93           C  \n",
      "ATOM    214  C   ASN A  25      24.744  52.867 -10.730  1.00 28.67           C  \n",
      "ATOM    215  O   ASN A  25      24.919  54.072 -10.932  1.00 31.58           O  \n",
      "ATOM    216  CB  ASN A  25      22.497  52.022 -10.029  1.00 28.47           C  \n",
      "ATOM    217  CG  ASN A  25      21.086  51.699 -10.481  1.00 37.30           C  \n",
      "ATOM    218  OD1 ASN A  25      20.730  51.895 -11.644  1.00 39.56           O  \n",
      "ATOM    219  ND2 ASN A  25      20.269  51.204  -9.560  1.00 38.11           N  \n",
      "ATOM    220  N   GLU A  26      25.632  52.094 -10.113  1.00 23.27           N  \n",
      "ATOM    221  CA  GLU A  26      26.891  52.608  -9.600  1.00 23.36           C  \n",
      "ATOM    222  C   GLU A  26      27.816  53.077 -10.717  1.00 27.70           C  \n",
      "ATOM    223  O   GLU A  26      28.464  54.123 -10.605  1.00 29.20           O  \n",
      "ATOM    224  CB  GLU A  26      27.590  51.531  -8.775  1.00 22.07           C  \n",
      "ATOM    225  CG  GLU A  26      27.051  51.408  -7.366  1.00 31.84           C  \n",
      "ATOM    226  CD  GLU A  26      27.244  52.679  -6.556  1.00 38.24           C  \n",
      "ATOM    227  OE1 GLU A  26      28.259  53.380  -6.772  1.00 39.82           O  \n",
      "ATOM    228  OE2 GLU A  26      26.380  52.973  -5.700  1.00 45.27           O  \n",
      "ATOM    229  N   ILE A  27      27.867  52.300 -11.797  1.00 26.94           N  \n",
      "ATOM    230  CA  ILE A  27      28.703  52.608 -12.946  1.00 32.04           C  \n",
      "ATOM    231  C   ILE A  27      28.112  53.751 -13.749  1.00 35.79           C  \n",
      "ATOM    232  O   ILE A  27      28.829  54.437 -14.474  1.00 39.60           O  \n",
      "ATOM    233  CB  ILE A  27      28.833  51.397 -13.880  1.00 33.47           C  \n",
      "ATOM    234  CG1 ILE A  27      29.355  50.193 -13.102  1.00 44.75           C  \n",
      "ATOM    235  CG2 ILE A  27      29.768  51.719 -15.029  1.00 31.14           C  \n",
      "ATOM    236  CD1 ILE A  27      28.963  48.869 -13.727  1.00 50.64           C  \n",
      "ATOM    237  N   ALA A  28      26.800  53.943 -13.631  1.00 38.76           N  \n",
      "ATOM    238  CA  ALA A  28      26.110  55.012 -14.347  1.00 35.42           C  \n",
      "ATOM    239  C   ALA A  28      26.315  56.336 -13.618  1.00 38.07           C  \n",
      "ATOM    240  O   ALA A  28      26.239  57.413 -14.216  1.00 38.44           O  \n",
      "ATOM    241  CB  ALA A  28      24.625  54.709 -14.453  1.00 30.26           C  \n",
      "ATOM    242  N   ARG A  29      26.574  56.239 -12.321  1.00 40.96           N  \n",
      "ATOM    243  CA  ARG A  29      26.802  57.410 -11.497  1.00 43.18           C  \n",
      "ATOM    244  C   ARG A  29      28.279  57.783 -11.568  1.00 41.63           C  \n",
      "ATOM    245  O   ARG A  29      28.609  58.928 -11.856  1.00 41.79           O  \n",
      "ATOM    246  CB  ARG A  29      26.400  57.123 -10.051  1.00 46.90           C  \n",
      "ATOM    247  CG  ARG A  29      26.890  58.157  -9.057  1.00 58.28           C  \n",
      "ATOM    248  CD  ARG A  29      26.587  57.737  -7.628  1.00 70.03           C  \n",
      "ATOM    249  NE  ARG A  29      27.701  57.006  -7.029  1.00 76.18           N  \n",
      "ATOM    250  CZ  ARG A  29      28.898  57.532  -6.793  1.00 74.14           C  \n",
      "ATOM    251  NH1 ARG A  29      29.139  58.797  -7.109  1.00 73.87           N  \n",
      "ATOM    252  NH2 ARG A  29      29.853  56.789  -6.250  1.00 75.03           N  \n",
      "ATOM    253  N   VAL A  30      29.158  56.810 -11.320  1.00 41.18           N  \n",
      "ATOM    254  CA  VAL A  30      30.602  57.035 -11.352  1.00 43.53           C  \n",
      "ATOM    255  C   VAL A  30      30.979  57.635 -12.697  1.00 49.31           C  \n",
      "ATOM    256  O   VAL A  30      31.682  58.644 -12.771  1.00 54.97           O  \n",
      "ATOM    257  CB  VAL A  30      31.398  55.705 -11.160  1.00 35.56           C  \n",
      "ATOM    258  CG1 VAL A  30      32.729  55.743 -11.926  1.00 37.10           C  \n",
      "ATOM    259  CG2 VAL A  30      31.667  55.482  -9.682  1.00 34.90           C  \n",
      "ATOM    260  N   LYS A  31      30.511  56.998 -13.758  1.00 52.73           N  \n",
      "ATOM    261  CA  LYS A  31      30.785  57.450 -15.106  1.00 58.55           C  \n",
      "ATOM    262  C   LYS A  31      29.716  58.459 -15.454  1.00 61.35           C  \n",
      "ATOM    263  O   LYS A  31      28.601  58.385 -14.948  1.00 62.77           O  \n",
      "ATOM    264  CB  LYS A  31      30.756  56.254 -16.060  1.00 59.31           C  \n",
      "ATOM    265  CG  LYS A  31      29.843  56.381 -17.264  1.00 52.47           C  \n",
      "ATOM    266  CD  LYS A  31      30.103  55.256 -18.249  1.00 50.94           C  \n",
      "ATOM    267  CE  LYS A  31      29.093  55.304 -19.377  1.00 53.05           C  \n",
      "ATOM    268  NZ  LYS A  31      27.690  55.495 -18.898  1.00 55.04           N  \n",
      "ATOM    269  N   LYS A  32      30.048  59.397 -16.326  1.00 64.57           N  \n",
      "ATOM    270  CA  LYS A  32      29.101  60.425 -16.700  1.00 65.77           C  \n",
      "ATOM    271  C   LYS A  32      28.718  61.117 -15.398  1.00 64.45           C  \n",
      "ATOM    272  O   LYS A  32      27.559  61.210 -14.972  1.00 59.42           O  \n",
      "ATOM    273  CB  LYS A  32      27.922  59.804 -17.445  1.00 68.22           C  \n",
      "ATOM    274  CG  LYS A  32      28.340  59.302 -18.838  1.00 66.37           C  \n",
      "ATOM    275  CD  LYS A  32      29.489  60.130 -19.445  1.00 60.54           C  \n",
      "ATOM    276  CE  LYS A  32      30.863  59.518 -19.162  1.00 61.35           C  \n",
      "ATOM    277  NZ  LYS A  32      31.643  59.352 -20.419  1.00 59.34           N  \n",
      "ATOM    278  N   LEU A  33      29.815  61.538 -14.782  1.00 63.21           N  \n",
      "ATOM    279  CA  LEU A  33      29.974  62.277 -13.545  1.00 58.92           C  \n",
      "ATOM    280  C   LEU A  33      31.404  62.625 -13.970  1.00 59.89           C  \n",
      "ATOM    281  O   LEU A  33      32.271  62.989 -13.172  1.00 58.24           O  \n",
      "ATOM    282  CB  LEU A  33      29.980  61.358 -12.328  1.00 52.22           C  \n",
      "ATOM    283  CG  LEU A  33      30.678  61.943 -11.092  1.00 50.32           C  \n",
      "ATOM    284  CD1 LEU A  33      29.725  61.926  -9.902  1.00 49.71           C  \n",
      "ATOM    285  CD2 LEU A  33      31.953  61.157 -10.787  1.00 43.22           C  \n",
      "ATOM    286  N   LEU A  34      31.603  62.451 -15.278  1.00 61.11           N  \n",
      "ATOM    287  CA  LEU A  34      32.853  62.692 -15.987  1.00 62.70           C  \n",
      "ATOM    288  C   LEU A  34      32.590  62.549 -17.492  1.00 65.93           C  \n",
      "ATOM    289  O   LEU A  34      33.289  61.812 -18.182  1.00 67.29           O  \n",
      "ATOM    290  CB  LEU A  34      33.939  61.685 -15.560  1.00 59.21           C  \n",
      "ATOM    291  CG  LEU A  34      33.660  60.235 -15.109  1.00 57.06           C  \n",
      "ATOM    292  CD1 LEU A  34      33.242  59.365 -16.286  1.00 57.10           C  \n",
      "ATOM    293  CD2 LEU A  34      34.935  59.677 -14.480  1.00 55.45           C  \n",
      "ATOM    294  N   GLY A  35      31.577  63.246 -17.997  1.00 67.18           N  \n",
      "ATOM    295  CA  GLY A  35      31.274  63.157 -19.415  1.00 71.02           C  \n",
      "ATOM    296  C   GLY A  35      29.997  63.862 -19.819  1.00 73.72           C  \n",
      "ATOM    297  O   GLY A  35      29.486  63.591 -20.929  1.00 77.04           O  \n",
      "TER     298      GLY A  35                                                      \n",
      "ATOM    299  N   ARG B   1      18.509  14.152 -17.855  1.00 50.36           N  \n",
      "ATOM    300  CA  ARG B   1      18.757  15.503 -18.428  1.00 45.77           C  \n",
      "ATOM    301  C   ARG B   1      18.657  16.610 -17.382  1.00 44.02           C  \n",
      "ATOM    302  O   ARG B   1      19.612  17.356 -17.165  1.00 43.98           O  \n",
      "ATOM    303  CB  ARG B   1      17.776  15.773 -19.575  1.00 48.14           C  \n",
      "ATOM    304  CG  ARG B   1      18.133  16.961 -20.459  1.00 53.97           C  \n",
      "ATOM    305  CD  ARG B   1      16.980  17.956 -20.514  1.00 61.11           C  \n",
      "ATOM    306  NE  ARG B   1      15.947  17.567 -21.475  1.00 72.38           N  \n",
      "ATOM    307  CZ  ARG B   1      14.706  18.046 -21.475  1.00 77.24           C  \n",
      "ATOM    308  NH1 ARG B   1      14.337  18.933 -20.560  1.00 83.55           N  \n",
      "ATOM    309  NH2 ARG B   1      13.833  17.639 -22.390  1.00 78.59           N  \n",
      "ATOM    310  N   LEU B   2      17.506  16.687 -16.720  1.00 39.26           N  \n",
      "ATOM    311  CA  LEU B   2      17.245  17.707 -15.714  1.00 35.42           C  \n",
      "ATOM    312  C   LEU B   2      18.206  17.803 -14.547  1.00 34.91           C  \n",
      "ATOM    313  O   LEU B   2      18.529  18.894 -14.108  1.00 34.01           O  \n",
      "ATOM    314  CB  LEU B   2      15.831  17.541 -15.180  1.00 32.40           C  \n",
      "ATOM    315  CG  LEU B   2      14.747  18.335 -15.934  1.00 38.30           C  \n",
      "ATOM    316  CD1 LEU B   2      15.097  18.381 -17.441  1.00 34.74           C  \n",
      "ATOM    317  CD2 LEU B   2      13.407  17.596 -15.784  1.00 44.02           C  \n",
      "ATOM    318  N   LEU B   3      18.661  16.666 -14.034  1.00 34.33           N  \n",
      "ATOM    319  CA  LEU B   3      19.598  16.656 -12.909  1.00 38.04           C  \n",
      "ATOM    320  C   LEU B   3      20.932  17.304 -13.295  1.00 36.32           C  \n",
      "ATOM    321  O   LEU B   3      21.527  18.058 -12.518  1.00 33.64           O  \n",
      "ATOM    322  CB  LEU B   3      19.864  15.215 -12.450  1.00 46.70           C  \n",
      "ATOM    323  CG  LEU B   3      19.149  14.720 -11.191  1.00 54.87           C  \n",
      "ATOM    324  CD1 LEU B   3      19.565  15.575 -10.007  1.00 54.24           C  \n",
      "ATOM    325  CD2 LEU B   3      17.645  14.777 -11.392  1.00 57.67           C  \n",
      "ATOM    326  N   GLN B   4      21.389  17.002 -14.508  1.00 31.24           N  \n",
      "ATOM    327  CA  GLN B   4      22.649  17.507 -15.042  1.00 36.61           C  \n",
      "ATOM    328  C   GLN B   4      22.579  18.993 -15.399  1.00 36.99           C  \n",
      "ATOM    329  O   GLN B   4      23.508  19.753 -15.117  1.00 42.05           O  \n",
      "ATOM    330  CB  GLN B   4      23.041  16.699 -16.262  1.00 31.57           C  \n",
      "ATOM    331  N   ARG B   5      21.478  19.382 -16.041  1.00 37.68           N  \n",
      "ATOM    332  CA  ARG B   5      21.206  20.764 -16.449  1.00 34.06           C  \n",
      "ATOM    333  C   ARG B   5      21.225  21.677 -15.219  1.00 29.91           C  \n",
      "ATOM    334  O   ARG B   5      21.760  22.789 -15.252  1.00 32.81           O  \n",
      "ATOM    335  CB  ARG B   5      19.815  20.847 -17.062  1.00 27.76           C  \n",
      "ATOM    336  CG  ARG B   5      19.693  20.835 -18.562  1.00 30.28           C  \n",
      "ATOM    337  CD  ARG B   5      18.214  21.084 -18.853  1.00 43.08           C  \n",
      "ATOM    338  NE  ARG B   5      17.862  21.036 -20.269  1.00 52.34           N  \n",
      "ATOM    339  CZ  ARG B   5      16.923  21.783 -20.827  1.00 57.45           C  \n",
      "ATOM    340  NH1 ARG B   5      16.212  22.642 -20.101  1.00 56.75           N  \n",
      "ATOM    341  NH2 ARG B   5      16.675  21.678 -22.128  1.00 67.11           N  \n",
      "ATOM    342  N   ILE B   6      20.595  21.198 -14.148  1.00 23.51           N  \n",
      "ATOM    343  CA  ILE B   6      20.501  21.922 -12.889  1.00 22.11           C  \n",
      "ATOM    344  C   ILE B   6      21.868  22.057 -12.227  1.00 26.65           C  \n",
      "ATOM    345  O   ILE B   6      22.175  23.093 -11.637  1.00 33.14           O  \n",
      "ATOM    346  CB  ILE B   6      19.556  21.194 -11.915  1.00 24.14           C  \n",
      "ATOM    347  CG1 ILE B   6      18.103  21.379 -12.366  1.00 35.26           C  \n",
      "ATOM    348  CG2 ILE B   6      19.779  21.691 -10.497  1.00 25.04           C  \n",
      "ATOM    349  CD1 ILE B   6      17.416  22.589 -11.790  1.00 42.46           C  \n",
      "ATOM    350  N   LYS B   7      22.678  21.004 -12.316  1.00 23.72           N  \n",
      "ATOM    351  CA  LYS B   7      24.011  21.007 -11.732  1.00 25.07           C  \n",
      "ATOM    352  C   LYS B   7      24.873  21.998 -12.504  1.00 27.83           C  \n",
      "ATOM    353  O   LYS B   7      25.686  22.717 -11.926  1.00 26.39           O  \n",
      "ATOM    354  CB  LYS B   7      24.634  19.612 -11.805  1.00 29.07           C  \n",
      "ATOM    355  CG  LYS B   7      26.138  19.600 -11.582  1.00 32.91           C  \n",
      "ATOM    356  CD  LYS B   7      26.481  19.563 -10.092  1.00 45.08           C  \n",
      "ATOM    357  CE  LYS B   7      26.299  20.922  -9.417  1.00 45.26           C  \n",
      "ATOM    358  NZ  LYS B   7      27.050  21.010  -8.135  1.00 46.39           N  \n",
      "ATOM    359  N   GLN B   8      24.673  22.020 -13.818  1.00 26.36           N  \n",
      "ATOM    360  CA  GLN B   8      25.393  22.908 -14.712  1.00 29.86           C  \n",
      "ATOM    361  C   GLN B   8      24.979  24.361 -14.461  1.00 29.21           C  \n",
      "ATOM    362  O   GLN B   8      25.810  25.269 -14.500  1.00 26.97           O  \n",
      "ATOM    363  CB  GLN B   8      25.100  22.518 -16.163  1.00 32.90           C  \n",
      "ATOM    364  CG  GLN B   8      25.706  23.451 -17.197  1.00 53.10           C  \n",
      "ATOM    365  CD  GLN B   8      24.696  23.906 -18.241  1.00 66.00           C  \n",
      "ATOM    366  OE1 GLN B   8      24.268  23.119 -19.087  1.00 72.55           O  \n",
      "ATOM    367  NE2 GLN B   8      24.315  25.179 -18.179  1.00 67.14           N  \n",
      "ATOM    368  N   GLN B   9      23.692  24.574 -14.202  1.00 23.57           N  \n",
      "ATOM    369  CA  GLN B   9      23.179  25.908 -13.946  1.00 21.88           C  \n",
      "ATOM    370  C   GLN B   9      23.694  26.429 -12.614  1.00 21.67           C  \n",
      "ATOM    371  O   GLN B   9      23.922  27.623 -12.454  1.00 25.76           O  \n",
      "ATOM    372  CB  GLN B   9      21.652  25.895 -13.926  1.00 25.78           C  \n",
      "ATOM    373  CG  GLN B   9      21.018  25.987 -15.298  1.00 37.18           C  \n",
      "ATOM    374  CD  GLN B   9      21.538  27.173 -16.088  1.00 44.80           C  \n",
      "ATOM    375  OE1 GLN B   9      21.357  28.330 -15.685  1.00 42.56           O  \n",
      "ATOM    376  NE2 GLN B   9      22.195  26.895 -17.217  1.00 33.23           N  \n",
      "ATOM    377  N   GLU B  10      23.874  25.525 -11.657  1.00 19.19           N  \n",
      "ATOM    378  CA  GLU B  10      24.370  25.902 -10.338  1.00 25.20           C  \n",
      "ATOM    379  C   GLU B  10      25.845  26.266 -10.410  1.00 26.04           C  \n",
      "ATOM    380  O   GLU B  10      26.317  27.131  -9.672  1.00 26.28           O  \n",
      "ATOM    381  CB  GLU B  10      24.189  24.760  -9.343  1.00 25.46           C  \n",
      "ATOM    382  CG  GLU B  10      22.764  24.567  -8.885  1.00 29.95           C  \n",
      "ATOM    383  CD  GLU B  10      22.520  23.179  -8.319  1.00 31.89           C  \n",
      "ATOM    384  OE1 GLU B  10      23.445  22.333  -8.372  1.00 25.37           O  \n",
      "ATOM    385  OE2 GLU B  10      21.401  22.941  -7.817  1.00 31.19           O  \n",
      "ATOM    386  N   ASP B  11      26.569  25.589 -11.296  1.00 29.14           N  \n",
      "ATOM    387  CA  ASP B  11      27.990  25.851 -11.475  1.00 28.23           C  \n",
      "ATOM    388  C   ASP B  11      28.156  27.182 -12.195  1.00 28.11           C  \n",
      "ATOM    389  O   ASP B  11      29.009  27.992 -11.836  1.00 30.06           O  \n",
      "ATOM    390  CB  ASP B  11      28.640  24.738 -12.301  1.00 25.28           C  \n",
      "ATOM    391  CG  ASP B  11      28.869  23.469 -11.500  1.00 28.64           C  \n",
      "ATOM    392  OD1 ASP B  11      28.729  23.504 -10.255  1.00 27.17           O  \n",
      "ATOM    393  OD2 ASP B  11      29.185  22.435 -12.120  1.00 32.54           O  \n",
      "ATOM    394  N   LYS B  12      27.324  27.399 -13.208  1.00 22.55           N  \n",
      "ATOM    395  CA  LYS B  12      27.371  28.619 -13.992  1.00 24.48           C  \n",
      "ATOM    396  C   LYS B  12      27.009  29.838 -13.154  1.00 25.89           C  \n",
      "ATOM    397  O   LYS B  12      27.608  30.900 -13.306  1.00 27.67           O  \n",
      "ATOM    398  CB  LYS B  12      26.429  28.512 -15.196  1.00 25.02           C  \n",
      "ATOM    399  CG  LYS B  12      26.481  29.713 -16.125  1.00 37.36           C  \n",
      "ATOM    400  CD  LYS B  12      27.616  29.576 -17.132  1.00 49.69           C  \n",
      "ATOM    401  CE  LYS B  12      28.738  30.571 -16.852  1.00 50.16           C  \n",
      "ATOM    402  NZ  LYS B  12      29.337  31.112 -18.108  1.00 47.58           N  \n",
      "ATOM    403  N   LEU B  13      26.035  29.683 -12.265  1.00 22.94           N  \n",
      "ATOM    404  CA  LEU B  13      25.608  30.785 -11.421  1.00 23.98           C  \n",
      "ATOM    405  C   LEU B  13      26.696  31.180 -10.420  1.00 25.44           C  \n",
      "ATOM    406  O   LEU B  13      26.858  32.362 -10.096  1.00 29.51           O  \n",
      "ATOM    407  CB  LEU B  13      24.317  30.415 -10.680  1.00 20.23           C  \n",
      "ATOM    408  CG  LEU B  13      23.700  31.487  -9.774  1.00 29.29           C  \n",
      "ATOM    409  CD1 LEU B  13      23.825  32.859 -10.426  1.00 28.41           C  \n",
      "ATOM    410  CD2 LEU B  13      22.248  31.161  -9.506  1.00 21.49           C  \n",
      "ATOM    411  N   GLU B  14      27.438  30.193  -9.931  1.00 23.33           N  \n",
      "ATOM    412  CA  GLU B  14      28.506  30.450  -8.974  1.00 25.70           C  \n",
      "ATOM    413  C   GLU B  14      29.649  31.206  -9.654  1.00 25.86           C  \n",
      "ATOM    414  O   GLU B  14      30.282  32.082  -9.058  1.00 29.51           O  \n",
      "ATOM    415  CB  GLU B  14      29.021  29.130  -8.397  1.00 25.23           C  \n",
      "ATOM    416  CG  GLU B  14      28.238  28.636  -7.189  1.00 48.41           C  \n",
      "ATOM    417  CD  GLU B  14      28.141  29.684  -6.089  1.00 59.01           C  \n",
      "ATOM    418  OE1 GLU B  14      29.178  29.980  -5.454  1.00 66.02           O  \n",
      "ATOM    419  OE2 GLU B  14      27.029  30.209  -5.866  1.00 61.59           O  \n",
      "ATOM    420  N   GLU B  15      29.912  30.857 -10.911  1.00 23.50           N  \n",
      "ATOM    421  CA  GLU B  15      30.964  31.507 -11.670  1.00 25.78           C  \n",
      "ATOM    422  C   GLU B  15      30.492  32.901 -12.055  1.00 26.03           C  \n",
      "ATOM    423  O   GLU B  15      31.290  33.831 -12.191  1.00 28.00           O  \n",
      "ATOM    424  CB  GLU B  15      31.292  30.695 -12.922  1.00 27.42           C  \n",
      "ATOM    425  CG  GLU B  15      31.889  29.327 -12.610  1.00 48.20           C  \n",
      "ATOM    426  CD  GLU B  15      31.850  28.376 -13.803  1.00 54.99           C  \n",
      "ATOM    427  OE1 GLU B  15      31.136  28.678 -14.783  1.00 57.78           O  \n",
      "ATOM    428  OE2 GLU B  15      32.536  27.330 -13.757  1.00 57.79           O  \n",
      "ATOM    429  N   THR B  16      29.183  33.037 -12.224  1.00 22.60           N  \n",
      "ATOM    430  CA  THR B  16      28.574  34.306 -12.578  1.00 23.42           C  \n",
      "ATOM    431  C   THR B  16      28.690  35.263 -11.392  1.00 25.01           C  \n",
      "ATOM    432  O   THR B  16      28.944  36.457 -11.566  1.00 21.41           O  \n",
      "ATOM    433  CB  THR B  16      27.079  34.120 -12.934  1.00 26.54           C  \n",
      "ATOM    434  OG1 THR B  16      26.964  33.371 -14.152  1.00 26.40           O  \n",
      "ATOM    435  CG2 THR B  16      26.389  35.477 -13.097  1.00 23.66           C  \n",
      "ATOM    436  N   LEU B  17      28.507  34.727 -10.187  1.00 18.63           N  \n",
      "ATOM    437  CA  LEU B  17      28.585  35.519  -8.965  1.00 19.40           C  \n",
      "ATOM    438  C   LEU B  17      30.017  35.968  -8.753  1.00 23.88           C  \n",
      "ATOM    439  O   LEU B  17      30.283  37.082  -8.302  1.00 24.28           O  \n",
      "ATOM    440  CB  LEU B  17      28.117  34.695  -7.765  1.00 15.60           C  \n",
      "ATOM    441  CG  LEU B  17      26.607  34.738  -7.527  1.00 26.76           C  \n",
      "ATOM    442  CD1 LEU B  17      26.210  33.642  -6.546  1.00 27.22           C  \n",
      "ATOM    443  CD2 LEU B  17      26.222  36.116  -7.001  1.00 28.05           C  \n",
      "ATOM    444  N   SER B  18      30.945  35.083  -9.083  1.00 22.20           N  \n",
      "ATOM    445  CA  SER B  18      32.350  35.384  -8.947  1.00 22.84           C  \n",
      "ATOM    446  C   SER B  18      32.670  36.630  -9.786  1.00 20.15           C  \n",
      "ATOM    447  O   SER B  18      33.361  37.544  -9.331  1.00 20.63           O  \n",
      "ATOM    448  CB  SER B  18      33.168  34.184  -9.417  1.00 21.82           C  \n",
      "ATOM    449  OG  SER B  18      34.458  34.208  -8.848  1.00 39.07           O  \n",
      "ATOM    450  N   LYS B  19      32.147  36.656 -11.009  1.00 19.69           N  \n",
      "ATOM    451  CA  LYS B  19      32.332  37.773 -11.934  1.00 25.75           C  \n",
      "ATOM    452  C   LYS B  19      31.822  39.067 -11.313  1.00 26.39           C  \n",
      "ATOM    453  O   LYS B  19      32.499  40.095 -11.360  1.00 28.40           O  \n",
      "ATOM    454  CB  LYS B  19      31.563  37.521 -13.238  1.00 29.38           C  \n",
      "ATOM    455  CG  LYS B  19      32.301  36.686 -14.268  1.00 40.00           C  \n",
      "ATOM    456  CD  LYS B  19      31.411  36.385 -15.474  1.00 44.79           C  \n",
      "ATOM    457  CE  LYS B  19      31.763  35.037 -16.078  1.00 48.16           C  \n",
      "ATOM    458  NZ  LYS B  19      33.109  35.084 -16.707  1.00 43.17           N  \n",
      "ATOM    459  N   ILE B  20      30.615  39.006 -10.752  1.00 24.11           N  \n",
      "ATOM    460  CA  ILE B  20      29.979  40.155 -10.120  1.00 22.86           C  \n",
      "ATOM    461  C   ILE B  20      30.799  40.663  -8.939  1.00 22.30           C  \n",
      "ATOM    462  O   ILE B  20      30.909  41.874  -8.731  1.00 24.95           O  \n",
      "ATOM    463  CB  ILE B  20      28.547  39.801  -9.637  1.00 13.11           C  \n",
      "ATOM    464  CG1 ILE B  20      27.591  39.821 -10.827  1.00 13.45           C  \n",
      "ATOM    465  CG2 ILE B  20      28.067  40.780  -8.574  1.00 11.22           C  \n",
      "ATOM    466  CD1 ILE B  20      26.656  38.652 -10.861  1.00 18.76           C  \n",
      "ATOM    467  N   TYR B  21      31.371  39.740  -8.167  1.00 20.17           N  \n",
      "ATOM    468  CA  TYR B  21      32.189  40.107  -7.013  1.00 22.13           C  \n",
      "ATOM    469  C   TYR B  21      33.375  40.919  -7.500  1.00 22.39           C  \n",
      "ATOM    470  O   TYR B  21      33.717  41.953  -6.920  1.00 21.56           O  \n",
      "ATOM    471  CB  TYR B  21      32.690  38.856  -6.289  1.00 18.47           C  \n",
      "ATOM    472  CG  TYR B  21      31.600  38.093  -5.579  1.00 16.68           C  \n",
      "ATOM    473  CD1 TYR B  21      30.353  38.675  -5.330  1.00 22.08           C  \n",
      "ATOM    474  CD2 TYR B  21      31.806  36.778  -5.172  1.00 20.38           C  \n",
      "ATOM    475  CE1 TYR B  21      29.342  37.966  -4.686  1.00 26.02           C  \n",
      "ATOM    476  CE2 TYR B  21      30.800  36.058  -4.524  1.00 24.13           C  \n",
      "ATOM    477  CZ  TYR B  21      29.569  36.659  -4.286  1.00 28.77           C  \n",
      "ATOM    478  OH  TYR B  21      28.577  35.945  -3.655  1.00 26.58           O  \n",
      "ATOM    479  N   HIS B  22      33.989  40.444  -8.579  1.00 20.09           N  \n",
      "ATOM    480  CA  HIS B  22      35.137  41.110  -9.156  1.00 22.60           C  \n",
      "ATOM    481  C   HIS B  22      34.778  42.517  -9.607  1.00 23.19           C  \n",
      "ATOM    482  O   HIS B  22      35.488  43.472  -9.303  1.00 25.15           O  \n",
      "ATOM    483  CB  HIS B  22      35.674  40.311 -10.344  1.00 21.46           C  \n",
      "ATOM    484  CG  HIS B  22      36.727  41.037 -11.117  1.00 23.23           C  \n",
      "ATOM    485  ND1 HIS B  22      36.537  41.462 -12.418  1.00 25.53           N  \n",
      "ATOM    486  CD2 HIS B  22      37.970  41.443 -10.769  1.00 19.59           C  \n",
      "ATOM    487  CE1 HIS B  22      37.619  42.098 -12.833  1.00 22.74           C  \n",
      "ATOM    488  NE2 HIS B  22      38.503  42.099 -11.849  1.00 23.74           N  \n",
      "ATOM    489  N   LEU B  23      33.675  42.644 -10.338  1.00 18.64           N  \n",
      "ATOM    490  CA  LEU B  23      33.226  43.938 -10.831  1.00 19.46           C  \n",
      "ATOM    491  C   LEU B  23      32.906  44.886  -9.677  1.00 20.71           C  \n",
      "ATOM    492  O   LEU B  23      33.208  46.079  -9.743  1.00 21.87           O  \n",
      "ATOM    493  CB  LEU B  23      31.990  43.766 -11.721  1.00 20.42           C  \n",
      "ATOM    494  CG  LEU B  23      32.163  43.099 -13.091  1.00 22.83           C  \n",
      "ATOM    495  CD1 LEU B  23      30.839  43.103 -13.861  1.00 22.44           C  \n",
      "ATOM    496  CD2 LEU B  23      33.229  43.846 -13.865  1.00 29.36           C  \n",
      "ATOM    497  N   GLU B  24      32.291  44.351  -8.625  1.00 17.66           N  \n",
      "ATOM    498  CA  GLU B  24      31.938  45.143  -7.452  1.00 20.64           C  \n",
      "ATOM    499  C   GLU B  24      33.220  45.690  -6.839  1.00 19.95           C  \n",
      "ATOM    500  O   GLU B  24      33.256  46.788  -6.285  1.00 21.32           O  \n",
      "ATOM    501  CB  GLU B  24      31.204  44.270  -6.432  1.00 11.49           C  \n",
      "ATOM    502  CG  GLU B  24      29.796  43.847  -6.867  1.00 15.73           C  \n",
      "ATOM    503  CD  GLU B  24      28.982  43.238  -5.730  1.00 16.56           C  \n",
      "ATOM    504  OE1 GLU B  24      29.587  42.581  -4.859  1.00 15.79           O  \n",
      "ATOM    505  OE2 GLU B  24      27.743  43.410  -5.710  1.00 15.03           O  \n",
      "ATOM    506  N   ASN B  25      34.281  44.903  -6.954  1.00 20.46           N  \n",
      "ATOM    507  CA  ASN B  25      35.569  45.281  -6.424  1.00 21.41           C  \n",
      "ATOM    508  C   ASN B  25      36.214  46.331  -7.320  1.00 22.51           C  \n",
      "ATOM    509  O   ASN B  25      36.902  47.229  -6.839  1.00 18.99           O  \n",
      "ATOM    510  CB  ASN B  25      36.463  44.049  -6.328  1.00 27.25           C  \n",
      "ATOM    511  CG  ASN B  25      36.510  43.487  -4.928  1.00 41.57           C  \n",
      "ATOM    512  OD1 ASN B  25      37.288  43.945  -4.089  1.00 52.51           O  \n",
      "ATOM    513  ND2 ASN B  25      35.673  42.490  -4.656  1.00 42.93           N  \n",
      "ATOM    514  N   GLU B  26      35.987  46.193  -8.625  1.00 19.70           N  \n",
      "ATOM    515  CA  GLU B  26      36.507  47.112  -9.623  1.00 21.77           C  \n",
      "ATOM    516  C   GLU B  26      35.852  48.464  -9.396  1.00 28.09           C  \n",
      "ATOM    517  O   GLU B  26      36.522  49.499  -9.395  1.00 27.80           O  \n",
      "ATOM    518  CB  GLU B  26      36.155  46.618 -11.026  1.00 22.94           C  \n",
      "ATOM    519  CG  GLU B  26      37.019  45.483 -11.512  1.00 30.00           C  \n",
      "ATOM    520  CD  GLU B  26      38.485  45.809 -11.418  1.00 33.50           C  \n",
      "ATOM    521  OE1 GLU B  26      38.970  46.573 -12.280  1.00 37.08           O  \n",
      "ATOM    522  OE2 GLU B  26      39.143  45.312 -10.475  1.00 42.27           O  \n",
      "ATOM    523  N   ILE B  27      34.534  48.426  -9.202  1.00 26.07           N  \n",
      "ATOM    524  CA  ILE B  27      33.715  49.609  -8.963  1.00 23.79           C  \n",
      "ATOM    525  C   ILE B  27      34.133  50.300  -7.666  1.00 25.82           C  \n",
      "ATOM    526  O   ILE B  27      34.116  51.523  -7.583  1.00 25.43           O  \n",
      "ATOM    527  CB  ILE B  27      32.208  49.225  -8.880  1.00 24.08           C  \n",
      "ATOM    528  CG1 ILE B  27      31.504  49.534 -10.210  1.00 29.99           C  \n",
      "ATOM    529  CG2 ILE B  27      31.535  49.970  -7.749  1.00 22.39           C  \n",
      "ATOM    530  CD1 ILE B  27      31.460  51.008 -10.580  1.00 31.59           C  \n",
      "ATOM    531  N   ALA B  28      34.489  49.511  -6.655  1.00 22.52           N  \n",
      "ATOM    532  CA  ALA B  28      34.928  50.041  -5.365  1.00 24.99           C  \n",
      "ATOM    533  C   ALA B  28      36.247  50.808  -5.494  1.00 31.55           C  \n",
      "ATOM    534  O   ALA B  28      36.399  51.889  -4.916  1.00 33.81           O  \n",
      "ATOM    535  CB  ALA B  28      35.095  48.905  -4.369  1.00 20.24           C  \n",
      "ATOM    536  N   ARG B  29      37.208  50.240  -6.239  1.00 31.51           N  \n",
      "ATOM    537  CA  ARG B  29      38.510  50.885  -6.440  1.00 32.69           C  \n",
      "ATOM    538  C   ARG B  29      38.359  52.157  -7.251  1.00 32.36           C  \n",
      "ATOM    539  O   ARG B  29      39.019  53.162  -6.978  1.00 37.17           O  \n",
      "ATOM    540  CB  ARG B  29      39.517  50.006  -7.200  1.00 35.55           C  \n",
      "ATOM    541  CG  ARG B  29      39.149  48.577  -7.433  1.00 55.50           C  \n",
      "ATOM    542  CD  ARG B  29      40.240  47.680  -6.879  1.00 57.54           C  \n",
      "ATOM    543  NE  ARG B  29      41.072  47.103  -7.937  1.00 62.26           N  \n",
      "ATOM    544  CZ  ARG B  29      41.720  45.948  -7.845  1.00 62.48           C  \n",
      "ATOM    545  NH1 ARG B  29      41.672  45.212  -6.740  1.00 63.96           N  \n",
      "ATOM    546  NH2 ARG B  29      42.434  45.533  -8.881  1.00 62.27           N  \n",
      "ATOM    547  N   VAL B  30      37.512  52.095  -8.275  1.00 29.71           N  \n",
      "ATOM    548  CA  VAL B  30      37.265  53.232  -9.152  1.00 32.76           C  \n",
      "ATOM    549  C   VAL B  30      36.642  54.379  -8.363  1.00 36.44           C  \n",
      "ATOM    550  O   VAL B  30      36.858  55.550  -8.675  1.00 40.16           O  \n",
      "ATOM    551  CB  VAL B  30      36.304  52.859 -10.304  1.00 32.29           C  \n",
      "ATOM    552  CG1 VAL B  30      35.635  54.116 -10.858  1.00 34.80           C  \n",
      "ATOM    553  CG2 VAL B  30      37.070  52.134 -11.407  1.00 33.46           C  \n",
      "ATOM    554  N   LYS B  31      35.860  54.031  -7.353  1.00 37.67           N  \n",
      "ATOM    555  CA  LYS B  31      35.208  55.026  -6.531  1.00 38.52           C  \n",
      "ATOM    556  C   LYS B  31      36.216  55.520  -5.522  1.00 42.87           C  \n",
      "ATOM    557  O   LYS B  31      36.204  56.683  -5.144  1.00 48.61           O  \n",
      "ATOM    558  CB  LYS B  31      34.000  54.422  -5.814  1.00 35.43           C  \n",
      "ATOM    559  CG  LYS B  31      32.722  54.403  -6.638  1.00 33.58           C  \n",
      "ATOM    560  CD  LYS B  31      31.665  53.510  -5.988  1.00 33.24           C  \n",
      "ATOM    561  CE  LYS B  31      31.160  54.086  -4.663  1.00 36.16           C  \n",
      "ATOM    562  NZ  LYS B  31      29.667  54.104  -4.561  1.00 38.76           N  \n",
      "ATOM    563  N   LYS B  32      37.095  54.628  -5.088  1.00 47.08           N  \n",
      "ATOM    564  CA  LYS B  32      38.127  54.983  -4.125  1.00 51.45           C  \n",
      "ATOM    565  C   LYS B  32      39.061  55.960  -4.839  1.00 55.24           C  \n",
      "ATOM    566  O   LYS B  32      39.586  56.909  -4.248  1.00 56.90           O  \n",
      "ATOM    567  CB  LYS B  32      38.890  53.728  -3.700  1.00 51.79           C  \n",
      "ATOM    568  CG  LYS B  32      39.195  53.669  -2.222  1.00 53.75           C  \n",
      "ATOM    569  CD  LYS B  32      39.691  52.285  -1.828  1.00 62.14           C  \n",
      "ATOM    570  CE  LYS B  32      38.604  51.227  -2.007  1.00 67.80           C  \n",
      "ATOM    571  NZ  LYS B  32      39.131  49.987  -2.665  1.00 62.55           N  \n",
      "ATOM    572  N   LEU B  33      39.244  55.707  -6.129  1.00 55.22           N  \n",
      "ATOM    573  CA  LEU B  33      40.071  56.520  -7.000  1.00 53.54           C  \n",
      "ATOM    574  C   LEU B  33      39.476  57.928  -7.021  1.00 56.78           C  \n",
      "ATOM    575  O   LEU B  33      40.125  58.896  -6.620  1.00 59.44           O  \n",
      "ATOM    576  CB  LEU B  33      40.049  55.906  -8.401  1.00 53.18           C  \n",
      "ATOM    577  CG  LEU B  33      40.980  56.439  -9.478  1.00 54.38           C  \n",
      "ATOM    578  CD1 LEU B  33      41.105  55.400 -10.579  1.00 51.35           C  \n",
      "ATOM    579  CD2 LEU B  33      40.427  57.738 -10.026  1.00 62.26           C  \n",
      "ATOM    580  N   LEU B  34      38.231  58.020  -7.482  1.00 56.27           N  \n",
      "ATOM    581  CA  LEU B  34      37.501  59.281  -7.573  1.00 55.23           C  \n",
      "ATOM    582  C   LEU B  34      37.446  60.020  -6.241  1.00 57.87           C  \n",
      "ATOM    583  O   LEU B  34      37.161  61.218  -6.204  1.00 61.22           O  \n",
      "ATOM    584  CB  LEU B  34      36.076  59.033  -8.059  1.00 51.23           C  \n",
      "ATOM    585  CG  LEU B  34      35.894  58.754  -9.549  1.00 52.33           C  \n",
      "ATOM    586  CD1 LEU B  34      34.533  58.117  -9.758  1.00 58.27           C  \n",
      "ATOM    587  CD2 LEU B  34      36.007  60.040 -10.352  1.00 55.80           C  \n",
      "ATOM    588  N   GLY B  35      37.687  59.300  -5.148  1.00 62.38           N  \n",
      "ATOM    589  CA  GLY B  35      37.688  59.929  -3.840  1.00 68.66           C  \n",
      "ATOM    590  C   GLY B  35      36.512  59.675  -2.920  1.00 72.67           C  \n",
      "ATOM    591  O   GLY B  35      36.455  60.242  -1.829  1.00 73.82           O  \n",
      "ATOM    592  N   GLU B  36      35.573  58.835  -3.341  1.00 78.25           N  \n",
      "ATOM    593  CA  GLU B  36      34.403  58.533  -2.517  1.00 82.50           C  \n",
      "ATOM    594  C   GLU B  36      34.438  57.109  -1.963  1.00 85.54           C  \n",
      "ATOM    595  O   GLU B  36      33.543  56.770  -1.159  1.00 87.95           O  \n",
      "ATOM    596  CB  GLU B  36      33.122  58.725  -3.329  1.00 81.58           C  \n",
      "ATOM    597  CG  GLU B  36      33.253  58.300  -4.778  1.00 79.82           C  \n",
      "ATOM    598  CD  GLU B  36      32.509  59.225  -5.717  1.00 80.53           C  \n",
      "ATOM    599  OE1 GLU B  36      33.062  60.288  -6.079  1.00 83.18           O  \n",
      "ATOM    600  OE2 GLU B  36      31.370  58.891  -6.095  1.00 75.68           O  \n",
      "TER     601      GLU B  36                                                      \n",
      "ATOM    602  N   ARG C   1       5.861  23.420 -14.309  1.00 53.63           N  \n",
      "ATOM    603  CA  ARG C   1       6.837  22.741 -15.205  1.00 53.64           C  \n",
      "ATOM    604  C   ARG C   1       8.287  22.843 -14.725  1.00 51.04           C  \n",
      "ATOM    605  O   ARG C   1       8.887  23.917 -14.730  1.00 57.13           O  \n",
      "ATOM    606  CB  ARG C   1       6.730  23.319 -16.623  1.00 56.09           C  \n",
      "ATOM    607  CG  ARG C   1       7.337  22.432 -17.704  1.00 58.95           C  \n",
      "ATOM    608  CD  ARG C   1       6.821  22.818 -19.087  1.00 62.91           C  \n",
      "ATOM    609  NE  ARG C   1       7.546  22.152 -20.169  1.00 74.62           N  \n",
      "ATOM    610  CZ  ARG C   1       7.356  22.402 -21.463  1.00 78.75           C  \n",
      "ATOM    611  NH1 ARG C   1       6.460  23.306 -21.840  1.00 82.48           N  \n",
      "ATOM    612  NH2 ARG C   1       8.056  21.746 -22.382  1.00 77.93           N  \n",
      "ATOM    613  N   LEU C   2       8.846  21.706 -14.315  1.00 47.72           N  \n",
      "ATOM    614  CA  LEU C   2      10.227  21.642 -13.858  1.00 44.53           C  \n",
      "ATOM    615  C   LEU C   2      11.083  21.822 -15.103  1.00 45.77           C  \n",
      "ATOM    616  O   LEU C   2      12.244  22.240 -15.039  1.00 50.84           O  \n",
      "ATOM    617  CB  LEU C   2      10.503  20.271 -13.239  1.00 42.74           C  \n",
      "ATOM    618  CG  LEU C   2      11.669  20.096 -12.270  1.00 48.36           C  \n",
      "ATOM    619  CD1 LEU C   2      11.156  19.382 -11.044  1.00 48.68           C  \n",
      "ATOM    620  CD2 LEU C   2      12.799  19.293 -12.915  1.00 49.95           C  \n",
      "ATOM    621  N   LEU C   3      10.484  21.497 -16.243  1.00 44.16           N  \n",
      "ATOM    622  CA  LEU C   3      11.141  21.595 -17.533  1.00 44.63           C  \n",
      "ATOM    623  C   LEU C   3      11.207  23.039 -18.026  1.00 46.45           C  \n",
      "ATOM    624  O   LEU C   3      12.263  23.509 -18.458  1.00 47.70           O  \n",
      "ATOM    625  CB  LEU C   3      10.385  20.740 -18.545  1.00 46.15           C  \n",
      "ATOM    626  CG  LEU C   3      11.165  19.715 -19.369  1.00 48.64           C  \n",
      "ATOM    627  CD1 LEU C   3      12.150  18.960 -18.490  1.00 43.75           C  \n",
      "ATOM    628  CD2 LEU C   3      10.181  18.753 -20.020  1.00 48.64           C  \n",
      "ATOM    629  N   GLN C   4      10.073  23.734 -17.972  1.00 42.69           N  \n",
      "ATOM    630  CA  GLN C   4       9.988  25.129 -18.404  1.00 44.71           C  \n",
      "ATOM    631  C   GLN C   4      10.703  26.065 -17.428  1.00 43.97           C  \n",
      "ATOM    632  O   GLN C   4      11.312  27.059 -17.833  1.00 49.49           O  \n",
      "ATOM    633  CB  GLN C   4       8.519  25.553 -18.552  1.00 42.85           C  \n",
      "ATOM    634  N   ARG C   5      10.629  25.747 -16.140  1.00 39.74           N  \n",
      "ATOM    635  CA  ARG C   5      11.272  26.562 -15.124  1.00 38.60           C  \n",
      "ATOM    636  C   ARG C   5      12.785  26.576 -15.328  1.00 41.27           C  \n",
      "ATOM    637  O   ARG C   5      13.433  27.609 -15.151  1.00 42.84           O  \n",
      "ATOM    638  CB  ARG C   5      10.958  26.014 -13.734  1.00 37.66           C  \n",
      "ATOM    639  CG  ARG C   5       9.603  26.392 -13.181  1.00 40.87           C  \n",
      "ATOM    640  CD  ARG C   5       9.326  25.578 -11.918  1.00 46.62           C  \n",
      "ATOM    641  NE  ARG C   5      10.011  26.143 -10.756  1.00 48.80           N  \n",
      "ATOM    642  CZ  ARG C   5       9.967  25.645  -9.512  1.00 48.36           C  \n",
      "ATOM    643  NH1 ARG C   5       9.257  24.562  -9.235  1.00 48.28           N  \n",
      "ATOM    644  NH2 ARG C   5      10.629  26.259  -8.545  1.00 47.22           N  \n",
      "ATOM    645  N   ILE C   6      13.337  25.425 -15.705  1.00 37.19           N  \n",
      "ATOM    646  CA  ILE C   6      14.771  25.281 -15.924  1.00 40.03           C  \n",
      "ATOM    647  C   ILE C   6      15.243  25.989 -17.194  1.00 37.26           C  \n",
      "ATOM    648  O   ILE C   6      16.306  26.611 -17.207  1.00 40.43           O  \n",
      "ATOM    649  CB  ILE C   6      15.172  23.777 -15.970  1.00 41.24           C  \n",
      "ATOM    650  CG1 ILE C   6      16.256  23.500 -14.921  1.00 50.98           C  \n",
      "ATOM    651  CG2 ILE C   6      15.642  23.382 -17.356  1.00 38.75           C  \n",
      "ATOM    652  CD1 ILE C   6      17.657  23.945 -15.311  1.00 56.17           C  \n",
      "ATOM    653  N   LYS C   7      14.455  25.878 -18.259  1.00 35.36           N  \n",
      "ATOM    654  CA  LYS C   7      14.778  26.527 -19.522  1.00 34.73           C  \n",
      "ATOM    655  C   LYS C   7      14.706  28.030 -19.293  1.00 32.38           C  \n",
      "ATOM    656  O   LYS C   7      15.442  28.806 -19.895  1.00 31.46           O  \n",
      "ATOM    657  CB  LYS C   7      13.765  26.131 -20.591  1.00 32.39           C  \n",
      "ATOM    658  CG  LYS C   7      14.398  25.756 -21.909  1.00 39.76           C  \n",
      "ATOM    659  CD  LYS C   7      14.683  26.991 -22.747  1.00 43.51           C  \n",
      "ATOM    660  CE  LYS C   7      13.558  27.257 -23.733  1.00 53.02           C  \n",
      "ATOM    661  NZ  LYS C   7      13.721  28.561 -24.423  1.00 44.58           N  \n",
      "ATOM    662  N   GLN C   8      13.790  28.412 -18.412  1.00 29.49           N  \n",
      "ATOM    663  CA  GLN C   8      13.557  29.793 -18.033  1.00 31.63           C  \n",
      "ATOM    664  C   GLN C   8      14.793  30.337 -17.314  1.00 32.38           C  \n",
      "ATOM    665  O   GLN C   8      15.285  31.426 -17.612  1.00 27.15           O  \n",
      "ATOM    666  CB  GLN C   8      12.357  29.836 -17.095  1.00 37.55           C  \n",
      "ATOM    667  CG  GLN C   8      11.713  31.184 -16.918  1.00 53.29           C  \n",
      "ATOM    668  CD  GLN C   8      10.311  31.072 -16.348  1.00 65.99           C  \n",
      "ATOM    669  OE1 GLN C   8      10.106  30.529 -15.256  1.00 68.91           O  \n",
      "ATOM    670  NE2 GLN C   8       9.332  31.583 -17.088  1.00 69.82           N  \n",
      "ATOM    671  N   GLN C   9      15.271  29.551 -16.356  1.00 30.31           N  \n",
      "ATOM    672  CA  GLN C   9      16.436  29.876 -15.555  1.00 28.14           C  \n",
      "ATOM    673  C   GLN C   9      17.669  29.978 -16.440  1.00 27.32           C  \n",
      "ATOM    674  O   GLN C   9      18.485  30.888 -16.292  1.00 28.39           O  \n",
      "ATOM    675  CB  GLN C   9      16.640  28.785 -14.508  1.00 32.97           C  \n",
      "ATOM    676  CG  GLN C   9      16.139  29.158 -13.138  1.00 44.16           C  \n",
      "ATOM    677  CD  GLN C   9      16.915  30.318 -12.556  1.00 48.78           C  \n",
      "ATOM    678  OE1 GLN C   9      18.122  30.214 -12.309  1.00 47.83           O  \n",
      "ATOM    679  NE2 GLN C   9      16.230  31.436 -12.337  1.00 44.65           N  \n",
      "ATOM    680  N   GLU C  10      17.794  29.027 -17.360  1.00 20.10           N  \n",
      "ATOM    681  CA  GLU C  10      18.909  28.981 -18.293  1.00 20.50           C  \n",
      "ATOM    682  C   GLU C  10      18.955  30.234 -19.142  1.00 23.48           C  \n",
      "ATOM    683  O   GLU C  10      20.010  30.842 -19.314  1.00 28.34           O  \n",
      "ATOM    684  CB  GLU C  10      18.778  27.766 -19.201  1.00 23.87           C  \n",
      "ATOM    685  CG  GLU C  10      19.235  26.488 -18.533  1.00 34.00           C  \n",
      "ATOM    686  CD  GLU C  10      18.763  25.235 -19.244  1.00 36.99           C  \n",
      "ATOM    687  OE1 GLU C  10      17.914  25.339 -20.158  1.00 35.90           O  \n",
      "ATOM    688  OE2 GLU C  10      19.248  24.141 -18.884  1.00 31.92           O  \n",
      "ATOM    689  N   ASP C  11      17.799  30.608 -19.679  1.00 20.31           N  \n",
      "ATOM    690  CA  ASP C  11      17.690  31.789 -20.516  1.00 23.49           C  \n",
      "ATOM    691  C   ASP C  11      18.023  33.022 -19.685  1.00 23.84           C  \n",
      "ATOM    692  O   ASP C  11      18.794  33.885 -20.109  1.00 26.66           O  \n",
      "ATOM    693  CB  ASP C  11      16.275  31.913 -21.090  1.00 23.92           C  \n",
      "ATOM    694  CG  ASP C  11      15.954  30.811 -22.086  1.00 32.86           C  \n",
      "ATOM    695  OD1 ASP C  11      16.857  30.015 -22.440  1.00 32.04           O  \n",
      "ATOM    696  OD2 ASP C  11      14.783  30.742 -22.514  1.00 34.62           O  \n",
      "ATOM    697  N   LYS C  12      17.436  33.086 -18.494  1.00 20.93           N  \n",
      "ATOM    698  CA  LYS C  12      17.666  34.197 -17.588  1.00 19.49           C  \n",
      "ATOM    699  C   LYS C  12      19.156  34.352 -17.314  1.00 21.43           C  \n",
      "ATOM    700  O   LYS C  12      19.708  35.446 -17.420  1.00 19.52           O  \n",
      "ATOM    701  CB  LYS C  12      16.934  33.956 -16.273  1.00 21.35           C  \n",
      "ATOM    702  CG  LYS C  12      16.820  35.186 -15.421  1.00 32.25           C  \n",
      "ATOM    703  CD  LYS C  12      15.418  35.300 -14.854  1.00 38.36           C  \n",
      "ATOM    704  CE  LYS C  12      15.443  35.227 -13.336  1.00 46.28           C  \n",
      "ATOM    705  NZ  LYS C  12      14.448  36.128 -12.685  1.00 47.14           N  \n",
      "ATOM    706  N   LEU C  13      19.799  33.248 -16.959  1.00 18.27           N  \n",
      "ATOM    707  CA  LEU C  13      21.220  33.239 -16.667  1.00 19.93           C  \n",
      "ATOM    708  C   LEU C  13      22.063  33.719 -17.847  1.00 24.26           C  \n",
      "ATOM    709  O   LEU C  13      23.065  34.416 -17.667  1.00 24.77           O  \n",
      "ATOM    710  CB  LEU C  13      21.658  31.833 -16.263  1.00 18.65           C  \n",
      "ATOM    711  CG  LEU C  13      22.553  31.711 -15.028  1.00 29.99           C  \n",
      "ATOM    712  CD1 LEU C  13      23.438  30.485 -15.183  1.00 30.00           C  \n",
      "ATOM    713  CD2 LEU C  13      23.412  32.970 -14.855  1.00 24.82           C  \n",
      "ATOM    714  N   GLU C  14      21.665  33.329 -19.057  1.00 20.75           N  \n",
      "ATOM    715  CA  GLU C  14      22.389  33.725 -20.258  1.00 23.59           C  \n",
      "ATOM    716  C   GLU C  14      22.331  35.244 -20.431  1.00 20.47           C  \n",
      "ATOM    717  O   GLU C  14      23.339  35.898 -20.715  1.00 19.82           O  \n",
      "ATOM    718  CB  GLU C  14      21.790  33.039 -21.492  1.00 18.72           C  \n",
      "ATOM    719  CG  GLU C  14      22.418  31.698 -21.817  1.00 38.24           C  \n",
      "ATOM    720  CD  GLU C  14      23.933  31.781 -21.991  1.00 49.91           C  \n",
      "ATOM    721  OE1 GLU C  14      24.663  31.678 -20.976  1.00 53.64           O  \n",
      "ATOM    722  OE2 GLU C  14      24.392  31.945 -23.146  1.00 49.34           O  \n",
      "ATOM    723  N   GLU C  15      21.135  35.795 -20.259  1.00 13.96           N  \n",
      "ATOM    724  CA  GLU C  15      20.937  37.225 -20.386  1.00 21.11           C  \n",
      "ATOM    725  C   GLU C  15      21.701  37.967 -19.287  1.00 19.02           C  \n",
      "ATOM    726  O   GLU C  15      22.227  39.057 -19.514  1.00 22.75           O  \n",
      "ATOM    727  CB  GLU C  15      19.449  37.563 -20.311  1.00 14.05           C  \n",
      "ATOM    728  CG  GLU C  15      19.144  39.060 -20.312  1.00 31.02           C  \n",
      "ATOM    729  CD  GLU C  15      19.959  39.845 -21.343  1.00 41.09           C  \n",
      "ATOM    730  OE1 GLU C  15      19.965  39.461 -22.537  1.00 36.91           O  \n",
      "ATOM    731  OE2 GLU C  15      20.595  40.852 -20.952  1.00 35.89           O  \n",
      "ATOM    732  N   THR C  16      21.764  37.362 -18.104  1.00 19.03           N  \n",
      "ATOM    733  CA  THR C  16      22.463  37.945 -16.971  1.00 15.23           C  \n",
      "ATOM    734  C   THR C  16      23.948  38.035 -17.298  1.00 20.18           C  \n",
      "ATOM    735  O   THR C  16      24.605  39.035 -17.005  1.00 18.97           O  \n",
      "ATOM    736  CB  THR C  16      22.282  37.083 -15.703  1.00 18.00           C  \n",
      "ATOM    737  OG1 THR C  16      20.895  37.081 -15.327  1.00 22.37           O  \n",
      "ATOM    738  CG2 THR C  16      23.118  37.633 -14.546  1.00 14.84           C  \n",
      "ATOM    739  N   LEU C  17      24.467  36.987 -17.923  1.00 16.33           N  \n",
      "ATOM    740  CA  LEU C  17      25.864  36.936 -18.312  1.00 18.96           C  \n",
      "ATOM    741  C   LEU C  17      26.177  37.942 -19.422  1.00 22.26           C  \n",
      "ATOM    742  O   LEU C  17      27.309  38.417 -19.546  1.00 22.54           O  \n",
      "ATOM    743  CB  LEU C  17      26.217  35.525 -18.767  1.00 15.96           C  \n",
      "ATOM    744  CG  LEU C  17      26.477  34.554 -17.616  1.00 28.60           C  \n",
      "ATOM    745  CD1 LEU C  17      26.582  33.143 -18.165  1.00 27.26           C  \n",
      "ATOM    746  CD2 LEU C  17      27.754  34.962 -16.885  1.00 28.48           C  \n",
      "ATOM    747  N   SER C  18      25.165  38.255 -20.230  1.00 18.13           N  \n",
      "ATOM    748  CA  SER C  18      25.302  39.210 -21.315  1.00 16.84           C  \n",
      "ATOM    749  C   SER C  18      25.383  40.615 -20.697  1.00 18.98           C  \n",
      "ATOM    750  O   SER C  18      26.147  41.482 -21.145  1.00 14.04           O  \n",
      "ATOM    751  CB  SER C  18      24.101  39.092 -22.245  1.00 15.51           C  \n",
      "ATOM    752  OG  SER C  18      24.069  40.152 -23.170  1.00 13.18           O  \n",
      "ATOM    753  N   LYS C  19      24.595  40.813 -19.642  1.00 17.71           N  \n",
      "ATOM    754  CA  LYS C  19      24.563  42.065 -18.904  1.00 19.24           C  \n",
      "ATOM    755  C   LYS C  19      25.938  42.284 -18.271  1.00 20.52           C  \n",
      "ATOM    756  O   LYS C  19      26.520  43.367 -18.373  1.00 25.19           O  \n",
      "ATOM    757  CB  LYS C  19      23.500  41.998 -17.803  1.00 17.79           C  \n",
      "ATOM    758  CG  LYS C  19      22.135  42.494 -18.227  1.00 31.49           C  \n",
      "ATOM    759  CD  LYS C  19      21.129  42.373 -17.091  1.00 38.89           C  \n",
      "ATOM    760  CE  LYS C  19      19.752  42.866 -17.517  1.00 38.53           C  \n",
      "ATOM    761  NZ  LYS C  19      19.239  42.176 -18.742  1.00 41.97           N  \n",
      "ATOM    762  N   ILE C  20      26.450  41.235 -17.628  1.00 19.93           N  \n",
      "ATOM    763  CA  ILE C  20      27.744  41.285 -16.966  1.00 20.04           C  \n",
      "ATOM    764  C   ILE C  20      28.868  41.570 -17.952  1.00 22.96           C  \n",
      "ATOM    765  O   ILE C  20      29.801  42.311 -17.641  1.00 28.18           O  \n",
      "ATOM    766  CB  ILE C  20      28.046  39.960 -16.245  1.00 20.04           C  \n",
      "ATOM    767  CG1 ILE C  20      27.135  39.825 -15.026  1.00 18.27           C  \n",
      "ATOM    768  CG2 ILE C  20      29.514  39.897 -15.831  1.00 13.16           C  \n",
      "ATOM    769  CD1 ILE C  20      26.969  38.400 -14.546  1.00 29.48           C  \n",
      "ATOM    770  N   TYR C  21      28.781  40.975 -19.139  1.00 24.54           N  \n",
      "ATOM    771  CA  TYR C  21      29.794  41.170 -20.170  1.00 22.46           C  \n",
      "ATOM    772  C   TYR C  21      29.879  42.657 -20.487  1.00 21.38           C  \n",
      "ATOM    773  O   TYR C  21      30.963  43.236 -20.553  1.00 17.97           O  \n",
      "ATOM    774  CB  TYR C  21      29.422  40.387 -21.434  1.00 18.91           C  \n",
      "ATOM    775  CG  TYR C  21      29.883  41.012 -22.742  1.00 16.95           C  \n",
      "ATOM    776  CD1 TYR C  21      29.077  41.936 -23.425  1.00 10.25           C  \n",
      "ATOM    777  CD2 TYR C  21      31.114  40.658 -23.316  1.00 13.20           C  \n",
      "ATOM    778  CE1 TYR C  21      29.489  42.491 -24.655  1.00 13.16           C  \n",
      "ATOM    779  CE2 TYR C  21      31.535  41.201 -24.537  1.00 11.65           C  \n",
      "ATOM    780  CZ  TYR C  21      30.717  42.115 -25.207  1.00 17.07           C  \n",
      "ATOM    781  OH  TYR C  21      31.121  42.622 -26.428  1.00 20.02           O  \n",
      "ATOM    782  N   HIS C  22      28.712  43.256 -20.677  1.00 22.38           N  \n",
      "ATOM    783  CA  HIS C  22      28.628  44.664 -20.995  1.00 22.88           C  \n",
      "ATOM    784  C   HIS C  22      29.171  45.533 -19.873  1.00 22.90           C  \n",
      "ATOM    785  O   HIS C  22      29.838  46.534 -20.123  1.00 26.35           O  \n",
      "ATOM    786  CB  HIS C  22      27.181  45.053 -21.269  1.00 21.56           C  \n",
      "ATOM    787  CG  HIS C  22      27.011  46.508 -21.527  1.00 31.33           C  \n",
      "ATOM    788  ND1 HIS C  22      27.568  47.136 -22.618  1.00 29.15           N  \n",
      "ATOM    789  CD2 HIS C  22      26.397  47.482 -20.809  1.00 26.40           C  \n",
      "ATOM    790  CE1 HIS C  22      27.311  48.430 -22.564  1.00 33.84           C  \n",
      "ATOM    791  NE2 HIS C  22      26.597  48.665 -21.471  1.00 32.72           N  \n",
      "ATOM    792  N   LEU C  23      28.858  45.168 -18.637  1.00 21.59           N  \n",
      "ATOM    793  CA  LEU C  23      29.317  45.917 -17.483  1.00 18.09           C  \n",
      "ATOM    794  C   LEU C  23      30.839  45.822 -17.346  1.00 22.60           C  \n",
      "ATOM    795  O   LEU C  23      31.497  46.773 -16.913  1.00 24.81           O  \n",
      "ATOM    796  CB  LEU C  23      28.637  45.394 -16.221  1.00 20.36           C  \n",
      "ATOM    797  CG  LEU C  23      27.173  45.808 -16.059  1.00 20.29           C  \n",
      "ATOM    798  CD1 LEU C  23      26.643  45.299 -14.731  1.00 19.95           C  \n",
      "ATOM    799  CD2 LEU C  23      27.060  47.317 -16.151  1.00 24.58           C  \n",
      "ATOM    800  N   GLU C  24      31.393  44.669 -17.712  1.00 17.11           N  \n",
      "ATOM    801  CA  GLU C  24      32.833  44.466 -17.644  1.00 17.67           C  \n",
      "ATOM    802  C   GLU C  24      33.486  45.512 -18.539  1.00 19.44           C  \n",
      "ATOM    803  O   GLU C  24      34.466  46.158 -18.164  1.00 14.54           O  \n",
      "ATOM    804  CB  GLU C  24      33.194  43.073 -18.156  1.00 17.24           C  \n",
      "ATOM    805  CG  GLU C  24      33.705  42.118 -17.095  1.00 29.64           C  \n",
      "ATOM    806  CD  GLU C  24      33.343  40.664 -17.393  1.00 35.57           C  \n",
      "ATOM    807  OE1 GLU C  24      33.141  40.336 -18.583  1.00 35.18           O  \n",
      "ATOM    808  OE2 GLU C  24      33.248  39.856 -16.442  1.00 38.90           O  \n",
      "ATOM    809  N   ASN C  25      32.915  45.673 -19.724  1.00 16.48           N  \n",
      "ATOM    810  CA  ASN C  25      33.406  46.626 -20.706  1.00 21.60           C  \n",
      "ATOM    811  C   ASN C  25      33.265  48.071 -20.208  1.00 22.36           C  \n",
      "ATOM    812  O   ASN C  25      34.184  48.875 -20.347  1.00 25.36           O  \n",
      "ATOM    813  CB  ASN C  25      32.645  46.454 -22.028  1.00 18.42           C  \n",
      "ATOM    814  CG  ASN C  25      32.864  45.077 -22.668  1.00 25.64           C  \n",
      "ATOM    815  OD1 ASN C  25      33.883  44.420 -22.444  1.00 20.82           O  \n",
      "ATOM    816  ND2 ASN C  25      31.902  44.646 -23.480  1.00 23.27           N  \n",
      "ATOM    817  N   GLU C  26      32.115  48.390 -19.615  1.00 21.67           N  \n",
      "ATOM    818  CA  GLU C  26      31.864  49.733 -19.105  1.00 24.31           C  \n",
      "ATOM    819  C   GLU C  26      32.859  50.155 -18.024  1.00 27.74           C  \n",
      "ATOM    820  O   GLU C  26      33.426  51.253 -18.083  1.00 25.65           O  \n",
      "ATOM    821  CB  GLU C  26      30.436  49.834 -18.552  1.00 23.82           C  \n",
      "ATOM    822  CG  GLU C  26      29.367  49.896 -19.624  1.00 32.95           C  \n",
      "ATOM    823  CD  GLU C  26      29.322  51.232 -20.339  1.00 43.11           C  \n",
      "ATOM    824  OE1 GLU C  26      28.850  52.222 -19.739  1.00 52.90           O  \n",
      "ATOM    825  OE2 GLU C  26      29.757  51.292 -21.507  1.00 50.93           O  \n",
      "ATOM    826  N   ILE C  27      33.066  49.277 -17.042  1.00 20.87           N  \n",
      "ATOM    827  CA  ILE C  27      33.978  49.538 -15.936  1.00 20.69           C  \n",
      "ATOM    828  C   ILE C  27      35.420  49.599 -16.417  1.00 24.92           C  \n",
      "ATOM    829  O   ILE C  27      36.226  50.357 -15.874  1.00 27.07           O  \n",
      "ATOM    830  CB  ILE C  27      33.865  48.446 -14.842  1.00 17.42           C  \n",
      "ATOM    831  CG1 ILE C  27      32.458  48.453 -14.239  1.00 22.49           C  \n",
      "ATOM    832  CG2 ILE C  27      34.892  48.689 -13.750  1.00 13.97           C  \n",
      "ATOM    833  CD1 ILE C  27      32.108  47.192 -13.476  1.00 25.00           C  \n",
      "ATOM    834  N   ALA C  28      35.747  48.797 -17.428  1.00 25.21           N  \n",
      "ATOM    835  CA  ALA C  28      37.101  48.780 -17.973  1.00 27.48           C  \n",
      "ATOM    836  C   ALA C  28      37.373  50.101 -18.681  1.00 27.14           C  \n",
      "ATOM    837  O   ALA C  28      38.487  50.621 -18.655  1.00 24.61           O  \n",
      "ATOM    838  CB  ALA C  28      37.267  47.621 -18.945  1.00 21.13           C  \n",
      "ATOM    839  N   ARG C  29      36.338  50.637 -19.315  1.00 23.33           N  \n",
      "ATOM    840  CA  ARG C  29      36.442  51.900 -20.023  1.00 25.50           C  \n",
      "ATOM    841  C   ARG C  29      36.593  53.028 -19.012  1.00 27.34           C  \n",
      "ATOM    842  O   ARG C  29      37.418  53.927 -19.186  1.00 29.51           O  \n",
      "ATOM    843  CB  ARG C  29      35.194  52.133 -20.874  1.00 31.98           C  \n",
      "ATOM    844  CG  ARG C  29      35.433  53.083 -22.033  1.00 42.60           C  \n",
      "ATOM    845  CD  ARG C  29      34.154  53.410 -22.795  1.00 54.13           C  \n",
      "ATOM    846  NE  ARG C  29      33.121  54.003 -21.946  1.00 66.12           N  \n",
      "ATOM    847  CZ  ARG C  29      33.175  55.227 -21.426  1.00 68.61           C  \n",
      "ATOM    848  NH1 ARG C  29      34.220  56.013 -21.660  1.00 68.23           N  \n",
      "ATOM    849  NH2 ARG C  29      32.180  55.667 -20.667  1.00 68.07           N  \n",
      "ATOM    850  N   VAL C  30      35.786  52.975 -17.960  1.00 25.07           N  \n",
      "ATOM    851  CA  VAL C  30      35.824  53.971 -16.905  1.00 27.71           C  \n",
      "ATOM    852  C   VAL C  30      37.197  53.928 -16.235  1.00 30.68           C  \n",
      "ATOM    853  O   VAL C  30      37.778  54.960 -15.895  1.00 28.97           O  \n",
      "ATOM    854  CB  VAL C  30      34.715  53.685 -15.862  1.00 23.35           C  \n",
      "ATOM    855  CG1 VAL C  30      35.229  53.924 -14.447  1.00 26.11           C  \n",
      "ATOM    856  CG2 VAL C  30      33.505  54.543 -16.150  1.00 32.88           C  \n",
      "ATOM    857  N   LYS C  31      37.716  52.719 -16.055  1.00 30.05           N  \n",
      "ATOM    858  CA  LYS C  31      39.008  52.535 -15.428  1.00 31.59           C  \n",
      "ATOM    859  C   LYS C  31      40.127  53.120 -16.283  1.00 33.88           C  \n",
      "ATOM    860  O   LYS C  31      41.030  53.779 -15.768  1.00 37.61           O  \n",
      "ATOM    861  CB  LYS C  31      39.255  51.048 -15.173  1.00 31.42           C  \n",
      "ATOM    862  CG  LYS C  31      39.029  50.634 -13.730  1.00 35.61           C  \n",
      "ATOM    863  CD  LYS C  31      40.220  49.867 -13.182  1.00 42.37           C  \n",
      "ATOM    864  CE  LYS C  31      40.472  48.604 -13.987  1.00 42.40           C  \n",
      "ATOM    865  NZ  LYS C  31      41.346  47.646 -13.258  1.00 42.64           N  \n",
      "ATOM    866  N   LYS C  32      40.070  52.884 -17.591  1.00 38.95           N  \n",
      "ATOM    867  CA  LYS C  32      41.084  53.401 -18.507  1.00 43.85           C  \n",
      "ATOM    868  C   LYS C  32      41.035  54.931 -18.512  1.00 45.37           C  \n",
      "ATOM    869  O   LYS C  32      42.065  55.610 -18.540  1.00 47.72           O  \n",
      "ATOM    870  CB  LYS C  32      40.842  52.864 -19.923  1.00 42.81           C  \n",
      "ATOM    871  CG  LYS C  32      42.097  52.380 -20.639  1.00 49.45           C  \n",
      "ATOM    872  CD  LYS C  32      41.802  51.974 -22.079  1.00 58.46           C  \n",
      "ATOM    873  CE  LYS C  32      42.763  52.648 -23.049  1.00 61.81           C  \n",
      "ATOM    874  NZ  LYS C  32      42.968  51.853 -24.293  1.00 57.54           N  \n",
      "ATOM    875  N   LEU C  33      39.817  55.462 -18.467  1.00 41.73           N  \n",
      "ATOM    876  CA  LEU C  33      39.587  56.896 -18.456  1.00 39.32           C  \n",
      "ATOM    877  C   LEU C  33      40.256  57.530 -17.241  1.00 40.78           C  \n",
      "ATOM    878  O   LEU C  33      40.964  58.531 -17.356  1.00 41.44           O  \n",
      "ATOM    879  CB  LEU C  33      38.084  57.178 -18.418  1.00 42.21           C  \n",
      "ATOM    880  CG  LEU C  33      37.532  58.326 -19.270  1.00 46.41           C  \n",
      "ATOM    881  CD1 LEU C  33      38.652  59.288 -19.662  1.00 52.92           C  \n",
      "ATOM    882  CD2 LEU C  33      36.857  57.748 -20.512  1.00 53.70           C  \n",
      "ATOM    883  N   LEU C  34      40.026  56.942 -16.072  1.00 41.13           N  \n",
      "ATOM    884  CA  LEU C  34      40.598  57.441 -14.830  1.00 45.20           C  \n",
      "ATOM    885  C   LEU C  34      42.123  57.331 -14.841  1.00 44.01           C  \n",
      "ATOM    886  O   LEU C  34      42.810  58.082 -14.148  1.00 46.97           O  \n",
      "ATOM    887  CB  LEU C  34      40.020  56.668 -13.640  1.00 45.43           C  \n",
      "ATOM    888  CG  LEU C  34      38.715  57.240 -13.073  1.00 53.73           C  \n",
      "ATOM    889  CD1 LEU C  34      37.708  57.462 -14.199  1.00 53.96           C  \n",
      "ATOM    890  CD2 LEU C  34      38.151  56.295 -12.025  1.00 60.34           C  \n",
      "ATOM    891  N   GLY C  35      42.644  56.398 -15.635  1.00 43.33           N  \n",
      "ATOM    892  CA  GLY C  35      44.079  56.223 -15.735  1.00 47.18           C  \n",
      "ATOM    893  C   GLY C  35      44.725  55.319 -14.703  1.00 50.79           C  \n",
      "ATOM    894  O   GLY C  35      45.894  54.942 -14.928  1.00 57.51           O  \n",
      "TER     895      GLY C  35                                                      \n",
      "HETATM  896  O   HOH A  38      26.147  45.148  -6.974  1.00 14.20           O  \n",
      "HETATM  897  O   HOH A  39      22.111  38.472  -1.719  1.00 33.28           O  \n",
      "HETATM  898  O   HOH A  40      18.578  43.388 -13.150  1.00 42.36           O  \n",
      "HETATM  899  O   HOH A  41      20.223  24.810  -6.395  1.00 31.43           O  \n",
      "HETATM  900  O   HOH A  42      21.983  27.259  -6.690  1.00 29.92           O  \n",
      "HETATM  901  O   HOH A  43      23.671  51.041  -6.619  1.00 40.80           O  \n",
      "HETATM  902  O   HOH A  44      21.198  51.153  -6.800  1.00 35.01           O  \n",
      "HETATM  903  O   HOH A  45      24.210  51.094  -4.029  1.00 47.69           O  \n",
      "HETATM  904  O   HOH A  46      26.568  31.827  -2.272  1.00 57.35           O  \n",
      "HETATM  905  O   HOH A  47      16.065  36.129  -3.676  1.00 51.30           O  \n",
      "HETATM  906  O   HOH A  48      23.815  34.353  -3.373  1.00 45.91           O  \n",
      "HETATM  907  O   HOH A  49      22.722  56.118 -10.350  1.00 40.51           O  \n",
      "HETATM  908  O   HOH A  50      20.563  54.249 -13.406  1.00 60.74           O  \n",
      "HETATM  909  O   HOH A  51      15.284  41.430 -11.913  1.00 56.98           O  \n",
      "HETATM  910  O   HOH A  52      10.934  31.087  -7.489  1.00 34.93           O  \n",
      "HETATM  911  O   HOH A  53      23.252  58.639 -12.671  1.00 49.05           O  \n",
      "HETATM  912  O   HOH A  54      19.560  48.860 -13.690  1.00 42.60           O  \n",
      "HETATM  913  O   HOH A  55      25.010  58.108 -16.363  1.00 59.63           O  \n",
      "HETATM  914  O   HOH A  56       9.501  30.190  -5.488  1.00 40.68           O  \n",
      "HETATM  915  O   HOH A  57      23.325  56.105  -7.472  1.00 48.15           O  \n",
      "HETATM  916  O   HOH A  58      12.120  29.382 -13.593  1.00 37.55           O  \n",
      "HETATM  917  O   HOH A  59      15.191  41.247 -14.433  1.00 48.30           O  \n",
      "HETATM  918  O   HOH A  60      20.688  60.012 -12.957  1.00 51.22           O  \n",
      "HETATM  919  O   HOH A  61      16.762  45.063  -7.083  1.00 49.82           O  \n",
      "HETATM  920  O   HOH A  62      26.790  61.199 -20.008  1.00 58.34           O  \n",
      "HETATM  921  O   HOH A  63      19.285  32.240  -1.256  1.00 49.19           O  \n",
      "HETATM  922  O   HOH A  64       8.911  29.563 -10.965  1.00 69.32           O  \n",
      "HETATM  923  O   HOH A  65      19.038  46.842 -17.128  1.00 59.83           O  \n",
      "HETATM  924  O   HOH A  66      15.068  33.512 -10.821  1.00 37.96           O  \n",
      "HETATM  925  O   HOH B  37      26.272  43.569  -3.438  1.00 12.05           O  \n",
      "HETATM  926  O   HOH B  38      32.067  42.144  -3.678  1.00  9.67           O  \n",
      "HETATM  927  O   HOH B  39      21.041  14.397 -15.666  1.00 35.68           O  \n",
      "HETATM  928  O   HOH B  40      27.216  54.942  -3.196  1.00 37.69           O  \n",
      "HETATM  929  O   HOH B  41      34.153  42.098  -2.247  1.00 49.29           O  \n",
      "HETATM  930  O   HOH B  42      33.763  44.781  -3.186  1.00 37.63           O  \n",
      "HETATM  931  O   HOH B  43      30.985  32.307  -6.361  1.00 36.82           O  \n",
      "HETATM  932  O   HOH B  44      35.337  35.952 -12.074  1.00 31.71           O  \n",
      "HETATM  933  O   HOH B  45      35.975  37.010  -8.059  1.00 32.10           O  \n",
      "HETATM  934  O   HOH B  46      32.826  54.837   0.403  1.00 49.90           O  \n",
      "HETATM  935  O   HOH B  47      35.960  37.859 -13.626  1.00 38.81           O  \n",
      "HETATM  936  O   HOH B  48      36.832  39.082  -6.942  1.00 41.26           O  \n",
      "HETATM  937  O   HOH B  49      38.808  40.987  -7.960  1.00 31.40           O  \n",
      "HETATM  938  O   HOH B  50      31.750  26.362 -10.615  1.00 44.87           O  \n",
      "HETATM  939  O   HOH B  51      29.346  19.950 -10.981  1.00 59.52           O  \n",
      "HETATM  940  O   HOH B  52      30.303  56.948  -2.899  1.00 39.68           O  \n",
      "HETATM  941  O   HOH B  53      28.951  59.166  -2.247  1.00 41.74           O  \n",
      "HETATM  942  O   HOH B  54      37.893  37.100 -10.232  1.00 39.90           O  \n",
      "HETATM  943  O   HOH B  55      25.103  26.316  -6.001  1.00 45.41           O  \n",
      "HETATM  944  O   HOH B  56      24.856  28.177  -7.638  1.00 47.09           O  \n",
      "HETATM  945  O   HOH B  57      19.119  21.798 -23.406  1.00 49.26           O  \n",
      "HETATM  946  O   HOH B  58      36.543  42.236 -15.543  1.00 43.68           O  \n",
      "HETATM  947  O   HOH B  59      34.018  33.799 -13.415  1.00 52.60           O  \n",
      "HETATM  948  O   HOH B  60      17.787  24.092 -24.644  1.00 48.17           O  \n",
      "HETATM  949  O   HOH B  61      21.857  23.517 -18.069  1.00 42.45           O  \n",
      "HETATM  950  O   HOH B  62      35.594  52.280  -2.508  1.00 39.62           O  \n",
      "HETATM  951  O   HOH B  63      31.101  27.041  -7.856  1.00 49.39           O  \n",
      "HETATM  952  O   HOH B  64      20.160  28.812 -13.594  1.00 51.09           O  \n",
      "HETATM  953  O   HOH C  38      29.487  46.140 -24.227  1.00 22.01           O  \n",
      "HETATM  954  O   HOH C  39      25.933  45.279 -24.577  1.00 29.79           O  \n",
      "HETATM  955  O   HOH C  40      29.442  44.794 -26.865  1.00 22.73           O  \n",
      "HETATM  956  O   HOH C  41      29.675  37.000 -19.017  1.00 36.28           O  \n",
      "HETATM  957  O   HOH C  42      25.606  35.152 -22.591  1.00 27.46           O  \n",
      "HETATM  958  O   HOH C  43      17.861  35.018 -22.940  1.00 27.12           O  \n",
      "HETATM  959  O   HOH C  44      14.022  31.502 -13.577  1.00 40.20           O  \n",
      "HETATM  960  O   HOH C  45      31.787  50.225 -22.342  1.00 55.17           O  \n",
      "HETATM  961  O   HOH C  46      34.982  41.475 -23.384  1.00 40.89           O  \n",
      "HETATM  962  O   HOH C  47      13.942  33.298 -19.059  1.00 44.84           O  \n",
      "HETATM  963  O   HOH C  48      22.479  29.335 -18.936  1.00 37.05           O  \n",
      "HETATM  964  O   HOH C  49      33.980  39.645 -13.896  1.00 42.81           O  \n",
      "HETATM  965  O   HOH C  50      12.838  31.754 -21.286  1.00 55.55           O  \n",
      "HETATM  966  O   HOH C  51      28.800  53.305 -23.641  1.00 55.52           O  \n",
      "HETATM  967  O   HOH C  52       7.146  23.355 -10.675  1.00 41.61           O  \n",
      "HETATM  968  O   HOH C  53      23.481  31.641 -25.532  1.00 56.17           O  \n",
      "HETATM  969  O   HOH C  54      17.526  26.703 -22.441  1.00 39.34           O  \n",
      "HETATM  970  O   HOH C  55       7.754  28.897 -15.231  1.00 62.98           O  \n",
      "HETATM  971  O   HOH C  56      40.530  48.973 -18.068  1.00 36.90           O  \n",
      "HETATM  972  O   HOH C  57      32.082  38.012 -19.424  1.00 38.87           O  \n",
      "HETATM  973  O   HOH C  58      34.918  37.727 -16.403  1.00 60.59           O  \n",
      "MASTER      287    0    0    3    0    0    0    6  970    3    0    9          \n",
      "END                                                                             \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "less 1ce0.pdb\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "import Bio.PDB.PDBParser\n",
    "import sys\n",
    "\n",
    "# Use QUIET=True to avoid lots of warnings...\n",
    "parser = Bio.PDB.PDBParser(QUIET=True)\n",
    "structure = parser.get_structure(\"1HPV\",\"1ce0.pdb\")\n",
    "\n",
    "achain = structure[0]['A']\n",
    "bchain = structure[0]['B']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.PDB.Polypeptide import is_aa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Residues LYS 12 in chain A and LYS 7 in chain B are close to each other: 11.29\n"
     ]
    }
   ],
   "source": [
    "# Busco en el carbono de la \"colita\"\n",
    "for res1 in achain:\n",
    "    # Se revisa que sea un aa y además que sea especificamente Lisina.\n",
    "    if is_aa(res1) and (res1.get_resname()==\"LYS\"):\n",
    "        r1ca = res1['NZ']\n",
    "        r1ind = res1.get_id()[1]\n",
    "        r1sym = res1.get_resname()\n",
    "        # Busco el carbono de la \"colita\" de la otra cadena\n",
    "        for res2 in bchain:\n",
    "            # Se revisa que sea un aa y además que sea especificamente Lisina.\n",
    "            if is_aa(res2) and (res2.get_resname()==\"LYS\") :\n",
    "                r2ca = res2['NZ']\n",
    "                r2ind = res2.get_id()[1]\n",
    "                r2sym = res2.get_resname()\n",
    "                \n",
    "                #las distancias entre lisinas deben ser menores a 12 A\n",
    "                if (r1ca - r2ca) < 12:\n",
    "                    print(\"Residues\",r1sym,r1ind,\"in chain A\", end=\" \")\n",
    "                    print(\"and\",r2sym,r2ind,\"in chain B\", end=\" \")\n",
    "                    print(\"are close to each other:\",round(r1ca-r2ca,2))\n",
    "                    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_3\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 3</h2>\n",
    "<p>Utilizando Entrez.einfo y Entrez.read, imprime la descripción de:\n",
    "    \n",
    "    \n",
    "+ field **'ECNO'**\n",
    "+ link **'protein_protein_small_genome'**\n",
    "    \n",
    "    \n",
    "de la base de datos 'protein'</p>\n",
    "\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = \"otokani@lcg.unam.mx\"\n",
    "#Se especifica la base protein\n",
    "handle = Entrez.einfo(db=\"protein\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "record = Entrez.read(handle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EC number for enzyme or CAS registry number\n"
     ]
    }
   ],
   "source": [
    "#Se checa cada field de la lista fields y cuando se encuentra \"ECNO\", se imprime su descripcion\n",
    "for field in record[\"DbInfo\"][\"FieldList\"]:    \n",
    "    if \"%(Name)s\" % field == \"ECNO\":\n",
    "        print(\"%(Description)s\"%field)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All proteins from this genome\n"
     ]
    }
   ],
   "source": [
    "#Se checa cada link de la lista LinkList y cuando se encuentra a \"protein_protein_small_genome\" , se imprime su descripcion\n",
    "for link in record[\"DbInfo\"][\"LinkList\"]:\n",
    "    if \"%(Name)s\" % link == \"protein_protein_small_genome\":\n",
    "        print(\"%(Description)s\"%link)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_4\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 4</h2>\n",
    "<p>Automaticen para buscar los siguientes genes en la base de datos nucleotide y guarden las secuencias de las proteínas en un archivo fasta: \n",
    "    \n",
    "    \n",
    "    Organismo: Notoryctes typhlops             Genes: PNOC y RUNX\n",
    "    Organismo: Cypripedium guttatum            Genes: matK\n",
    "    Organismo: Hepatitis B virus               Genes: X                     Aislado: migrant 467\n",
    "    \n",
    "    \n",
    "</p>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"otokani@lcg.unam.mx\"\n",
    "#Para obtener todos los genes de todos los organismos se usan simplememte operaciones logicas con \"AND\" y \"OR\"\n",
    "search_handle = Entrez.esearch(db=\"nucleotide\",term=\"(Notoryctes typhlops AND (RUNX2 OR PNOC)) OR (Cypripedium guttatum AND matK) OR (Hepatitis B virus AND (X Aislado: migrant 467))  \", usehistory=\"y\")\n",
    "search_results = Entrez.read(search_handle)\n",
    "search_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "count =int(search_results[\"Count\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "webenv = search_results[\"WebEnv\"]\n",
    "query_key = search_results[\"QueryKey\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Going to download record 1 to 5\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "import time\n",
    "from urllib.error import HTTPError \n",
    "\n",
    "# Indicamos el numero de genes que vamos a recuperar\n",
    "batch_size = 5\n",
    "\n",
    "#archivo de salida\n",
    "out_handle = open(\"genes.fasta\", \"w\")\n",
    "\n",
    "for start in range(0, count, batch_size):\n",
    "    end = min(count, start+batch_size)\n",
    "    print(\"Going to download record %i to %i\" % (start+1, end))\n",
    "    attempt = 1\n",
    "    while attempt <= 3:\n",
    "        try:\n",
    "            fetch_handle = Entrez.efetch(db=\"nucleotide\", rettype=\"fasta\", retmode=\"text\",\n",
    "                                         retstart=start, retmax=batch_size,\n",
    "                                         webenv=webenv, query_key=query_key)\n",
    "            \n",
    "            break\n",
    "        except HTTPError as err:\n",
    "            if 500 <= err.code <= 599:\n",
    "                print(\"Received error from server %s\" % err)\n",
    "                print(\"Attempt %i of 3\" % attempt)\n",
    "                attempt += 1\n",
    "                time.sleep(15)\n",
    "            else:\n",
    "                raise\n",
    "                \n",
    "    data = fetch_handle.read()\n",
    "    fetch_handle.close()\n",
    "    out_handle.write(data)\n",
    "out_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KY938699.1 Notoryctes typhlops runt-related transcription factor 2 (RUNX2) gene, partial cds\n",
      "ACCCGGCAGATGAGCGACGTGAGCCCGGTAGTGGCTGCCCAGCAGCAGCAGCAGCAGCAGCAGCAGCAAC\n",
      "AGCAACAGCAGCAACAGCAGCAGCAGCAGCAGCAGCAGCAGGAGGCGGCGGCCGCAGCAGCGGCGGCGGC\n",
      "GGCGGCGGCGGCGGCGGCGGCAGCGGCAGCGGCAGCCGCGGTGCCGAGGCTGCGCCCGCCTCACGACAAC\n",
      "CGCACGATG\n",
      "\n",
      ">GQ855547.1 Hepatitis B virus isolate migrant 467 X protein (X), precore/core protein (C), and core protein (C) genes, partial cds\n",
      "TGTGCCTTCTCATCTGCCGGTCCGTGTGCACTTCGCTTCACCTCTGCACGTCGCATGGAGACCACCGTGA\n",
      "ACGCCCGCCAGGTCTTGCCCAAGGTCTTACATAAGAGGACTCTTGGACTCTCAGCAATGTCAACGACCGA\n",
      "CCTTGAGGCATACTTCAAAGACTGTGTATTTAAAGACTGGGAGGAGTTGGGGGAGGAGATTAGGTTAAAG\n",
      "GTCTTTGTACTGGGAGGCTGTAGGCATAAATTGGTCTGTTCACCAGCACCATGCAACTTTTTCACCTCTG\n",
      "CCTAATCATCTCATGTTCATGTCCTACTGTTCAAGCCTCCAAGCTGTGCCTTGGGTGGCTTTGGGGCATG\n",
      "GACATTGACCCGTATAAAGAATTTGGAGCTTCTGTGGAGTTACTCTCTTTTTTGCCTTCTGACTTCTTTC\n",
      "CTTCTATTCGAGATCTCCTCGACACCGCCTCTGCTCTATATCGGGAGGCCTTAGAGTCTCCGGAACATTG\n",
      "TTCACCTCAC\n",
      "\n",
      ">KP769247.1 Cypripedium guttatum isolate BOP011182 maturase K (matK) gene, partial cds; chloroplast\n",
      "AAATCCTTCAATGCTGGATCAAGGATGTTCCTTCTTTGCATTTATTGCGATTGCTTTTCCACGAATATCA\n",
      "TTATTTTAATAGTCTCATTACTTCAAAAAAATTAATTTACGCCTTTTCAAGAATAAATAAAAAATTCCTT\n",
      "TGGTTCCTATATAATTCTTATGTATATGAATGCGAATATCTATTCCATTTTCTTCGTAAACAGTCTTCTT\n",
      "ATTTACGATCAACATCTTCTGGAGTGTTTCTTGAGCGAACACATTTCTATGTAAAAATAGAACATCTTAT\n",
      "AGTAGTGTGTTGTAATTCTTTTCATAGGATCCTATGCTTTCTCAAGGATCCTTTCATGCATTATGTTAGA\n",
      "TATCAAGGAAA\n",
      "\n",
      ">KF261992.1 Cypripedium guttatum isolate CG_20120624_74 maturase K (matK) gene, partial cds; chloroplast\n",
      "GTTCCTTCTTTGCATTTATTGCGATTGCTTTTCCACGAATATCATTATTTTAATAGTCTCATTACTTCAA\n",
      "AGAAAGGCATTTACGCCTTTTCAAGAATAAATAAAAGATTCCTTTGGTTCCTATATAATTCTTATGTATA\n",
      "TGAATGCGAATATCTATTCCATTTTCTTCGTAAACAGTCTTCTTATTTACGATCAACATCTTCTGGAGTG\n",
      "TTTCTTGAGCGAACACATTTCTATGTAAAAATAGAACATCTTATAGTAGTGTGTTGTAATTCTTTTCATA\n",
      "GGATCCTATGCTTTCTCAAGGATCCTTTCATGCATTATGTTCGATATCAAGGAAAAGCAATTCTGGCTTC\n",
      "AAAGGGAACTCTTATTATGATGAAGAAATGGAAATTTCATCTTGTGAATTTTTGGCAATCTTATTTTCAC\n",
      "TTTTGGTCTCAACCGTATAGGATCCATATAAAGCAATTATCCAACTATTCCTTCTCTTTTCTGGGGTATT\n",
      "TTTCAAGTGTACTAGAAAATAATTTGGTAGTAAGAAATCAAATGCTAGAGAATTCATTTCTAATAAATAT\n",
      "TATGACTAAGAAATTAGATACCATAGCCCCAGTTATTTCTCTTATTGGATCATTGTCGAAAGCTCAATTT\n",
      "TGTACTGTATTGGGCCATCCTATTAGTAAACCGATCTGGACCGATTTATCGGATTCTGATATTCTTGATC\n",
      "GATTTTGCCGGATATGTAGAAATCTTTGTCGTTATC\n",
      "\n",
      ">JN633765.1 Notoryctes typhlops prepronociceptin (PNOC) gene, partial cds\n",
      "TTCTGGGCTCCTTGCATCAAGGCCATGAATAAGGGGCCCTGGCAGCTCAGCTCAGCCAGCAAGGACAAGG\n",
      "CAATGACATCCTTCTATCAGCCACAGTTGATGGAGGCACAATCCCTAAGCCAGATGGAGGGCAAGAACCG\n",
      "AGTGTCTCGTGTCAGAAGTTTGTTCCATGCCCAGGAACAAGAGGCAGCAGAAGGCTTAGGGGAGTCAGGG\n",
      "GAAGAGATGCAGAAGAAATTACAGAAACGGTTTGGGGGCTTCACTGGGGCACGGAAATCAGCCCGGAAGT\n",
      "TGGCCAACCAGAAG\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head -200 genes.fasta"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
