{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_1\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 1</h2>\n",
    "<p>Un ORF inicia con un codón inicial y termina, ya sea con un codón final o al final de la cádena.\n",
    "</p>\n",
    "<p>Utilizando como input una cadena de ADN, obtener la cadena proteica de mayor longitud que puede ser traducida de cualquiera de sus ORF.\n",
    "</p>\n",
    "<p>Si más de una cadena protéica de longitud maxima existe, entonces podrás emitir cualquier solución.</p>\n",
    "<h3 id=\"sample-dataset\">Conjunto de datos de muestra</h3>\n",
    "<div class=\"codehilite\"><pre>AGCCATGTAGCTAACTCAGGTTACATGGGGATGACCCCGCGACTTGGATTAGAGTCTCTTTTGGAATAAGCCTGAATGATCCGAGTAGCATCTCAG\n",
    "</pre></div>\n",
    "\n",
    "\n",
    "<h3 id=\"sample-output\">Resultado emitido de muestra</h3>\n",
    "<div class=\"codehilite\"><pre>MLLGSFRLIPKETLIQVAGSSPCNLS\n",
    "</pre></div>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Larger protein: MLLGSFRLIPKETLIQVAGSSPCNLS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.5/dist-packages/Bio/Seq.py:2859: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n",
      "  BiopythonWarning,\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "ADN_forward= Seq(\"AGCCATGTAGCTAACTCAGGTTACATGGGGATGACCCCGCGACTTGGATTAGAGTCTCTTTTGGAATAAGCCTGAATGATCCGAGTAGCATCTCAG\")\n",
    "ADN_reverse=ADN_forward.reverse_complement()\n",
    "#Se usan dos listas. Una para guardar las proteinas y otra para guardar sus respectivos tamaños.\n",
    "list_protein=[]\n",
    "list_pro_length=[]\n",
    "\n",
    "for ORF in range(3):\n",
    "#Se hace un corte en la secuecnia forward y reverse dependiendo del marco de lectura correspondiente.\n",
    "    strand_forward=ADN_forward[ORF:len(ADN_forward)]\n",
    "    strand_reverse=ADN_reverse[ORF:len(ADN_forward)]\n",
    "#Al hacer la traducción nos arroja la proteína con codones de paro y de inicio juntos en una misma secuencia.\n",
    "    strand_forward=strand_forward.translate()\n",
    "    strand_reverse=strand_reverse.translate()\n",
    "#Dividimos la proteína (en proteínas más chicas) cada que encuentre un codon de paro.\n",
    "    for protein in strand_forward.split(\"*\"):\n",
    "        #Volvemos a hacer cortes en las proteínas (en los codones de inicio) para tener una proteína \"real\".\n",
    "        if protein.find(\"M\")!=-1:\n",
    "            start=protein[protein.find(\"M\"):len(protein)]\n",
    "            list_protein.append(start)\n",
    "            list_pro_length.append(len(start))\n",
    "            \n",
    "            \n",
    "    for protein in strand_reverse.split(\"*\"):\n",
    "        if protein.find(\"M\")!=-1:\n",
    "            start=protein[protein.find(\"M\"):len(protein)]\n",
    "            list_protein.append(start)\n",
    "            list_pro_length.append(len(start))\n",
    "#Cada proteína \"real\" se guarda en una lista al igual que su respectivo tamaño en la otra lista.Se obtiene la más grande y se imprime\n",
    "index_max=(list_pro_length.index(max(list_pro_length)))\n",
    "large_protein=list_protein[index_max]\n",
    "\n",
    "print(\"Larger protein: {}\".format(large_protein))\n",
    "            \n",
    "            \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_2\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">\n",
    "<h2 id=\"problem\">Ejercicio 2</h2>\n",
    "    \n",
    "1. Teniendo como input un archivo fasta, crea un diccionario de todas las secuencias en tu atchivo FASTA.\n",
    "\n",
    "2. Encuentra todos los codones en tu primer marco de lectura. \n",
    "\n",
    "3. Imprime los codones de cada secuncia en formato FASTA. \n",
    "\n",
    "   ```\n",
    "   >my_seq_marco_1\n",
    "   ATG TTC ATC\n",
    "   ```\n",
    "\n",
    "\n",
    "4. Imprime los codones de cada secuencia en los 6 marcos lectura.\n",
    "</div>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter the path of your fasta file\n",
      "seq.nt.fa\n",
      "\n",
      ">my_seq_marco_0\n",
      "AAG\tAGC\tAGC\tTCG\tCGC\tTAA\tTGT\tGAT\tAGA\tTGG\tCGG\tTAA\tAGT\tAAA\tTGT\tCCT\tATG\tGGC\tCAC\tCAA\tTTA\tTGG\tTGT\tATG\tAGT\tGAA\tTCT\tCTG\tGTC\tCGA\tGAT\tTCA\tCTG\tAGT\tAAC\tTGC\tTGT\tACA\tCAG\tTAG\tTAA\tCAC\tGTG\tGAG\tATC\tCCA\tTAA\tGCT\tTCA\tCGT\tGTG\tGTC\tCAA\tTAA\tAAC\tACT\tCCG\tTTG\tGTC\tAAC\t\n",
      ">my_seq_marco_-0\n",
      "GTT\tGAC\tCAA\tCGG\tAGT\tGTT\tTTA\tTTG\tGAC\tCAC\tACG\tTGA\tAGC\tTTA\tTGG\tGAT\tCTC\tCAC\tGTG\tTTA\tCTA\tCTG\tTGT\tACA\tGCA\tGTT\tACT\tCAG\tTGA\tATC\tTCG\tGAC\tCAG\tAGA\tTTC\tACT\tCAT\tACA\tCCA\tTAA\tTTG\tGTG\tGCC\tCAT\tAGG\tACA\tTTT\tACT\tTTA\tCCG\tCCA\tTCT\tATC\tACA\tTTA\tGCG\tCGA\tGCT\tGCT\tCTT\t\n",
      ">my_seq_marco_1\n",
      "AGA\tGCA\tGCT\tCGC\tGCT\tAAT\tGTG\tATA\tGAT\tGGC\tGGT\tAAA\tGTA\tAAT\tGTC\tCTA\tTGG\tGCC\tACC\tAAT\tTAT\tGGT\tGTA\tTGA\tGTG\tAAT\tCTC\tTGG\tTCC\tGAG\tATT\tCAC\tTGA\tGTA\tACT\tGCT\tGTA\tCAC\tAGT\tAGT\tAAC\tACG\tTGG\tAGA\tTCC\tCAT\tAAG\tCTT\tCAC\tGTG\tTGG\tTCC\tAAT\tAAA\tACA\tCTC\tCGT\tTGG\tTCA\t\n",
      ">my_seq_marco_-1\n",
      "TTG\tACC\tAAC\tGGA\tGTG\tTTT\tTAT\tTGG\tACC\tACA\tCGT\tGAA\tGCT\tTAT\tGGG\tATC\tTCC\tACG\tTGT\tTAC\tTAC\tTGT\tGTA\tCAG\tCAG\tTTA\tCTC\tAGT\tGAA\tTCT\tCGG\tACC\tAGA\tGAT\tTCA\tCTC\tATA\tCAC\tCAT\tAAT\tTGG\tTGG\tCCC\tATA\tGGA\tCAT\tTTA\tCTT\tTAC\tCGC\tCAT\tCTA\tTCA\tCAT\tTAG\tCGC\tGAG\tCTG\tCTC\t\n",
      ">my_seq_marco_2\n",
      "GAG\tCAG\tCTC\tGCG\tCTA\tATG\tTGA\tTAG\tATG\tGCG\tGTA\tAAG\tTAA\tATG\tTCC\tTAT\tGGG\tCCA\tCCA\tATT\tATG\tGTG\tTAT\tGAG\tTGA\tATC\tTCT\tGGT\tCCG\tAGA\tTTC\tACT\tGAG\tTAA\tCTG\tCTG\tTAC\tACA\tGTA\tGTA\tACA\tCGT\tGGA\tGAT\tCCC\tATA\tAGC\tTTC\tACG\tTGT\tGGT\tCCA\tATA\tAAA\tCAC\tTCC\tGTT\tGGT\tCAA\t\n",
      ">my_seq_marco_-2\n",
      "TGA\tCCA\tACG\tGAG\tTGT\tTTT\tATT\tGGA\tCCA\tCAC\tGTG\tAAG\tCTT\tATG\tGGA\tTCT\tCCA\tCGT\tGTT\tACT\tACT\tGTG\tTAC\tAGC\tAGT\tTAC\tTCA\tGTG\tAAT\tCTC\tGGA\tCCA\tGAG\tATT\tCAC\tTCA\tTAC\tACC\tATA\tATT\tGGT\tGGC\tCCA\tTAG\tGAC\tATT\tTAC\tTTT\tACC\tGCC\tATC\tTAT\tCAC\tATT\tAGC\tGCG\tAGC\tTGC\tTCT\t\n",
      ">my_seq_marco_0\n",
      "GCC\tACA\tGAG\tCCT\tAGG\tACC\tCCA\tACC\tTAA\tCCT\tAAC\tCTA\tACC\tTAA\tCCT\tACA\tGTT\tTGA\tTCT\tTAA\tCCA\tTGA\tGGC\tTGA\tGAA\tGCG\tATG\tTCC\tTGA\tCCG\tGCC\tTGT\tCCT\tAAC\tCGC\tCCT\tGAC\tCTA\tACC\tGGC\tTTG\tACC\tTAA\tCCG\tCCC\tTGA\tCCT\tAAC\tCAG\tGCT\tAAC\tCTA\tACC\tAAA\tCCG\tTGA\tAAA\tAAG\tGAA\tTCT\t\n",
      ">my_seq_marco_-0\n",
      "AGA\tTTC\tCTT\tTTT\tTCA\tCGG\tTTT\tGGT\tTAG\tGTT\tAGC\tCTG\tGTT\tAGG\tTCA\tGGG\tCGG\tTTA\tGGT\tCAA\tGCC\tGGT\tTAG\tGTC\tAGG\tGCG\tGTT\tAGG\tACA\tGGC\tCGG\tTCA\tGGA\tCAT\tCGC\tTTC\tTCA\tGCC\tTCA\tTGG\tTTA\tAGA\tTCA\tAAC\tTGT\tAGG\tTTA\tGGT\tTAG\tGTT\tAGG\tTTA\tGGT\tTGG\tGGT\tCCT\tAGG\tCTC\tTGT\tGGC\t\n",
      ">my_seq_marco_1\n",
      "CCA\tCAG\tAGC\tCTA\tGGA\tCCC\tCAA\tCCT\tAAC\tCTA\tACC\tTAA\tCCT\tAAC\tCTA\tCAG\tTTT\tGAT\tCTT\tAAC\tCAT\tGAG\tGCT\tGAG\tAAG\tCGA\tTGT\tCCT\tGAC\tCGG\tCCT\tGTC\tCTA\tACC\tGCC\tCTG\tACC\tTAA\tCCG\tGCT\tTGA\tCCT\tAAC\tCGC\tCCT\tGAC\tCTA\tACC\tAGG\tCTA\tACC\tTAA\tCCA\tAAC\tCGT\tGAA\tAAA\tAGG\tAAT\t\n",
      ">my_seq_marco_-1\n",
      "GAT\tTCC\tTTT\tTTT\tCAC\tGGT\tTTG\tGTT\tAGG\tTTA\tGCC\tTGG\tTTA\tGGT\tCAG\tGGC\tGGT\tTAG\tGTC\tAAG\tCCG\tGTT\tAGG\tTCA\tGGG\tCGG\tTTA\tGGA\tCAG\tGCC\tGGT\tCAG\tGAC\tATC\tGCT\tTCT\tCAG\tCCT\tCAT\tGGT\tTAA\tGAT\tCAA\tACT\tGTA\tGGT\tTAG\tGTT\tAGG\tTTA\tGGT\tTAG\tGTT\tGGG\tGTC\tCTA\tGGC\tTCT\tGTG\t\n",
      ">my_seq_marco_2\n",
      "CAC\tAGA\tGCC\tTAG\tGAC\tCCC\tAAC\tCTA\tACC\tTAA\tCCT\tAAC\tCTA\tACC\tTAC\tAGT\tTTG\tATC\tTTA\tACC\tATG\tAGG\tCTG\tAGA\tAGC\tGAT\tGTC\tCTG\tACC\tGGC\tCTG\tTCC\tTAA\tCCG\tCCC\tTGA\tCCT\tAAC\tCGG\tCTT\tGAC\tCTA\tACC\tGCC\tCTG\tACC\tTAA\tCCA\tGGC\tTAA\tCCT\tAAC\tCAA\tACC\tGTG\tAAA\tAAA\tGGA\tATC\t\n",
      ">my_seq_marco_-2\n",
      "ATT\tCCT\tTTT\tTTC\tACG\tGTT\tTGG\tTTA\tGGT\tTAG\tCCT\tGGT\tTAG\tGTC\tAGG\tGCG\tGTT\tAGG\tTCA\tAGC\tCGG\tTTA\tGGT\tCAG\tGGC\tGGT\tTAG\tGAC\tAGG\tCCG\tGTC\tAGG\tACA\tTCG\tCTT\tCTC\tAGC\tCTC\tATG\tGTT\tAAG\tATC\tAAA\tCTG\tTAG\tGTT\tAGG\tTTA\tGGT\tTAG\tGTT\tAGG\tTTG\tGGG\tTCC\tTAG\tGCT\tCTG\tTGG\t\n",
      ">my_seq_marco_0\n",
      "ATG\tAAA\tGTT\tACA\tTAA\tAGA\tCTA\tTTC\tGAT\tGCA\tTAA\tATA\tGTT\tCAG\tTTT\tTGA\tAAA\tCTT\tACA\tTTT\tTGT\tTAA\tAGT\tCAG\tGTA\tCTT\tGTG\tTAT\tAAT\tATC\tAAC\tTAA\t\n",
      ">my_seq_marco_-0\n",
      "ATT\tTAG\tTTG\tATA\tTTA\tTAC\tACA\tAGT\tACC\tTGA\tCTT\tTAA\tCAA\tAAT\tGTA\tAGT\tTTT\tCAA\tAAC\tTGA\tACT\tATT\tTAT\tGCA\tTCG\tAAT\tAGT\tCTT\tTAT\tGTA\tACT\tTTC\t\n",
      ">my_seq_marco_1\n",
      "TGA\tAAG\tTTA\tCAT\tAAA\tGAC\tTAT\tTCG\tATG\tCAT\tAAA\tTAG\tTTC\tAGT\tTTT\tGAA\tAAC\tTTA\tCAT\tTTT\tGTT\tAAA\tGTC\tAGG\tTAC\tTTG\tTGT\tATA\tATA\tTCA\tACT\tAAA\t\n",
      ">my_seq_marco_-1\n",
      "TTT\tAGT\tTGA\tTAT\tTAT\tACA\tCAA\tGTA\tCCT\tGAC\tTTT\tAAC\tAAA\tATG\tTAA\tGTT\tTTC\tAAA\tACT\tGAA\tCTA\tTTT\tATG\tCAT\tCGA\tATA\tGTC\tTTT\tATG\tTAA\tCTT\tTCA\t\n",
      ">my_seq_marco_2\n",
      "GAA\tAGT\tTAC\tATA\tAAG\tACT\tATT\tCGA\tTGC\tATA\tAAT\tAGT\tTCA\tGTT\tTTG\tAAA\tACT\tTAC\tATT\tTTG\tTTA\tAAG\tTCA\tGGT\tACT\tTGT\tGTA\tTAA\tTAT\tCAA\tCTA\tAAT\t\n",
      ">my_seq_marco_-2\n",
      "TTA\tGTT\tGAT\tATT\tATA\tCAC\tAAG\tTAC\tCTG\tACT\tTTA\tACA\tAAA\tTGT\tAAG\tTTT\tTCA\tAAA\tCTG\tAAC\tTAT\tTTA\tTGC\tATC\tGAA\tTAG\tTCT\tTTA\tTGT\tAAC\tTTT\tCAT\t\n",
      ">my_seq_marco_0\n",
      "ATG\tCTA\tACC\tAAA\tGTT\tTCA\tGTT\tCGG\tACG\tTGT\tCGA\tTGA\tGCG\tACG\tCTC\tAAA\tAAG\tGAA\tACA\tACA\tTGC\tCAA\tATA\tGAA\tACG\tATC\tAAT\tTCG\tGCG\tATG\tGAA\tATC\tAGA\tACA\tACG\tATC\tAGT\tTTG\tGAA\tATC\tAAA\tATA\tGAA\tATA\tACG\tGGA\tACG\tATC\tAGT\tTTA\tATA\tACA\tTGA\tTGC\tAGA\tATA\tAAG\tGGA\tATA\tATC\tAAT\tTTA\tATC\tCAG\tGTA\tATC\tAGA\tACA\tGAG\t\n",
      ">my_seq_marco_-0\n",
      "ACC\tTCT\tGTT\tCTG\tATT\tACC\tTGG\tATT\tAAA\tTTG\tATT\tATT\tCCC\tTTT\tATT\tCTG\tCAT\tCAT\tGTT\tATT\tAAA\tCTG\tATC\tGTT\tCCC\tGTT\tATT\tTCT\tATT\tTTG\tATT\tTCC\tAAA\tCTG\tATC\tGTT\tGTT\tCTG\tATT\tTCC\tATC\tGCC\tGAA\tTTG\tATC\tGTT\tTCT\tATT\tTGG\tCAT\tGTT\tGTT\tTCC\tTTT\tTTG\tAGC\tGTC\tGCT\tCAT\tCGA\tCAC\tGTC\tCGA\tACT\tGAA\tACT\tTTG\tGTT\tAGC\t\n",
      ">my_seq_marco_1\n",
      "TGC\tTAA\tCCA\tAAG\tTTT\tCAG\tTTC\tGGA\tCGT\tGTC\tGAT\tGAG\tCGA\tCGC\tTCA\tAAA\tAGG\tAAA\tCAA\tCAT\tGCC\tAAA\tTAG\tAAA\tCGA\tTCA\tATT\tCGG\tCGA\tTGG\tAAA\tTCA\tGAA\tCAA\tCGA\tTCA\tGTT\tTGG\tAAA\tTCA\tAAA\tTAG\tAAA\tTAA\tCGG\tGAA\tCGA\tTCA\tGTT\tTAA\tTAA\tCAT\tGAT\tGCA\tGAA\tTAA\tAGG\tGAA\tTAA\tTCA\tATT\tTAA\tTCC\tAGG\tTAA\tTCA\tGAA\tCAG\tAGG\t\n",
      ">my_seq_marco_-1\n",
      "CCT\tCTG\tTTC\tTGA\tTTA\tCCT\tGGA\tTTA\tAAT\tTGA\tTTA\tTTC\tCCT\tTTA\tTTC\tTGC\tATC\tATG\tTTA\tTTA\tAAC\tTGA\tTCG\tTTC\tCCG\tTTA\tTTT\tCTA\tTTT\tTGA\tTTT\tCCA\tAAC\tTGA\tTCG\tTTG\tTTC\tTGA\tTTT\tCCA\tTCG\tCCG\tAAT\tTGA\tTCG\tTTT\tCTA\tTTT\tGGC\tATG\tTTG\tTTT\tCCT\tTTT\tTGA\tGCG\tTCG\tCTC\tATC\tGAC\tACG\tTCC\tGAA\tCTG\tAAA\tCTT\tTGG\tTTA\tGCA\t\n",
      ">my_seq_marco_2\n",
      "GCT\tAAC\tCAA\tAGT\tTTC\tAGT\tTCG\tGAC\tGTG\tTCG\tATG\tAGC\tGAC\tGCT\tCAA\tAAA\tGGA\tAAC\tAAC\tATG\tCCA\tAAT\tAGA\tAAC\tGAT\tCAA\tTTC\tGGC\tGAT\tGGA\tAAT\tCAG\tAAC\tAAC\tGAT\tCAG\tTTT\tGGA\tAAT\tCAA\tAAT\tAGA\tAAT\tAAC\tGGG\tAAC\tGAT\tCAG\tTTT\tAAT\tAAC\tATG\tATG\tCAG\tAAT\tAAA\tGGG\tAAT\tAAT\tCAA\tTTT\tAAT\tCCA\tGGT\tAAT\tCAG\tAAC\tAGA\tGGT\t\n",
      ">my_seq_marco_-2\n",
      "CTC\tTGT\tTCT\tGAT\tTAC\tCTG\tGAT\tTAA\tATT\tGAT\tTAT\tTCC\tCTT\tTAT\tTCT\tGCA\tTCA\tTGT\tTAT\tTAA\tACT\tGAT\tCGT\tTCC\tCGT\tTAT\tTTC\tTAT\tTTT\tGAT\tTTC\tCAA\tACT\tGAT\tCGT\tTGT\tTCT\tGAT\tTTC\tCAT\tCGC\tCGA\tATT\tGAT\tCGT\tTTC\tTAT\tTTG\tGCA\tTGT\tTGT\tTTC\tCTT\tTTT\tGAG\tCGT\tCGC\tTCA\tTCG\tACA\tCGT\tCCG\tAAC\tTGA\tAAC\tTTT\tGGT\tTAG\tCAT\t"
     ]
    }
   ],
   "source": [
    "#seq.nt.fa\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "import re\n",
    "print(\"Enter the path of your fasta file\")\n",
    "path=input()\n",
    "#Se crea un diccionario con el archivo fasta\n",
    "dictionary = SeqIO.to_dict(SeqIO.parse(path, 'fasta'))\n",
    "\n",
    "for sequence in dictionary:\n",
    "    #Se ingresa a cada secuencia del archivo fasta y se almacena esa secuencia en ADN_forward\n",
    "    ADN_forward= dictionary[sequence].seq\n",
    "    ADN_reverse=ADN_forward.reverse_complement()\n",
    "    for ORF in range(3):\n",
    "        #Dos listas donde se guardarán los codones de la secuencia reverse y forward.De cada marco de lectura\n",
    "        codons_f=[]\n",
    "        codons_r=[]\n",
    "        strand_forward=ADN_forward[ORF:len(ADN_forward)]\n",
    "        strand_reverse=ADN_reverse[ORF:len(ADN_forward)]\n",
    "        print(\"\\n>my_seq_marco_{}\".format(ORF))\n",
    "        #Se divide la secuencia forward en codones y posteriormente se imprimen.\n",
    "        for codon in re.findall(r\"(.{3})\",str(strand_forward)):\n",
    "            codons_f.append(codon)\n",
    "        for codon in codons_f:\n",
    "            print(codon,end=\"\\t\") \n",
    "        #Se divide la secuencia reverse en codones y posteriormente se imprimen.\n",
    "        print(\"\\n>my_seq_marco_-{}\".format(ORF))\n",
    "        for codon in re.findall(r\"(.{3})\",str(strand_reverse)):\n",
    "            codons_r.append(codon)\n",
    "        for codon in codons_r:\n",
    "            print(codon,end=\"\\t\")\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p id=\"Ejercicio_3\"></p>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 3</h2>\n",
    "<ul>\n",
    "</ul>\n",
    "    <p>Obtén el <b>número</b> de lecturas cuyo promedio de calidad esté por debajo del umbral dado.</p>\n",
    "<p><span class=\"given-return\">Input:</span> Un umbral de calidad, Junto con un fastq.</p>\n",
    "\n",
    "<h3 id=\"sample-dataset\">Sample Dataset</h3>\n",
    "<div class=\"codehilite\"><pre>\n",
    "@Rosalind_0041\n",
    "GGCCGGTCTATTTACGTTCTCACCCGACGTGACGTACGGTCC\n",
    "+\n",
    "6.3536354;.151&lt;211/0?::6/-2051)-*&quot;40/.,+%)\n",
    "@Rosalind_0041\n",
    "TCGTATGCGTAGCACTTGGTACAGGAAGTGAACATCCAGGAT\n",
    "+\n",
    "AH@FGGGJ&lt;GB&lt;&lt;9:GD=D@GG9=?A@DC=;:?&gt;839/4856\n",
    "@Rosalind_0041\n",
    "ATTCGGTAATTGGCGTGAATCTGTTCTGACTGATAGAGACAA\n",
    "+\n",
    "@DJEJEA?JHJ@8?F?IA3=;8@C95=;=?;&gt;D/:;74792.\n",
    "</pre></div>\n",
    "\n",
    "\n",
    "<h3 id=\"sample-output\">Sample Output</h3>\n",
    "<div class=\"codehilite\"><pre>1\n",
    "</pre></div>\n",
    "</div> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter the path where your fastq is located\n",
      "sample.fastq\n",
      "Give a threshold of quality for the readings\n",
      "30\n",
      "Readings with quality threshold under the specified= 5\n"
     ]
    }
   ],
   "source": [
    "#sample.fastq\n",
    "from Bio import SeqIO\n",
    "from statistics import mean\n",
    "print(\"Enter the path where your fastq is located\")\n",
    "path=input()\n",
    "print(\"Give a threshold of quality for the readings\")\n",
    "quality=int(input())\n",
    "bad_quality=0\n",
    "for record in SeqIO.parse(path, \"fastq\"):\n",
    "   \n",
    "    if mean(record.letter_annotations[\"phred_quality\"])<quality:\n",
    "        bad_quality+=1\n",
    "\n",
    "print(\"Readings with quality threshold under the specified= {}\".format(bad_quality))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_4\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 4</h2>\n",
    "<ul>\n",
    "    El archivo hepatitis.gb contiene información sobre la región de ADN que codifica HBx, una proteína viral de la hepatitis B multifuncional involucrada en la modulación de varias vías al interactuar directa o indirectamente con factores del hospedero (degradación de proteínas, apoptosis, transcripción, transducción de señales, progreso del ciclo celular y estabilidad genética). Escribe un programa de Python que, utilizando este registro de genbank, guarde la secuencia de proteínas correspondiente en formato fasta.\n",
    "    </ul>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAARLYCQLDSSRDVLCLRPVGAESRGRPLARPLGAVSSPSPSAVPSDHGAHLSLRGLPVCAFSSXRSVCTSLHLCTLHGDHRERPSXPAQGLTKEDSWTLSNVNDRPRGLLQRLCDKRLGGVGGGDKVXXLCIRRL\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "#Se lee el archivo de GenBank y se guarda la info. en una variable\n",
    "gb_record= SeqIO.read(\"hepatitis.gb\", \"genbank\")\n",
    "#features[2]: ahí se encuentra \"CDS\". En CDS se encuentra la proteína que codifica la region de ADN.\n",
    "#Accedemos con la llave \"protein\" y accesamos a la casilla 0 para que nos regrese la proteína.\n",
    "protein=gb_record.features[2].qualifiers['translation'][0]\n",
    "protein=protein.format(\"fasta\")\n",
    "print(protein)\n",
    "#Creamos un archivo con la proteína obtenida en formato fasta.\n",
    "with open(\"protein.fasta\",\"w+\") as file:\n",
    "        file.write(\">seq1\\n\")\n",
    "        file.write(protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_5\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 5</h2>\n",
    "<ul>\n",
    "    \n",
    "+ Consigue el *structure* del archivo pdb 4hhb\n",
    "\n",
    "+ Imprime las coordenadas 3d del átomo C-alpha del aminoacido con residuo número 20 en chain C y asigna ese residuo a una variable.\n",
    "\n",
    "+ Calcula el centro geométrico (centroid) del aminoacido mencionado anteriormente e imprímelo (esto asume que las masas de los átomos son iguales)\n",
    "\n",
    "Para calular el centro geométrico se necesitan sacar las coordenadas de todos los átomos del residuo y sacar el promedio.\n",
    "</ul>\n",
    "</div>\n",
    "\n",
    "Con el ejemplo del residuo utilizado en **clase** debe darles:\n",
    "```python\n",
    ">>> centroid(residue)\n",
    "[27.96233346727159, -10.11222219467163, 56.41411124335395]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3D coordinates=[-26.912   8.767  11.705]\n",
      "centroid=[-28.1675      8.4300995  10.921499 ]\n"
     ]
    }
   ],
   "source": [
    "from Bio import PDB\n",
    "parser = PDB.PDBParser(QUIET=True)\n",
    "#Se parsea el archivo PDB y se consigue el structure. Se accede especificamente al residuo solicitado.\n",
    "structure = parser.get_structure(\"protein\", \"4hhb.pdb\")\n",
    "residue=structure[0][\"C\"][20]\n",
    "print(\"3D coordinates={}\".format(residue[\"CA\"].get_coord()))\n",
    "#Para obtener el centroid se usa un \"for\" para sumar todos las coordenadas y posteriormente se divide entre el total de coordenadas.\n",
    "n=0\n",
    "atoms=0\n",
    "for atom in residue:\n",
    "                atoms = atom.get_coord()+atoms\n",
    "                n+=1\n",
    "centroid=atoms/n\n",
    "\n",
    "print(\"centroid={}\".format(centroid))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_6\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">\n",
    "<h2 id=\"problem\">Ejercicio 6</h2>\n",
    "\n",
    "1. Usando el archivo test.blastout.xml parsea los resultados e imprime lo siguiente \"queryName\thitID\te-value\thitDesc\tHsp_num\" Si el valor es mejor (meno) que 1e-10. (el query name lo encuentran en el blast_record y el hitDesc dentro del alignment.hit_def)\n",
    "\n",
    "```python\n",
    ">>>blast_record.query_id\n",
    "'Query_1'\n",
    ">>>alignment.hit_id\n",
    "'sp|Q9Y267|S22AE_HUMAN'\n",
    ">>>alignment.hit_def\n",
    "'Solute carrier family 22 member 14 OS=Homo sapiens GN=SLC22A14 PE=2 SV=4'\n",
    ">>>hsp.expect\n",
    "2.35119e-06\n",
    "```\n",
    "\n",
    "|Query Name|Hit ID|e-Value|Hit Description|Hsp Number|\n",
    "|:---------:|:----:|:------:|:----:|:------:|\n",
    "|Query_1|sp\\|Q9Y267\\|S22AE_HUMAN|2.35119e-06|Solute carrier family 22 member 14|1|\n",
    "\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query Name\tHit ID\te-Value\tHit Description\tHsp Number\n",
      "Query_1\tsp|Q8CFZ5|S22AC_MOUSE\t1.47916e-109\tSolute carrier family 22 member 12 \t1\n",
      "Query_1\tsp|Q8CFZ5|S22AC_MOUSE\t1.20034e-74\tSolute carrier family 22 member 12 \t2\n",
      "Query_1\tsp|Q3ZAV1|S22AC_RAT\t7.69823e-103\tSolute carrier family 22 member 12 \t1\n",
      "Query_1\tsp|Q3ZAV1|S22AC_RAT\t1.31588e-62\tSolute carrier family 22 member 12 \t2\n",
      "Query_1\tsp|Q96S37|S22AC_HUMAN\t3.00718e-78\tSolute carrier family 22 member 12 \t1\n",
      "Query_1\tsp|Q96S37|S22AC_HUMAN\t4.9569e-46\tSolute carrier family 22 member 12 \t2\n",
      "Query_1\tsp|Q9NSA0|S22AB_HUMAN\t4.379e-59\tSolute carrier family 22 member 11 \t1\n",
      "Query_1\tsp|Q9NSA0|S22AB_HUMAN\t6.63169e-25\tSolute carrier family 22 member 11 \t2\n",
      "Query_1\tsp|Q2KIV1|S22A9_BOVIN\t2.16919e-49\tSolute carrier family 22 member 9 \t1\n",
      "Query_1\tsp|Q2KIV1|S22A9_BOVIN\t1.78901e-27\tSolute carrier family 22 member 9 \t2\n",
      "Query_1\tsp|Q8IVM8|S22A9_HUMAN\t7.4769e-47\tSolute carrier family 22 member 9 \t1\n",
      "Query_1\tsp|Q8IVM8|S22A9_HUMAN\t7.93154e-18\tSolute carrier family 22 member 9 \t2\n",
      "Query_1\tsp|Q76M72|S22AR_MOUSE\t8.2208e-47\tSolute carrier family 22 member 27 \t1\n",
      "Query_1\tsp|Q76M72|S22AR_MOUSE\t4.58388e-20\tSolute carrier family 22 member 27 \t2\n",
      "Query_1\tsp|Q63ZE4|S22AA_HUMAN\t9.76472e-47\tSolute carrier family 22 member 10 \t1\n",
      "Query_1\tsp|Q63ZE4|S22AA_HUMAN\t1.21117e-22\tSolute carrier family 22 member 10 \t2\n",
      "Query_1\tsp|Q6T423|S22AP_HUMAN\t1.417e-45\tSolute carrier family 22 member 25 \t1\n",
      "Query_1\tsp|Q6T423|S22AP_HUMAN\t5.2991e-20\tSolute carrier family 22 member 25 \t2\n",
      "Query_1\tsp|Q8N4F4|S22AO_HUMAN\t1.47336e-45\tSolute carrier family 22 member 24 \t1\n",
      "Query_1\tsp|Q5RC45|S22AO_PONAB\t2.77334e-45\tSolute carrier family 22 member 24 \t1\n",
      "Query_1\tsp|Q8VCA0|S22AJ_MOUSE\t4.76077e-42\tSolute carrier family 22 member 19 \t1\n",
      "Query_1\tsp|Q8VCA0|S22AJ_MOUSE\t6.39257e-21\tSolute carrier family 22 member 19 \t2\n",
      "Query_1\tsp|Q8R0S9|S22AM_MOUSE\t1.04333e-41\tSolute carrier family 22 member 22 \t1\n",
      "Query_1\tsp|Q8R0S9|S22AM_MOUSE\t1.43928e-19\tSolute carrier family 22 member 22 \t2\n",
      "Query_1\tsp|O35956|S22A6_RAT\t5.54738e-35\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|O35956|S22A6_RAT\t1.9653e-24\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q66J52|S226B_XENLA\t2.35073e-34\tSolute carrier family 22 member 6-B \t1\n",
      "Query_1\tsp|Q66J52|S226B_XENLA\t7.47759e-22\tSolute carrier family 22 member 6-B \t2\n",
      "Query_1\tsp|Q4U2R8|S22A6_HUMAN\t2.88427e-33\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q4U2R8|S22A6_HUMAN\t3.99476e-25\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q66J54|S226A_XENLA\t6.81013e-33\tSolute carrier family 22 member 6-A \t1\n",
      "Query_1\tsp|Q66J54|S226A_XENLA\t8.12649e-24\tSolute carrier family 22 member 6-A \t2\n",
      "Query_1\tsp|Q5RCH6|S22A6_PONAB\t1.66812e-32\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q5RCH6|S22A6_PONAB\t3.33423e-25\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q4W8A3|S22A6_MACFA\t5.67735e-32\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q4W8A3|S22A6_MACFA\t3.30207e-25\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q8VC69|S22A6_MOUSE\t7.98384e-32\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q8VC69|S22A6_MOUSE\t8.58509e-23\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q6NYN7|S22A6_DANRE\t1.28659e-30\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q6NYN7|S22A6_DANRE\t7.18428e-22\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q5R9C4|S22A8_PONAB\t1.87627e-30\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q5R9C4|S22A8_PONAB\t2.20276e-22\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|Q9TSY7|S22A6_RABIT\t3.07812e-30\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q9TSY7|S22A6_RABIT\t1.3715e-25\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q4W8A2|S22A8_MACFA\t8.12098e-30\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q4W8A2|S22A8_MACFA\t3.38035e-23\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|Q864Z3|S22A6_BOVIN\t9.46439e-30\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q864Z3|S22A6_BOVIN\t8.84813e-23\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|A6NK97|S22AK_HUMAN\t1.72054e-29\tSolute carrier family 22 member 20 \t1\n",
      "Query_1\tsp|A6NK97|S22AK_HUMAN\t4.39103e-24\tSolute carrier family 22 member 20 \t2\n",
      "Query_1\tsp|Q8TCC7|S22A8_HUMAN\t8.93333e-29\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q8TCC7|S22A8_HUMAN\t3.02317e-22\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|Q80UJ1|S22AK_MOUSE\t1.13083e-28\tSolute carrier family 22 member 20 \t1\n",
      "Query_1\tsp|Q80UJ1|S22AK_MOUSE\t1.29389e-26\tSolute carrier family 22 member 20 \t2\n",
      "Query_1\tsp|Q70BM6|S22A8_PIG\t5.43083e-28\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q70BM6|S22A8_PIG\t1.32801e-22\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|O88909|S22A8_MOUSE\t7.26336e-28\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|O88909|S22A8_MOUSE\t8.97063e-23\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|O57379|S22A6_PSEAM\t2.18954e-27\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|O57379|S22A6_PSEAM\t2.06833e-19\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|Q9R1U7|S22A8_RAT\t4.89152e-27\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q9R1U7|S22A8_RAT\t3.92799e-22\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|Q8HY24|S22A8_RABIT\t1.58173e-26\tSolute carrier family 22 member 8 \t1\n",
      "Query_1\tsp|Q8HY24|S22A8_RABIT\t1.62805e-20\tSolute carrier family 22 member 8 \t2\n",
      "Query_1\tsp|Q8MK48|S22A6_PIG\t3.11031e-25\tSolute carrier family 22 member 6 \t1\n",
      "Query_1\tsp|Q8MK48|S22A6_PIG\t8.65746e-24\tSolute carrier family 22 member 6 \t2\n",
      "Query_1\tsp|A6QLW8|S22A7_BOVIN\t4.80411e-23\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|A6QLW8|S22A7_BOVIN\t3.59154e-17\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q9Y694|S22A7_HUMAN\t5.10092e-22\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q9Y694|S22A7_HUMAN\t2.89038e-15\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q1RPP5|S22A7_PIG\t2.87209e-21\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q1RPP5|S22A7_PIG\t1.02504e-17\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q5RLM2|S22A7_RAT\t4.01662e-21\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q5RLM2|S22A7_RAT\t1.46591e-16\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q5R540|S22A7_PONAB\t5.45494e-21\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q5R540|S22A7_PONAB\t1.13206e-15\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q91WU2|S22A7_MOUSE\t3.67797e-20\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q91WU2|S22A7_MOUSE\t3.67157e-16\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q3YAW7|S22A7_RABIT\t1.70574e-19\tSolute carrier family 22 member 7 \t1\n",
      "Query_1\tsp|Q3YAW7|S22A7_RABIT\t7.37303e-16\tSolute carrier family 22 member 7 \t2\n",
      "Query_1\tsp|Q9Z0E8|S22A5_MOUSE\t5.39632e-17\tSolute carrier family 22 member 5 \t1\n",
      "Query_1\tsp|Q9WTN6|S22AL_MOUSE\t2.46943e-16\tSolute carrier family 22 member 21 \t1\n",
      "Query_1\tsp|Q9WTN6|S22AL_MOUSE\t4.08638e-08\tSolute carrier family 22 member 21 \t2\n",
      "Query_1\tsp|O70594|S22A5_RAT\t4.44505e-15\tSolute carrier family 22 member 5 \t1\n",
      "Query_1\tsp|Q9Z306|S22A4_MOUSE\t1.7029e-14\tSolute carrier family 22 member 4 \t1\n",
      "Query_1\tsp|Q9Y226|S22AD_HUMAN\t4.16716e-14\tSolute carrier family 22 member 13 \t1\n",
      "Query_1\tsp|Q9Y226|S22AD_HUMAN\t1.8416e-09\tSolute carrier family 22 member 13 \t2\n",
      "Query_1\tsp|Q9R141|S22A4_RAT\t8.25159e-14\tSolute carrier family 22 member 4 \t1\n",
      "Query_1\tsp|Q9H015|S22A4_HUMAN\t3.12319e-13\tSolute carrier family 22 member 4 \t1\n",
      "Query_1\tsp|A9CB25|S22A4_PAPAN\t1.56162e-12\tSolute carrier family 22 member 4 \t1\n",
      "Query_1\tsp|O76082|S22A5_HUMAN\t1.5994e-12\tSolute carrier family 22 member 5 \t1\n",
      "Query_1\tsp|O76082|S22A5_HUMAN\t6.99512e-06\tSolute carrier family 22 member 5 \t2\n",
      "Query_1\tsp|Q17QN9|S22AG_BOVIN\t4.53963e-12\tSolute carrier family 22 member 16 \t1\n",
      "Query_1\tsp|Q6A4L0|S22AD_MOUSE\t6.54368e-11\tSolute carrier family 22 member 13 \t1\n",
      "Query_1\tsp|Q6A4L0|S22AD_MOUSE\t2.53797e-08\tSolute carrier family 22 member 13 \t2\n",
      "Query_1\tsp|Q86VW1|S22AG_HUMAN\t1.02212e-10\tSolute carrier family 22 member 16 \t1\n",
      "Query_1\tsp|Q66KG0|S22AG_XENLA\t1.96989e-10\tSolute carrier family 22 member 16 \t1\n",
      "Query_1\tsp|Q9VIK2|CART_DROME\t1.99728e-10\tCarcinine transporter \t1\n",
      "Query_1\tsp|O08966|S22A1_MOUSE\t2.24409e-10\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|O75751|S22A3_HUMAN\t1.3777e-09\tSolute carrier family 22 member 3 \t1\n",
      "Query_1\tsp|Q497L8|S22AG_MOUSE\t2.23665e-09\tSolute carrier family 22 member 16 \t1\n",
      "Query_1\tsp|Q63089|S22A1_RAT\t3.03707e-09\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|Q9VCA2|ORCT_DROME\t5.53116e-09\tOrganic cation transporter protein \t1\n",
      "Query_1\tsp|Q9VCA2|ORCT_DROME\t8.87519e-06\tOrganic cation transporter protein \t2\n",
      "Query_1\tsp|O15244|S22A2_HUMAN\t7.19125e-09\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|O15244|S22A2_HUMAN\t1.09637e-08\tSolute carrier family 22 member 2 \t2\n",
      "Query_1\tsp|Q6NUB3|S22AF_XENLA\t1.84566e-08\tSolute carrier family 22 member 15 \t1\n",
      "Query_1\tsp|Q6DFR1|S22AF_XENTR\t2.74301e-08\tSolute carrier family 22 member 15 \t1\n",
      "Query_1\tsp|Q9U539|OCT1_CAEEL\t3.39166e-08\tOrganic cation transporter 1 \t1\n",
      "Query_1\tsp|Q9U539|OCT1_CAEEL\t2.05964e-06\tOrganic cation transporter 1 \t2\n",
      "Query_1\tsp|O88446|S22A3_RAT\t3.40113e-08\tSolute carrier family 22 member 3 \t1\n",
      "Query_1\tsp|Q9WTW5|S22A3_MOUSE\t3.7956e-08\tSolute carrier family 22 member 3 \t1\n",
      "Query_1\tsp|O77504|S22A1_RABIT\t6.46373e-08\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|O02713|S22A2_PIG\t6.82784e-08\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|O02713|S22A2_PIG\t3.93141e-07\tSolute carrier family 22 member 2 \t2\n",
      "Query_1\tsp|A7MBE0|S22A1_BOVIN\t7.39466e-08\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|A7MBE0|S22A1_BOVIN\t2.22956e-07\tSolute carrier family 22 member 1 \t2\n",
      "Query_1\tsp|Q95R48|OCTL_DROME\t9.31713e-08\tOrganic cation transporter-like protein \t1\n",
      "Query_1\tsp|Q95R48|OCTL_DROME\t8.74847e-06\tOrganic cation transporter-like protein \t2\n",
      "Query_1\tsp|Q9R0W2|S22A2_RAT\t1.45742e-07\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|Q9R0W2|S22A2_RAT\t4.36294e-06\tSolute carrier family 22 member 2 \t2\n",
      "Query_1\tsp|Q5R5H7|S22A2_PONAB\t1.59657e-07\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|Q5R5H7|S22A2_PONAB\t2.99398e-07\tSolute carrier family 22 member 2 \t2\n",
      "Query_1\tsp|Q9CAT6|OCT1_ARATH\t2.48766e-07\tOrganic cation/carnitine transporter 1 \t1\n",
      "Query_1\tsp|Q863T6|S22A1_PIG\t2.96466e-07\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|Q863T6|S22A1_PIG\t5.65724e-07\tSolute carrier family 22 member 1 \t2\n",
      "Query_1\tsp|O70577|S22A2_MOUSE\t4.22523e-07\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|O70577|S22A2_MOUSE\t1.79204e-06\tSolute carrier family 22 member 2 \t2\n",
      "Query_1\tsp|Q8MJI6|S22A2_RABIT\t5.65724e-07\tSolute carrier family 22 member 2 \t1\n",
      "Query_1\tsp|Q8IZD6|S22AF_HUMAN\t5.83661e-07\tSolute carrier family 22 member 15 \t1\n",
      "Query_1\tsp|Q504N2|S22AF_MOUSE\t9.86673e-07\tSolute carrier family 22 member 15 \t1\n",
      "Query_1\tsp|O15245|S22A1_HUMAN\t1.03072e-06\tSolute carrier family 22 member 1 \t1\n",
      "Query_1\tsp|O15245|S22A1_HUMAN\t3.47635e-06\tSolute carrier family 22 member 1 \t2\n",
      "Query_1\tsp|Q9Y267|S22AE_HUMAN\t2.35119e-06\tSolute carrier family 22 member 14 \t1\n",
      "Query_1\tsp|Q9Y267|S22AE_HUMAN\t6.30785e-06\tSolute carrier family 22 member 14 \t2\n"
     ]
    }
   ],
   "source": [
    "\n",
    "from Bio.Blast import NCBIXML\n",
    "print(\"{}\\t{}\\t{}\\t{}\\t{}\".format(\"Query Name\",\"Hit ID\",\"e-Value\",\"Hit Description\",\"Hsp Number\"))\n",
    "result_handle = open(\"test.blastout.xml\")\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "for record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        i=0\n",
    "        for hsp in alignment.hsps:\n",
    "            i+=1\n",
    "            hit_def=alignment.hit_def.split(\"OS=\")\n",
    "            hit_def=hit_def[0]\n",
    "            print(\"{}\\t{}\\t{}\\t{}\\t{}\".format(record.query_id,alignment.hit_id,hsp.expect,hit_def,i))\n",
    "        "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
