{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Copyright 2007 by Michiel de Hoon.  All rights reserved.\n",
    "# This code is part of the Biopython distribution and governed by its\n",
    "# license.  Please see the LICENSE file that should have been included\n",
    "# as part of this package.\n",
    "\"\"\"Code to work with the sprotXX.dat file from SwissProt.\n",
    "\n",
    "https://web.expasy.org/docs/userman.html\n",
    "\n",
    "Classes:\n",
    " - Record             Holds SwissProt data.\n",
    " - Reference          Holds reference data from a SwissProt record.\n",
    "\n",
    "Functions:\n",
    " - read               Read one SwissProt record\n",
    " - parse              Read multiple SwissProt records\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "\n",
    "import io\n",
    "\n",
    "from Bio.SeqFeature import (\n",
    "    SeqFeature,\n",
    "    FeatureLocation,\n",
    "    ExactPosition,\n",
    "    BeforePosition,\n",
    "    AfterPosition,\n",
    "    UncertainPosition,\n",
    "    UnknownPosition,\n",
    ")\n",
    "\n",
    "\n",
    "class SwissProtParserError(ValueError):\n",
    "    \"\"\"An error occurred while parsing a SwissProt file.\"\"\"\n",
    "\n",
    "    def __init__(self, *args, line=None):\n",
    "        \"\"\"Create a SwissProtParserError object with the offending line.\"\"\"\n",
    "        super().__init__(*args)\n",
    "        self.line = line\n",
    "\n",
    "\n",
    "class Record:\n",
    "    \"\"\"Holds information from a SwissProt record.\n",
    "\n",
    "    Attributes:\n",
    "     - entry_name        Name of this entry, e.g. RL1_ECOLI.\n",
    "     - data_class        Either 'STANDARD' or 'PRELIMINARY'.\n",
    "     - molecule_type     Type of molecule, 'PRT',\n",
    "     - sequence_length   Number of residues.\n",
    "     - accessions        List of the accession numbers, e.g. ['P00321']\n",
    "     - created           A tuple of (date, release).\n",
    "     - sequence_update   A tuple of (date, release).\n",
    "     - annotation_update A tuple of (date, release).\n",
    "     - description       Free-format description.\n",
    "     - gene_name         Gene name.  See userman.txt for description.\n",
    "     - organism          The source of the sequence.\n",
    "     - organelle         The origin of the sequence.\n",
    "     - organism_classification  The taxonomy classification.  List of strings.\n",
    "       (http://www.ncbi.nlm.nih.gov/Taxonomy/)\n",
    "     - taxonomy_id       A list of NCBI taxonomy id's.\n",
    "     - host_organism     A list of names of the hosts of a virus, if any.\n",
    "     - host_taxonomy_id  A list of NCBI taxonomy id's of the hosts, if any.\n",
    "     - references        List of Reference objects.\n",
    "     - comments          List of strings.\n",
    "     - cross_references  List of tuples (db, id1[, id2][, id3]).  See the docs.\n",
    "     - keywords          List of the keywords.\n",
    "     - features          List of tuples (key name, from, to, description).\n",
    "       from and to can be either integers for the residue\n",
    "       numbers, '<', '>', or '?'\n",
    "     - protein_existence Numerical value describing the evidence for the existence of the protein.\n",
    "     - seqinfo           tuple of (length, molecular weight, CRC32 value)\n",
    "     - sequence          The sequence.\n",
    "\n",
    "    Examples\n",
    "    --------\n",
    "    >>> from Bio import SwissProt\n",
    "    >>> example_filename = \"SwissProt/sp008\"\n",
    "    >>> with open(example_filename) as handle:\n",
    "    ...     records = SwissProt.parse(handle)\n",
    "    ...     for record in records:\n",
    "    ...         print(record.entry_name)\n",
    "    ...         print(\",\".join(record.accessions))\n",
    "    ...         print(record.keywords)\n",
    "    ...         print(repr(record.organism))\n",
    "    ...         print(record.sequence[:20] + \"...\")\n",
    "    ...\n",
    "    1A02_HUMAN\n",
    "    P01892,P06338,P30514,P30444,P30445,P30446,Q29680,Q29899,Q95352,Q29837,Q95380\n",
    "    ['MHC I', 'Transmembrane', 'Glycoprotein', 'Signal', 'Polymorphism', '3D-structure']\n",
    "    'Homo sapiens (Human).'\n",
    "    MAVMAPRTLVLLLSGALALT...\n",
    "\n",
    "    \"\"\"\n",
    "\n",
    "    def __init__(self):\n",
    "        \"\"\"Initialize the class.\"\"\"\n",
    "        self.entry_name = None\n",
    "        self.data_class = None\n",
    "        self.molecule_type = None\n",
    "        self.sequence_length = None\n",
    "\n",
    "        self.accessions = []\n",
    "        self.created = None\n",
    "        self.sequence_update = None\n",
    "        self.annotation_update = None\n",
    "\n",
    "        self.description = []\n",
    "        self.gene_name = \"\"\n",
    "        self.organism = []\n",
    "        self.organelle = \"\"\n",
    "        self.organism_classification = []\n",
    "        self.taxonomy_id = []\n",
    "        self.host_organism = []\n",
    "        self.host_taxonomy_id = []\n",
    "        self.references = []\n",
    "        self.comments = []\n",
    "        self.cross_references = []\n",
    "        self.keywords = []\n",
    "        self.features = []\n",
    "        self.protein_existence = \"\"\n",
    "\n",
    "        self.seqinfo = None\n",
    "        self.sequence = \"\"\n",
    "\n",
    "\n",
    "class Reference:\n",
    "    \"\"\"Holds information from one reference in a SwissProt entry.\n",
    "\n",
    "    Attributes:\n",
    "     - number      Number of reference in an entry.\n",
    "     - evidence    Evidence code.  List of strings.\n",
    "     - positions   Describes extent of work.  List of strings.\n",
    "     - comments    Comments.  List of (token, text).\n",
    "     - references  References.  List of (dbname, identifier).\n",
    "     - authors     The authors of the work.\n",
    "     - title       Title of the work.\n",
    "     - location    A citation for the work.\n",
    "\n",
    "    \"\"\"\n",
    "\n",
    "    def __init__(self):\n",
    "        \"\"\"Initialize the class.\"\"\"\n",
    "        self.number = None\n",
    "        self.positions = []\n",
    "        self.comments = []\n",
    "        self.references = []\n",
    "        self.authors = []\n",
    "        self.title = []\n",
    "        self.location = []\n",
    "\n",
    "\n",
    "class FeatureTable(SeqFeature):\n",
    "    \"\"\"Stores feature annotations for specific regions of the sequence.\n",
    "\n",
    "    This is a subclass of SeqFeature, defined in Bio.SeqFeature, where the\n",
    "    attributes are used as follows:\n",
    "\n",
    "     - ``location``: location of the feature on the canonical or isoform\n",
    "       sequence; the location is stored as an instance of FeatureLocation,\n",
    "       defined in Bio.SeqFeature, with the ref attribute set to the isoform\n",
    "       ID referring to the canonical or isoform sequence on which the feature\n",
    "       is defined\n",
    "     - ``id``: unique and stable identifier (FTId), only provided for features\n",
    "       belonging to the types CARBOHYD, CHAIN, PEPTIDE, PROPEP, VARIANT, or\n",
    "       VAR_SEQ\n",
    "     - ``type``: indicates the type of feature, as defined by the UniProt\n",
    "       Knowledgebase documentation:\n",
    "\n",
    "        - ACT_SITE: amino acid(s) involved in the activity of an enzyme\n",
    "        - BINDING:  binding site for any chemical group\n",
    "        - CARBOHYD: glycosylation site; an FTId identifier to the GlyConnect\n",
    "          database is provided if annotated there\n",
    "        - CA_BIND:  calcium-binding region\n",
    "        - CHAIN:    polypeptide chain in the mature protein\n",
    "        - COILED:   coiled-coil region\n",
    "        - COMPBIAS: compositionally biased region\n",
    "        - CONFLICT: different sources report differing sequences\n",
    "        - CROSSLNK: posttransationally formed amino acid bond\n",
    "        - DISULFID: disulfide bond\n",
    "        - DNA_BIND: DNA-binding region\n",
    "        - DOMAIN:   domain, defined as a specific combination of secondary\n",
    "          structures organized into a characteristic three-dimensional\n",
    "          structure or fold\n",
    "        - INIT_MET: initiator methionine\n",
    "        - INTRAMEM: region located in a membrane without crossing it\n",
    "        - HELIX:    alpha-, 3(10)-, or pi-helix secondary structure\n",
    "        - LIPID:    covalent binding of a lipid moiety\n",
    "        - METAL:    binding site for a metal ion\n",
    "        - MOD_RES:  posttranslational modification (PTM) of a residue,\n",
    "          annotated by the controlled vocabulary defined by the ptmlist.txt\n",
    "          document on the UniProt website\n",
    "        - MOTIF:    short sequence motif of biological interest\n",
    "        - MUTAGEN:  site experimentally altered by mutagenesis\n",
    "        - NON_CONS: non-consecutive residues\n",
    "        - NON_STD:  non-standard amino acid\n",
    "        - NON_TER:  the residue at an extremity of the sequence is not the\n",
    "          terminal residue\n",
    "        - NP_BIND:  nucleotide phosphate-binding region\n",
    "        - PEPTIDE:  released active mature polypeptide\n",
    "        - PROPEP:   any processed propeptide\n",
    "        - REGION:   region of interest in the sequence\n",
    "        - REPEAT:   internal sequence repetition\n",
    "        - SIGNAL:   signal sequence (prepeptide)\n",
    "        - SITE:     amino-acid site of interest not represented by another\n",
    "          feature key\n",
    "        - STRAND:   beta-strand secondary structure; either a hydrogen-bonded\n",
    "          extended beta strand or a residue in an isolated beta-bridge\n",
    "        - TOPO_DOM: topological domain\n",
    "        - TRANSIT:  transit peptide (mitochondrion, chloroplast, thylakoid,\n",
    "          cyanelle, peroxisome, etc.)\n",
    "        - TRANSMEM: transmembrane region\n",
    "        - TURN:     H-bonded turn (3-, 4-, or 5-turn)\n",
    "        - UNSURE:   uncertainties in the sequence\n",
    "        - VARIANT:  sequence variant; an FTId is provided for protein sequence\n",
    "          variants of Hominidae (great apes and humans)\n",
    "        - VAR_SEQ:  sequence variant produced by alternative splicing,\n",
    "          alternative promoter usage, alternative initiation, or ribosomal\n",
    "          frameshifting\n",
    "        - ZN_FING:  zinc finger region\n",
    "\n",
    "     - qualifiers   A dictionary of additional information, which may include\n",
    "       the feature evidence and free-text notes. While SwissProt includes the\n",
    "       feature identifier code (FTId) as a qualifier, it is stored as the\n",
    "       attribute ID of the FeatureTable object.\n",
    "\n",
    "    \"\"\"\n",
    "\n",
    "\n",
    "def parse(source):\n",
    "    \"\"\"Read multiple SwissProt records from file.\n",
    "\n",
    "    Argument source is a file-like object or a path to a file.\n",
    "\n",
    "    Returns a generator object which yields Bio.SwissProt.Record() objects.\n",
    "    \"\"\"\n",
    "    handle = _open(source)\n",
    "    try:\n",
    "        while True:\n",
    "            record = _read(handle)\n",
    "            if not record:\n",
    "                return\n",
    "            yield record\n",
    "    finally:\n",
    "        if handle is not source:\n",
    "            handle.close()\n",
    "\n",
    "\n",
    "def read(source):\n",
    "    \"\"\"Read one SwissProt record from file.\n",
    "\n",
    "    Argument source is a file-like object or a path to a file.\n",
    "\n",
    "    Returns a Record() object.\n",
    "    \"\"\"\n",
    "    handle = _open(source)\n",
    "    try:\n",
    "        record = _read(handle)\n",
    "        if not record:\n",
    "            raise ValueError(\"No SwissProt record found\")\n",
    "        # We should have reached the end of the record by now.\n",
    "        # Try to read one more line to be sure:\n",
    "        try:\n",
    "            next(handle)\n",
    "        except StopIteration:\n",
    "            return record\n",
    "        raise ValueError(\"More than one SwissProt record found\")\n",
    "    finally:\n",
    "        if handle is not source:\n",
    "            handle.close()\n",
    "\n",
    "\n",
    "# Everything below is considered private\n",
    "\n",
    "\n",
    "def _open(source):\n",
    "    try:\n",
    "        handle = open(source)\n",
    "        return handle\n",
    "    except TypeError:\n",
    "        handle = source\n",
    "        if handle.read(0) == \"\":\n",
    "            # handle is text; assume the encoding is compatible with ASCII\n",
    "            return handle\n",
    "        # handle is binary; SwissProt encoding is always ASCII\n",
    "        return io.TextIOWrapper(handle, encoding=\"ASCII\")\n",
    "\n",
    "\n",
    "def _read(handle):\n",
    "    record = None\n",
    "    unread = \"\"\n",
    "    try:\n",
    "        line = next(handle)\n",
    "    except StopIteration:\n",
    "        return record\n",
    "    key, value = line[:2], line[5:].rstrip()\n",
    "    if key != \"ID\":\n",
    "        raise SwissProtParserError(\"Failed to find ID in first line\", line=line)\n",
    "    record = Record()\n",
    "    _read_id(record, line)\n",
    "    _sequence_lines = []\n",
    "    for line in handle:\n",
    "        key, value = line[:2], line[5:].rstrip()\n",
    "        if unread:\n",
    "            value = unread + \" \" + value\n",
    "            unread = \"\"\n",
    "        if key == \"AC\":\n",
    "            accessions = value.rstrip(\";\").split(\"; \")\n",
    "            record.accessions.extend(accessions)\n",
    "        elif key == \"DT\":\n",
    "            _read_dt(record, line)\n",
    "        elif key == \"DE\":\n",
    "            record.description.append(value.strip())\n",
    "        elif key == \"GN\":\n",
    "            if record.gene_name:\n",
    "                record.gene_name += \" \"\n",
    "            record.gene_name += value\n",
    "        elif key == \"OS\":\n",
    "            record.organism.append(value)\n",
    "        elif key == \"OG\":\n",
    "            record.organelle += line[5:]\n",
    "        elif key == \"OC\":\n",
    "            cols = value.rstrip(\";.\").split(\"; \")\n",
    "            record.organism_classification.extend(cols)\n",
    "        elif key == \"OX\":\n",
    "            _read_ox(record, line)\n",
    "        elif key == \"OH\":\n",
    "            _read_oh(record, line)\n",
    "        elif key == \"RN\":\n",
    "            reference = Reference()\n",
    "            _read_rn(reference, value)\n",
    "            record.references.append(reference)\n",
    "        elif key == \"RP\":\n",
    "            assert record.references, \"RP: missing RN\"\n",
    "            record.references[-1].positions.append(value)\n",
    "        elif key == \"RC\":\n",
    "            assert record.references, \"RC: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            unread = _read_rc(reference, value)\n",
    "        elif key == \"RX\":\n",
    "            assert record.references, \"RX: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            _read_rx(reference, value)\n",
    "        elif key == \"RL\":\n",
    "            assert record.references, \"RL: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            reference.location.append(value)\n",
    "        # In UniProt release 1.12 of 6/21/04, there is a new RG\n",
    "        # (Reference Group) line, which references a group instead of\n",
    "        # an author.  Each block must have at least 1 RA or RG line.\n",
    "        elif key == \"RA\":\n",
    "            assert record.references, \"RA: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            reference.authors.append(value)\n",
    "        elif key == \"RG\":\n",
    "            assert record.references, \"RG: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            reference.authors.append(value)\n",
    "        elif key == \"RT\":\n",
    "            assert record.references, \"RT: missing RN\"\n",
    "            reference = record.references[-1]\n",
    "            reference.title.append(value)\n",
    "        elif key == \"CC\":\n",
    "            _read_cc(record, line)\n",
    "        elif key == \"DR\":\n",
    "            _read_dr(record, value)\n",
    "        elif key == \"PE\":\n",
    "            _read_pe(record, value)\n",
    "        elif key == \"KW\":\n",
    "            _read_kw(record, value)\n",
    "        elif key == \"FT\":\n",
    "            _read_ft(record, line)\n",
    "        elif key == \"SQ\":\n",
    "            cols = value.split()\n",
    "            assert len(cols) == 7, \"I don't understand SQ line %s\" % line\n",
    "            # Do more checking here?\n",
    "            record.seqinfo = int(cols[1]), int(cols[3]), cols[5]\n",
    "        elif key == \"  \":\n",
    "            _sequence_lines.append(value.replace(\" \", \"\").rstrip())\n",
    "        elif key == \"//\":\n",
    "            # Join multiline data into one string\n",
    "            record.description = \" \".join(record.description)\n",
    "            record.organism = \" \".join(record.organism)\n",
    "            record.organelle = record.organelle.rstrip()\n",
    "            for reference in record.references:\n",
    "                reference.authors = \" \".join(reference.authors).rstrip(\";\")\n",
    "                if reference.title:\n",
    "                    title = reference.title[0]\n",
    "                    for fragment in reference.title[1:]:\n",
    "                        if not title.endswith(\"-\"):\n",
    "                            title += \" \"\n",
    "                        title += fragment\n",
    "                    title = title.rstrip(\";\")\n",
    "                    if title.startswith('\"') and title.endswith('\"'):\n",
    "                        title = title[1:-1]  # remove quotes\n",
    "                else:\n",
    "                    title = \"\"\n",
    "                reference.title = title\n",
    "                reference.location = \" \".join(reference.location)\n",
    "            record.sequence = \"\".join(_sequence_lines)\n",
    "            return record\n",
    "        elif key == \"**\":\n",
    "            # Do this one last, as it will almost never occur.\n",
    "            # See Bug 2353, some files from the EBI have extra lines\n",
    "            # starting \"**\" (two asterisks/stars).  They appear\n",
    "            # to be unofficial automated annotations. e.g.\n",
    "            # **\n",
    "            # **   #################    INTERNAL SECTION    ##################\n",
    "            # **HA SAM; Annotated by PicoHamap 1.88; MF_01138.1; 09-NOV-2003.\n",
    "            pass\n",
    "        else:\n",
    "            raise SwissProtParserError(\"Unknown keyword '%s' found\" % key, line=line)\n",
    "    if record:\n",
    "        raise ValueError(\"Unexpected end of stream.\")\n",
    "\n",
    "\n",
    "def _read_id(record, line):\n",
    "    cols = line[5:].split()\n",
    "    # Prior to release 51, included with MoleculeType:\n",
    "    # ID   EntryName DataClass; MoleculeType; SequenceLength AA.\n",
    "    #\n",
    "    # Newer files lack the MoleculeType:\n",
    "    # ID   EntryName DataClass; SequenceLength AA.\n",
    "    if len(cols) == 5:\n",
    "        record.entry_name = cols[0]\n",
    "        record.data_class = cols[1].rstrip(\";\")\n",
    "        record.molecule_type = cols[2].rstrip(\";\")\n",
    "        record.sequence_length = int(cols[3])\n",
    "    elif len(cols) == 4:\n",
    "        record.entry_name = cols[0]\n",
    "        record.data_class = cols[1].rstrip(\";\")\n",
    "        record.molecule_type = None\n",
    "        record.sequence_length = int(cols[2])\n",
    "    else:\n",
    "        raise SwissProtParserError(\"ID line has unrecognised format\", line=line)\n",
    "    # check if the data class is one of the allowed values\n",
    "    allowed = (\"STANDARD\", \"PRELIMINARY\", \"IPI\", \"Reviewed\", \"Unreviewed\")\n",
    "    if record.data_class not in allowed:\n",
    "        message = \"Unrecognized data class '%s'\" % record.data_class\n",
    "        raise SwissProtParserError(message, line=line)\n",
    "\n",
    "    # molecule_type should be 'PRT' for PRoTein\n",
    "    # Note that has been removed in recent releases (set to None)\n",
    "    if record.molecule_type not in (None, \"PRT\"):\n",
    "        message = \"Unrecognized molecule type '%s'\" % record.molecule_type\n",
    "        raise SwissProtParserError(message, line=line)\n",
    "\n",
    "\n",
    "def _read_dt(record, line):\n",
    "    value = line[5:]\n",
    "    uprline = value.upper()\n",
    "    cols = value.rstrip().split()\n",
    "    if (\n",
    "        \"CREATED\" in uprline\n",
    "        or \"LAST SEQUENCE UPDATE\" in uprline\n",
    "        or \"LAST ANNOTATION UPDATE\" in uprline\n",
    "    ):\n",
    "        # Old style DT line\n",
    "        # =================\n",
    "        # e.g.\n",
    "        # DT   01-FEB-1995 (Rel. 31, Created)\n",
    "        # DT   01-FEB-1995 (Rel. 31, Last sequence update)\n",
    "        # DT   01-OCT-2000 (Rel. 40, Last annotation update)\n",
    "        #\n",
    "        # or:\n",
    "        # DT   08-JAN-2002 (IPI Human rel. 2.3, Created)\n",
    "        # ...\n",
    "\n",
    "        # find where the version information will be located\n",
    "        # This is needed for when you have cases like IPI where\n",
    "        # the release version is in a different spot:\n",
    "        # DT   08-JAN-2002 (IPI Human rel. 2.3, Created)\n",
    "        uprcols = uprline.split()\n",
    "        rel_index = -1\n",
    "        for index in range(len(uprcols)):\n",
    "            if \"REL.\" in uprcols[index]:\n",
    "                rel_index = index\n",
    "        assert rel_index >= 0, \"Could not find Rel. in DT line: %s\" % line\n",
    "        version_index = rel_index + 1\n",
    "        # get the version information\n",
    "        str_version = cols[version_index].rstrip(\",\")\n",
    "        # no version number\n",
    "        if str_version == \"\":\n",
    "            version = 0\n",
    "        # dot versioned\n",
    "        elif \".\" in str_version:\n",
    "            version = str_version\n",
    "        # integer versioned\n",
    "        else:\n",
    "            version = int(str_version)\n",
    "        date = cols[0]\n",
    "\n",
    "        if \"CREATED\" in uprline:\n",
    "            record.created = date, version\n",
    "        elif \"LAST SEQUENCE UPDATE\" in uprline:\n",
    "            record.sequence_update = date, version\n",
    "        elif \"LAST ANNOTATION UPDATE\" in uprline:\n",
    "            record.annotation_update = date, version\n",
    "        else:\n",
    "            raise SwissProtParserError(\"Unrecognised DT (DaTe) line\", line=line)\n",
    "    elif (\n",
    "        \"INTEGRATED INTO\" in uprline\n",
    "        or \"SEQUENCE VERSION\" in uprline\n",
    "        or \"ENTRY VERSION\" in uprline\n",
    "    ):\n",
    "        # New style DT line\n",
    "        # =================\n",
    "        # As of UniProt Knowledgebase release 7.0 (including\n",
    "        # Swiss-Prot release 49.0 and TrEMBL release 32.0) the\n",
    "        # format of the DT lines and the version information\n",
    "        # in them was changed - the release number was dropped.\n",
    "        #\n",
    "        # For more information see bug 1948 and\n",
    "        # http://ca.expasy.org/sprot/relnotes/sp_news.html#rel7.0\n",
    "        #\n",
    "        # e.g.\n",
    "        # DT   01-JAN-1998, integrated into UniProtKB/Swiss-Prot.\n",
    "        # DT   15-OCT-2001, sequence version 3.\n",
    "        # DT   01-APR-2004, entry version 14.\n",
    "        #\n",
    "        # This is a new style DT line...\n",
    "\n",
    "        # The date should be in string cols[1]\n",
    "        # Get the version number if there is one.\n",
    "        # For the three DT lines above: 0, 3, 14\n",
    "        try:\n",
    "            version = 0\n",
    "            for s in cols[-1].split(\".\"):\n",
    "                if s.isdigit():\n",
    "                    version = int(s)\n",
    "        except ValueError:\n",
    "            version = 0\n",
    "        date = cols[0].rstrip(\",\")\n",
    "\n",
    "        # Re-use the historical property names, even though\n",
    "        # the meaning has changed slighty:\n",
    "        if \"INTEGRATED\" in uprline:\n",
    "            record.created = date, version\n",
    "        elif \"SEQUENCE VERSION\" in uprline:\n",
    "            record.sequence_update = date, version\n",
    "        elif \"ENTRY VERSION\" in uprline:\n",
    "            record.annotation_update = date, version\n",
    "        else:\n",
    "            raise SwissProtParserError(\"Unrecognised DT (DaTe) line\", line=line)\n",
    "    else:\n",
    "        raise SwissProtParserError(\"Failed to parse DT (DaTe) line\", line=line)\n",
    "\n",
    "\n",
    "def _read_ox(record, line):\n",
    "    # The OX line used to be in the simple format:\n",
    "    # OX   DESCRIPTION=ID[, ID]...;\n",
    "    # If there are too many id's to fit onto a line, then the ID's\n",
    "    # continue directly onto the next line, e.g.\n",
    "    # OX   DESCRIPTION=ID[, ID]...\n",
    "    # OX   ID[, ID]...;\n",
    "    # Currently, the description is always \"NCBI_TaxID\".\n",
    "    # To parse this, I need to check to see whether I'm at the\n",
    "    # first line.  If I am, grab the description and make sure\n",
    "    # it's an NCBI ID.  Then, grab all the id's.\n",
    "    #\n",
    "    # As of the 2014-10-01 release, there may be an evidence code, e.g.\n",
    "    # OX   NCBI_TaxID=418404 {ECO:0000313|EMBL:AEX14553.1};\n",
    "    # In the short term, we will ignore any evidence codes:\n",
    "    line = line.split(\"{\")[0]\n",
    "    if record.taxonomy_id:\n",
    "        ids = line[5:].rstrip().rstrip(\";\")\n",
    "    else:\n",
    "        descr, ids = line[5:].rstrip().rstrip(\";\").split(\"=\")\n",
    "        assert descr == \"NCBI_TaxID\", \"Unexpected taxonomy type %s\" % descr\n",
    "    record.taxonomy_id.extend(ids.split(\", \"))\n",
    "\n",
    "\n",
    "def _read_oh(record, line):\n",
    "    # Line type OH (Organism Host) for viral hosts\n",
    "    assert line[5:].startswith(\"NCBI_TaxID=\"), \"Unexpected %s\" % line\n",
    "    line = line[16:].rstrip()\n",
    "    assert line[-1] == \".\" and line.count(\";\") == 1, line\n",
    "    taxid, name = line[:-1].split(\";\")\n",
    "    record.host_taxonomy_id.append(taxid.strip())\n",
    "    record.host_organism.append(name.strip())\n",
    "\n",
    "\n",
    "def _read_rn(reference, rn):\n",
    "    # This used to be a very simple line with a reference number, e.g.\n",
    "    # RN   [1]\n",
    "    # As of the 2014-10-01 release, there may be an evidence code, e.g.\n",
    "    # RN   [1] {ECO:0000313|EMBL:AEX14553.1}\n",
    "    words = rn.split(None, 1)\n",
    "    number = words[0]\n",
    "    assert number.startswith(\"[\") and number.endswith(\"]\"), (\n",
    "        \"Missing brackets %s\" % number\n",
    "    )\n",
    "    reference.number = int(number[1:-1])\n",
    "    if len(words) > 1:\n",
    "        evidence = words[1]\n",
    "        assert evidence.startswith(\"{\") and evidence.endswith(\"}\"), (\n",
    "            \"Missing braces %s\" % evidence\n",
    "        )\n",
    "        reference.evidence = evidence[1:-1].split(\"|\")\n",
    "\n",
    "\n",
    "def _read_rc(reference, value):\n",
    "    cols = value.split(\";\")\n",
    "    if value[-1] == \";\":\n",
    "        unread = \"\"\n",
    "    else:\n",
    "        cols, unread = cols[:-1], cols[-1]\n",
    "    for col in cols:\n",
    "        if not col:  # last column will be the empty string\n",
    "            return\n",
    "        # The token is everything before the first '=' character.\n",
    "        i = col.find(\"=\")\n",
    "        if i >= 0:\n",
    "            token, text = col[:i], col[i + 1 :]\n",
    "            comment = token.lstrip(), text\n",
    "            reference.comments.append(comment)\n",
    "        else:\n",
    "            comment = reference.comments[-1]\n",
    "            comment = \"%s %s\" % (comment, col)\n",
    "            reference.comments[-1] = comment\n",
    "    return unread\n",
    "\n",
    "\n",
    "def _read_rx(reference, value):\n",
    "    # The basic (older?) RX line is of the form:\n",
    "    # RX   MEDLINE; 85132727.\n",
    "    # but there are variants of this that need to be dealt with (see below)\n",
    "\n",
    "    # CLD1_HUMAN in Release 39 and DADR_DIDMA in Release 33\n",
    "    # have extraneous information in the RX line.  Check for\n",
    "    # this and chop it out of the line.\n",
    "    # (noticed by katel@worldpath.net)\n",
    "    value = value.replace(\" [NCBI, ExPASy, Israel, Japan]\", \"\")\n",
    "\n",
    "    # RX lines can also be used of the form\n",
    "    # RX   PubMed=9603189;\n",
    "    # reported by edvard@farmasi.uit.no\n",
    "    # and these can be more complicated like:\n",
    "    # RX   MEDLINE=95385798; PubMed=7656980;\n",
    "    # RX   PubMed=15060122; DOI=10.1136/jmg 2003.012781;\n",
    "    # We look for these cases first and deal with them\n",
    "    warn = False\n",
    "    if \"=\" in value:\n",
    "        cols = value.split(\"; \")\n",
    "        cols = [x.strip() for x in cols]\n",
    "        cols = [x for x in cols if x]\n",
    "        for col in cols:\n",
    "            x = col.split(\"=\")\n",
    "            if len(x) != 2 or x == (\"DOI\", \"DOI\"):\n",
    "                warn = True\n",
    "                break\n",
    "            assert len(x) == 2, \"I don't understand RX line %s\" % value\n",
    "            reference.references.append((x[0], x[1].rstrip(\";\")))\n",
    "    # otherwise we assume we have the type 'RX   MEDLINE; 85132727.'\n",
    "    else:\n",
    "        cols = value.split(\"; \")\n",
    "        # normally we split into the three parts\n",
    "        if len(cols) != 2:\n",
    "            warn = True\n",
    "        else:\n",
    "            reference.references.append((cols[0].rstrip(\";\"), cols[1].rstrip(\".\")))\n",
    "    if warn:\n",
    "        import warnings\n",
    "        from Bio import BiopythonParserWarning\n",
    "\n",
    "        warnings.warn(\"Possibly corrupt RX line %r\" % value, BiopythonParserWarning)\n",
    "\n",
    "\n",
    "def _read_cc(record, line):\n",
    "    key, value = line[5:8], line[9:].rstrip()\n",
    "    if key == \"-!-\":  # Make a new comment\n",
    "        record.comments.append(value)\n",
    "    elif key == \"   \":  # add to the previous comment\n",
    "        if not record.comments:\n",
    "            # TCMO_STRGA in Release 37 has comment with no topic\n",
    "            record.comments.append(value)\n",
    "        else:\n",
    "            record.comments[-1] += \" \" + value\n",
    "\n",
    "\n",
    "def _read_dr(record, value):\n",
    "    cols = value.rstrip(\".\").split(\"; \")\n",
    "    record.cross_references.append(tuple(cols))\n",
    "\n",
    "\n",
    "def _read_pe(record, value):\n",
    "    pe = value.split(\":\")\n",
    "    record.protein_existence = int(pe[0])\n",
    "\n",
    "\n",
    "def _read_kw(record, value):\n",
    "    # Old style - semi-colon separated, multi-line. e.g. Q13639.txt\n",
    "    # KW   Alternative splicing; Cell membrane; Complete proteome;\n",
    "    # KW   Disulfide bond; Endosome; G-protein coupled receptor; Glycoprotein;\n",
    "    # KW   Lipoprotein; Membrane; Palmitate; Polymorphism; Receptor; Transducer;\n",
    "    # KW   Transmembrane.\n",
    "    #\n",
    "    # New style as of 2014-10-01 release with evidence codes, e.g. H2CNN8.txt\n",
    "    # KW   Monooxygenase {ECO:0000313|EMBL:AEX14553.1};\n",
    "    # KW   Oxidoreductase {ECO:0000313|EMBL:AEX14553.1}.\n",
    "    # For now to match the XML parser, drop the evidence codes.\n",
    "    for value in value.rstrip(\";.\").split(\"; \"):\n",
    "        if value.endswith(\"}\"):\n",
    "            # Discard the evidence code\n",
    "            value = value.rsplit(\"{\", 1)[0]\n",
    "        record.keywords.append(value.strip())\n",
    "\n",
    "\n",
    "def _read_ft(record, line):\n",
    "    name = line[5:13].rstrip()\n",
    "    if name:\n",
    "        if line[13:21] == \"        \":  # new-style FT line\n",
    "            location = line[21:80].rstrip()\n",
    "            try:\n",
    "                isoform_id, location = location.split(\":\")\n",
    "            except ValueError:\n",
    "                isoform_id = None\n",
    "            try:\n",
    "                from_res, to_res = location.split(\"..\")\n",
    "            except ValueError:\n",
    "                from_res = location\n",
    "                to_res = \"\"\n",
    "            qualifiers = {}\n",
    "        else:  # old-style FT line\n",
    "            from_res = line[14:20].lstrip()\n",
    "            to_res = line[21:27].lstrip()\n",
    "            isoform_id = None\n",
    "            description = line[34:75].rstrip()\n",
    "            qualifiers = {\"description\": description}\n",
    "        if from_res == \"?\":\n",
    "            from_res = UnknownPosition()\n",
    "        elif from_res.startswith(\"?\"):\n",
    "            position = int(from_res[1:]) - 1  # Python zero-based counting\n",
    "            from_res = UncertainPosition(position)\n",
    "        elif from_res.startswith(\"<\"):\n",
    "            position = int(from_res[1:]) - 1  # Python zero-based counting\n",
    "            from_res = BeforePosition(position)\n",
    "        else:\n",
    "            position = int(from_res) - 1  # Python zero-based counting\n",
    "            from_res = ExactPosition(position)\n",
    "        if to_res == \"\":\n",
    "            position = from_res + 1\n",
    "            to_res = ExactPosition(position)\n",
    "        elif to_res == \"?\":\n",
    "            to_res = UnknownPosition()\n",
    "        elif to_res.startswith(\"?\"):\n",
    "            position = int(to_res[1:])\n",
    "            to_res = UncertainPosition(position)\n",
    "        elif to_res.startswith(\">\"):\n",
    "            position = int(to_res[1:])\n",
    "            to_res = AfterPosition(position)\n",
    "        else:\n",
    "            position = int(to_res)\n",
    "            to_res = ExactPosition(position)\n",
    "        location = FeatureLocation(from_res, to_res, ref=isoform_id)\n",
    "        feature = FeatureTable(\n",
    "            location=location, type=name, id=None, qualifiers=qualifiers\n",
    "        )\n",
    "        record.features.append(feature)\n",
    "        return\n",
    "    # this line is a continuation of the previous feature\n",
    "    feature = record.features[-1]\n",
    "    if line[5:34] == \"                             \":  # old-style FT line\n",
    "        description = line[34:75].rstrip()\n",
    "        if description.startswith(\"/FTId=\"):\n",
    "            # store the FTId as the feature ID\n",
    "            feature.id = description[6:].rstrip(\".\")\n",
    "            return\n",
    "        # this line is a continuation of the description of the previous feature\n",
    "        old_description = feature.qualifiers[\"description\"]\n",
    "        if old_description.endswith(\"-\"):\n",
    "            description = \"%s%s\" % (old_description, description)\n",
    "        else:\n",
    "            description = \"%s %s\" % (old_description, description)\n",
    "\n",
    "        if feature.type in (\"VARSPLIC\", \"VAR_SEQ\"):  # special case\n",
    "            # Remove unwanted spaces in sequences.\n",
    "            # During line carryover, the sequences in VARSPLIC/VAR_SEQ can get\n",
    "            # mangled with unwanted spaces like:\n",
    "            # 'DISSTKLQALPSHGLESIQT -> PCRATGWSPFRRSSPC LPTH'\n",
    "            # We want to check for this case and correct it as it happens.\n",
    "            try:\n",
    "                first_seq, second_seq = description.split(\" -> \")\n",
    "            except ValueError:\n",
    "                pass\n",
    "            else:\n",
    "                extra_info = \"\"\n",
    "                # we might have more information at the end of the\n",
    "                # second sequence, which should be in parenthesis\n",
    "                extra_info_pos = second_seq.find(\" (\")\n",
    "                if extra_info_pos != -1:\n",
    "                    extra_info = second_seq[extra_info_pos:]\n",
    "                    second_seq = second_seq[:extra_info_pos]\n",
    "                # now clean spaces out of the first and second string\n",
    "                first_seq = first_seq.replace(\" \", \"\")\n",
    "                second_seq = second_seq.replace(\" \", \"\")\n",
    "                # reassemble the description\n",
    "                description = first_seq + \" -> \" + second_seq + extra_info\n",
    "        feature.qualifiers[\"description\"] = description\n",
    "    else:  # new-style FT line\n",
    "        value = line[21:].rstrip()\n",
    "        if value.startswith(\"/id=\"):\n",
    "            qualifier_type = \"id\"\n",
    "            value = value[4:]\n",
    "            assert value.startswith('\"')\n",
    "            assert value.endswith('\"')\n",
    "            feature.id = value[1:-1]\n",
    "            return\n",
    "        elif value.startswith(\"/evidence=\"):\n",
    "            value = value[10:]\n",
    "            assert value.startswith('\"')\n",
    "            if value.endswith('\"'):\n",
    "                value = value[1:-1]\n",
    "            else:  # continues on the next line\n",
    "                value = value[1:]\n",
    "            assert \"evidence\" not in feature.qualifiers\n",
    "            feature.qualifiers[\"evidence\"] = value\n",
    "            return\n",
    "        elif value.startswith(\"/note=\"):\n",
    "            value = value[6:]\n",
    "            assert value.startswith('\"')\n",
    "            if value.endswith('\"'):\n",
    "                value = value[1:-1]\n",
    "            else:  # continues on the next line\n",
    "                value = value[1:]\n",
    "            assert \"note\" not in feature.qualifiers\n",
    "            feature.qualifiers[\"note\"] = value\n",
    "            return\n",
    "        # this line is a continuation of the description of the previous feature\n",
    "        keys = list(feature.qualifiers.keys())\n",
    "        key = keys[-1]\n",
    "        description = value.rstrip('\"')\n",
    "        old_description = feature.qualifiers[key]\n",
    "        if key == \"evidence\" or old_description.endswith(\"-\"):\n",
    "            description = \"%s%s\" % (old_description, description)\n",
    "        else:\n",
    "            description = \"%s %s\" % (old_description, description)\n",
    "        if feature.type == \"VAR_SEQ\":  # see VARSPLIC above\n",
    "            try:\n",
    "                first_seq, second_seq = description.split(\" -> \")\n",
    "            except ValueError:\n",
    "                pass\n",
    "            else:\n",
    "                extra_info = \"\"\n",
    "                # we might have more information at the end of the\n",
    "                # second sequence, which should be in parenthesis\n",
    "                extra_info_pos = second_seq.find(\" (\")\n",
    "                if extra_info_pos != -1:\n",
    "                    extra_info = second_seq[extra_info_pos:]\n",
    "                    second_seq = second_seq[:extra_info_pos]\n",
    "                # now clean spaces out of the first and second string\n",
    "                first_seq = first_seq.replace(\" \", \"\")\n",
    "                second_seq = second_seq.replace(\" \", \"\")\n",
    "                # reassemble the description\n",
    "                description = first_seq + \" -> \" + second_seq + extra_info\n",
    "        feature.qualifiers[key] = description\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from urllib.request import urlopen\n",
    "from urllib.parse import urlencode\n",
    "\n",
    "def scan(seq=\"\", mirror=\"https://prosite.expasy.org\", output=\"xml\", **keywords):\n",
    "    \n",
    "    #Borré la descripción de la función solo para que la visualización de los cambios fuera más sencilla\n",
    "    parameters = {\"seq\": seq, \"output\": output}\n",
    "    \n",
    "    ##Cambio para solo usar sig\n",
    "    if seq is None:\n",
    "        parameters = {\"output\": output}\n",
    "    ##\n",
    "    \n",
    "    for key, value in keywords.items():\n",
    "        if value is not None:\n",
    "            parameters[key] = value\n",
    "\n",
    "        \n",
    "    command = urlencode(parameters)\n",
    "    url = \"%s/cgi-bin/prosite/PSScan.cgi?%s\" % (mirror, command)\n",
    "    handle = urlopen(url)\n",
    "    return handle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ejercicios\n",
    "+ <a href=\"#Ejercicio_1\" style=\"color:#058B8C\">Ejercicio 1</a>\n",
    "+ <a href=\"#Ejercicio_2\" style=\"color:#058B8C\">Ejercicio 2</a>\n",
    "+ <a href=\"#Ejercicio_3\" style=\"color:#058B8C\">Ejercicio 3</a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "usar solo read() en vez de SwissProt.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_1\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 1</h2>\n",
    "<p>Busca los patrones de Prosite PS00079 y PS00080  en la Ferroxidasa humana HEPHL1 e imprme la o las partes de la secuencia que entra en el patron y el rango (donde inica y termina el patrón en la secuenca)</p>\n",
    "\n",
    "\n",
    "<h3 id=\"sample-output\">Resultado esperado (datos al azar, no reales)</h3>\n",
    "<div class=\"codehilite\"><pre>Prosite # Secuencia Rango\n",
    "PS00079 1 ACESYCRDHK 54-63\n",
    "PS00079 2 ACASYCRDHH 32-41\n",
    "PS00080 1 WNYLPYVVQF 102-101\n",
    "\n",
    "</pre></div>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import ExPASy\n",
    "from Bio.ExPASy import Prosite\n",
    "from Bio.ExPASy import ScanProsite\n",
    "#Se obtiene la secuencia del humano HPHL1_HUMAN\n",
    "ExPAS_raw = ExPASy.get_sprot_raw('HPHL1_HUMAN')\n",
    "ExPAS_raw_record = read(ExPAS_raw)\n",
    "sequence= ExPAS_raw_record.sequence\n",
    "#Se hace una lista con los accession numbers\n",
    "PROSITE_accession = [\"PS00079\",\"PS00080\"] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Prosite 1 PS00079 Rango: 341 - 361\n",
      "KWMITCQVSDHLQAGMLGQY\n",
      "Prosite 2 PS00079 Rango: 694 - 714\n",
      "IFRVFCATMPHLSRGMGQIY\n",
      "Prosite 3 PS00079 Rango: 1043 - 1063\n",
      "TWLLHCHVSDHIHAGMETTY\n",
      "Prosite 1 PS00080 Rango: 1048 - 1059\n",
      "CHVSDHIHAGM\n"
     ]
    }
   ],
   "source": [
    "for  accession in PROSITE_accession:\n",
    "    \n",
    "    \n",
    "    handle = ExPASy.get_prosite_raw(accession)\n",
    "    record = Prosite.read(handle)\n",
    "    #El punto de al final que arroja record.pattern causa ruido.Entonces lo quitamos\n",
    "    pattern = record.pattern[:len(record.pattern)-1]\n",
    "    #Se usa la funcion scan y se le da el patron a buscar en la secuencia que teniamos.\n",
    "    handle = scan(seq=sequence, sig=pattern) \n",
    "    #Se imprimen los resultados\n",
    "    results = ScanProsite.read(handle)\n",
    "    for r in range(len(results)):\n",
    "    \n",
    "        print(\"Prosite\",r+1,accession,\"Rango:\",results[r][\"start\"] ,\"-\",results[r][\"stop\"])\n",
    "        print(sequence[results[r][\"start\"]:results[r][\"stop\"]])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_2\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 2</h2>\n",
    "<p>Obtén el nombre de las enzimas de una lista que contenga accesion numbers de SwissProt (los que quieras) utilizando expresiones regulares</p>\n",
    "\n",
    "\n",
    "<h3 id=\"sample-output\">Resultado esperado</h3>\n",
    "<div class=\"codehilite\"><pre>O23729    2.3.1.74\n",
    "O46647    3.1.1.34\n",
    "</pre></div>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "accession= O23729 EC=2.3.1.74\n",
      "accession= O23731 EC=2.3.1.74\n",
      "accession= O23732 EC=6.3.2.3\n",
      "accession= O23733 EC=2.5.1.47\n",
      "accession= O23734 EC=2.5.1.47\n",
      "accession= O23735 EC=2.5.1.47\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "accession_no = [\"O23729\",\"O23731\",\"O23732\",\"O23733\",\"O23734\",\"O23735\"]\n",
    "for a in accession_no:\n",
    "    \n",
    "    handle = ExPASy.get_sprot_raw(a)\n",
    "    record = read(handle)\n",
    "    #Se busca un match que empiece con \"EC=\" y después que tenga  un punto o un numero,de 2 a 15 veces.\n",
    "    match=re.search(\"EC=[\\d.]{2,15}\",record.description)\n",
    "    print(\"accession=\",a,match.group())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"Ejercicio_3\"></a>\n",
    "\n",
    "<div class=\"alert alert-block alert-info\">    \n",
    "<h2 id=\"problem\">Ejercicio 3</h2>\n",
    "<p>Escojan un organismo de los presentes en Cellosaurus y después escojan una categoría para graficar los porcentajes.</p>\n",
    "<p>Por ejemplo, en la categoría de enfermedades cuáles son las más representadas o en sexo cuáles son los porcentajes (x% mixed sex, y% female, z% male)</p>\n",
    "\n",
    "\n",
    "\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex_male=0\n",
    "sex_female=0\n",
    "sex_mixed=0\n",
    "total_count = 0\n",
    "\n",
    "from Bio.ExPASy import cellosaurus \n",
    "with open(\"cellosaurus.txt\") as handle:\n",
    "    records = cellosaurus.parse(handle)\n",
    "    for record in records:\n",
    "        #Se revisa celulas especificas de humano\n",
    "        if \"Homo sapiens\" in record[\"OX\"][0]:\n",
    "            #Se revisa si el sexo es masculino\n",
    "            if \"Male\" in record[\"SX\"]:\n",
    "                sex_male+=1\n",
    "            #Se revisa si el sexo es femenino\n",
    "            if \"Female\" in record[\"SX\"]:\n",
    "                sex_female+=1\n",
    "            total_count+=1\n",
    "           "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "93535"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "sex_mixed=total_count-sex_male-sex_female"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "45977\n",
      "39293\n",
      "8265\n"
     ]
    }
   ],
   "source": [
    "print(sex_male)\n",
    "print(sex_female)\n",
    "print(sex_mixed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAkMAAAGRCAYAAABmJ7ciAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAIABJREFUeJzt3Xec3HWB//HXZ/smIZNCExAHAT0RKdJVBBXrir3g4d3asJx6eOrPG8vPG9u5p/7UU89TwAqKFU9kBDyVfnQpA5w03ZBCKCmTbG/f3x/fCZuETd/dz8x8X8/HYx7ZMrvz3hjJO58akiRBkiQpq5piB5AkSYrJMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiRJkjLNMiSppoUQkhDCeRu93xJCeCSEcNE2vu7kbT1HksAyJKn29QOHhhA6q++/EFgeMY+kBmMZklQPfgt0Vd9+E3D+hk+EEI4NIVwbQrglhPA/IYSnbv7FIYS5IYTvhhBuqD7vlbOUW1IdsAxJqgc/AU4LIXQAhwHXb/S5PwMnJklyJPBJ4F+n+PqPA39MkuRY4HnAF0MIc2c4s6Q60RI7gCRtS5Ikt4cQ8qSjQr/d7NM54AchhIOBBGid4lu8CHhFCOHD1fc7gP2B/52RwJLqimVIUr24EPgScDKweKOPfwa4LEmSV1cL0+VTfG0AXpskyd0zG1FSPXKaTFK9+C7wqSRJypt9PMfkguq3bOFrLwXeH0IIACGEI2ckoaS6ZBmSVBeSJFmWJMnXpvjUF4DPhxBuYcuj3Z8hnT67PYRwZ/V9SQIgJEkSO4MkSVI0jgxJkqRMswxJkqRMswxJkqRMswxJkqRMswxJkqRMswxJkqRMswxJkqRMswxJkqRMswxJkqRM86JWSTMqXyh1ArsB84C5QDMwUX0kG7091WMEGAKGgeHeni6PzJc07byOQ9Lj5AulxcBeTJaYDb9u6e0tfWxD+Zkum5Sjjd4eAFYBD2/0eGSz9x/u7ekamsYskhqEZUjKmOpIzf7AE7fy65xoAWdWH1spS5t97JHenq6xSDklzSLLkNRA8oVSM7APWy46+wOLowWsLwmwArh7s8efgQd6e7omImaTNI0sQ1IdyhdKC4HDN3o8hbTo7MP0TktpakPAvTy+KN3d29NViRlM0o6zDEk1LF8oNQEHsWnxOZx0pEe1aSVTlCTgr709XeMxg0mammVIqhH5QmkecBiThecI4FDSRciqfyPA/cAtwLXA/wC3uy5Jis8yJEWQL5TyPH6058lAiBhLs28AuJG0GF0LXNvb0/Vo3EhS9liGpBmWL5QCadl5PvA84DnAgqihVMvuY3Lk6FrgDqfXpJllGZJmQL5QOoS0+DwfOAl3cGnn9QE3MFmOruvt6VodN5LUWCxD0jTIF0oHMjny8zxg77iJ1MAS4B4my9GVvT1dd8eNJNU3y5C0E/KF0n6k5WdDAdo/biJl3P3ARcBvSMvRaOQ8Ul2xDEnbIV8o7cnktNfzSbe7S7VoHfA70mL0WxdkS9tmGZK2IF8oPRN4HXAq6RZ3qd5MANdRHTXq7em6I3IeqSZZhqSN5Aul44DXVh9PjhxHmm69QIl01Ojy3p6u4bhxpNpgGVKmVbe9P5u0/LwG1/4oO/qA35MWo1JvT9dDkfNI0ViGlDnVKy6eSzoF9mrS+7ykLEtID3/8DfCz3p6ueyLnkWaVZUiZkC+UWkgXQL8OeBWwZ9xEUk27FvgB8BMvnlUWWIbU0PKF0jOAtwKnYwGSdtQQ8Gvg+8B/exK2GpVlSA0nXygtJi0/bwGOjJtGahgrgPOAH/T2dN0VO4w0nSxDagjVabCXko4CdQFtcRNJDe1G0mm0870aRI3AMqS6li+UDgbeBbwZ2CtyHClrRkgXXX8fuNhpNNUry5DqTnU3WBfwXuBFQIibSBKwFPgWcHZvT9cjscNIO8IypLpRXQv0duA9QD5uGklbMAz8FPh6b0/XTbHDSNvDMqSaly+UjgLeB5wGdESOI2n7XQ98Hfh5b0/XSOww0pZYhlSTqlNhrwU+CBwfOY6kXfMQcBbwzd6erpWxw0ibswyppuQLpXagG/gwcHDkOJKm1zDpYusv9vZ03R85i/QYy5BqQr5QypGuBToT2DtyHEkzaxz4OdDT29N1W+wwkmVIUeULpScAHwDeDcyPHEfS7LsY+HxvT9dVsYMouyxDiiJfKD0J+BjplFh75DiS4ruGdKToothBlD2WIc2q6kjQJ4B34CnRkh7vZuATvT1dl8QOouywDGlW5Aul3YEC8A9AZ+Q4kmrfVcDHnT7TbLAMaUblC6UFpDvDzgTmRY4jqf5cQlqK/hQ7iBqXZUgzIl8ozSMtQB8GFkSOI6m+JcAFwP/t7en639hh1HgsQ5pW1cMS3wp8FrfIS5peE8C5wMd6e7pWxA6jxmEZ0rTJF0onAV8BjoydRVJD6wc+D/y/3p6uodhhVP8sQ9pl+ULpAOBLwGtiZ5GUKb3AR3p7un4eO4jqm2VIOy1fKO1Guk3+TDwrSFI8VwJn9vZ03Ro7iOqTZUg7LF8oBeBtwOeAvSLHkSRI1xN9h/SMoodjh1F9sQxph+QLpcOAbwEnxM4iSVOoAB8FvtXb0+VfcNouliFtl3yhNBcokt4j1hI3jSRt0zXAO3t7uu6KHUS1zzKkbcoXSi8D/hPYP3YWSdoBI0AP8K+9PV3DscOodlmGtEX5QmkP4GvAabGzSNIu+DPpKJFXe2hKliFNKV8o/R3pmUGLY2eRpGmQAGeTbsWvxA6j2mIZ0iaqo0FnAa+KnUWSZsCDwBm9PV2l2EFUOyxDeky+UHo5cA5ul5fU+M4GPtjb09UXO4jiswxpw6WqXwbOiJ1FkmbRX4C/7+3puiZ2EMVlGcq4fKH0LOCHwIGxs0hSBOOk95x9qrenayx2GMVhGcqofKHUCnwK+AjQHDmOJMV2PXB6b0/X/bGDaPZZhjIoXyjtD/wcODZ2FkmqIX3AP/T2dJ0bO4hml2UoY/KF0kuB84BFsbNIUo06C/hHD2rMDstQRuQLpSbg08DHgBA5jiTVupuA1/X2dC2JHUQzzzKUAflCaU/gx8ALYmeRpDqyGnhzb0/XxbGDaGZZhhpcvlB6DvBTYJ/YWSSpDiXA54B/6e3pmogdRjPDMtTA8oXSB4Av4i3zkrSr/hv4296erkdjB9H0sww1oOq2+W8C74idRZIayBLg5b09XXfEDqLpZRlqMPlCaSHwS+B5sbNIUgNaB7yht6fr0thBNH2aYgfQ9MkXSgcD12ERkqSZMh8o5Qul98QOounjyFCDyBdKJ5OOCHl+kGpaMjHOgz/4J1p2W8yer/sXBpfcxtrLvksyPkrb3gex+KVnEpqa6b/7GipX/Yimznns8ZpP0Nw5n9E1D7L2yh+yxyv/OfaPIQH8O+llry6srnOODDWAfKH0duB3WIRUB9bfdCGti58IQJJMsKr0FXZ/xUfY5+3fpGX+nvSV/5A+7+bfsHf3l5l3xEvpv+sKANZedS4LTnxztOzSZs4E/itfKM2NHUS7xjJU5/KF0meBc4DW2FmkbRlb9yiDf7mReYe/CICJwfWE5hZaF+0LQEf+CAbuqV4gHppIxsdIRocJTc0MLb2D5rkLH3uuVCNOBa7KF0r+waxjlqE6lS+UmvKF0jeBj8fOIm2vNX84iwUnv40Q0kPQmzrnk0yMM/zgvQAM3H0N4+vSncu541/Pwz/5OIP3Xc/cQ06i8j8/Jfes06Jll7biSOCafKF0UOwg2jmeP1OH8oVSC/BD4E2xs0jba+C+G2iau4D2vQ9i6IHbAQghsMcrPsKaP55NMj5KR/6Z0JT+G63zgCPpPOBIAPru+AOdTz6asdXLWX3DBTR1zGPhKe+kqbUj2s8jbeZJpCNEL3Trff1xAXWdyRdKnaQ3znfFziLtiDVXfJ/+Oy6DpmaS8RGS4UHmPOUEdj/1w489Z/Cvf6Lvtt+xx6sKj31sYnSIh3/xafZ6w6d5+BefYo9Xf5yBu68hGR9ltyNeEuNHkbZmNfCS3p6uG2MH0fZzZKiO5Aul+cBvgOfGziLtqIUnvYWFJ70FgKEHbmfdDb9i91M/zHj/WprnLiAZG2Xd9b9g/glv3OTr1l1/AfOPOpXQ3EIyNpJeMxwCyZgXiqsmLQL+kC+UTu3t6boidhhtH9cM1Yl8obQ7cBkWITWYdTdcwPKz382K772PzgOPo/NJhz/2ubH1qxh58B7mPOUEAHY76lRW/uCD9N16MXMPOTlSYmmbdgMuyRdKjuDXCafJ6kC+UNoDuBw4JHIUSdL2GwX+rren66exg2jrLEM1Ll8oLSIdETosdhZJ0g4bJ72+44LYQbRllqEali+UcsDvgaNjZ5Ek7bQR4FW9PV0Xxw6iqVmGalS+UJpHeqr0CbGzSJJ22RDwst6erstiB9HjuYC6BlW3z1+ERUiSGkUHcGG+UPK/6zXIMlRj8oVSO/Br4KTYWSRJ02oecHG+UHpm7CDalGWohuQLpSbgp8ALY2eRJM2IHPC7fKH09NhBNMkyVFv+HXhl7BCSpBm1mPQcon1iB1HKMlQj8oXSPwHvi51DkjQr9gMuyhdKc2MHkbvJakK+UHoN6X1jllNJypaLgFf29nRNxA6SZf7lG1m+UDoOOA//t5CkLHo58NXYIbLOkaGI8oXSk4HrgD1iZ5EkRXVmb0/X12KHyCrLUCT5QmkBaRF6auwskqToJkhPqf5N7CBZ5NRMBPlCKQA/xCIkSUo1Aee75T4Oy1AcHwVOjR1CklRT5gK/qF7HpFlkGZpl+ULpFOAzsXNIkmrS3wDnxA6RNZahWZQvlJ4InI+/75KkLXtjvlB6f+wQWeIC6lmSL5TagKuAY2NnkSTVvBHgub09XdfHDpIFjlDMnq9iEZIkbZ824Of5Qmlx7CBZYBmaBdUTpt8TO4ckqa48EfhRdQeyZpBlaIblC6W9gG/HziFJqksvxnsrZ5xlaOadDeweO4QkqW715Aulg2OHaGSWoRmUL5TejucJSZJ2zRzgB/lCqTl2kEZlGZoh+ULpAOArsXNIkhrCCcD/iR2iUbm1fgbkC6Um4DLgubGzSJIaxghwVG9P1x2xgzQaR4ZmxgewCEmSplcb8MN8odQaO0ijsQxNs+op05+OnUOS1JCOBAqxQzQay9D0+yrpZXuSJM2Ej+YLpXzsEI3EMjSN8oXSS4DXxM4hSWponbhBZ1q5gHqa5AulduAO4KDYWSRJmfDS3p6uS2KHaASODE2fj2ARkiTNnq9VLwHXLrIMTYPqmUIfjZ1DkpQpBwMfih2iEViGpseXSedwJUmaTR+v7mLWLrAM7aJ8oXQ88KrYOSRJmTQX+FzsEPXOMrTr/jV2AElSpp2eL5SeFjtEPbMM7YJ8oXQK8LzYOSRJmdaEh/3uEsvQrnFUSJJUC16bL5SOiB2iXlmGdlK+UHo1cEzsHJIkAQH4TOwQ9cpDF3dC9Vb624Gnx84iSdJGju/t6bo+doh648jQznk9FiFJUu35bOwA9cgytHM85EqSVItOyRdKLuHYQZahHZQvlE7EtUKSpNr1gdgB6o1laMd9MHYASZK24vX5Qmmf2CHqiWVoB+QLpQOBV8TOIUnSVrQC740dop5YhnbMB/D3TJJU+96VL5S8M3M7+Rf7dsoXSguAt8bOIUnSdlgM/F3sEPXCMrT93kZ6IZ4kSfXgzNgB6oVlaPu9JXYASZJ2wCH5Quk5sUPUA8vQdqje9/KM2DkkSdpBTpVtB8vQ9umOHUCSpJ3whnyh1B47RK2zDG1DvlBqAf42dg5JknbCAqArdohaZxnatpcAe8YOIUnSTnKqbBssQ9vmFJkkqZ69LF8oLYodopZZhrYiXyjtBpwaO4ckSbugDXhj7BC1zDK0dS8CXHgmSap3r48doJZZhrbu5bEDSJI0DZ5Tne3QFCxDW5AvlALwstg5JEmaBq3AC2OHqFWWoS07FneRSZIah//A3wLL0JZ5LoMkqZG8JHaAWmUZ2jLXC0mSGsm++ULp8NghapFlaAr5QukJwJGxc0iSNM2cKpuCZWhqz44dQJKkGeAi6ilYhqZ2QuwAkiTNgGPyhZJ/92/G35CpWYYkSY1oHvD02CFqjWVoM/lCqQ14ZuwckiTNkONiB6g1lqHHeyZewSFJalzHxw5QayxDj+cUmSSpkTkytBnL0ONZhiRJjewQ7ynblGXo8Q6LHUCSpBnUBBwVO0QtsQxtJF8oNQNPjp1DkqQZ9tTYAWqJZWhTB5De7CtJUiM7OHaAWmIZ2pRNWZKUBZahjViGNvWU2AEkSZoFB8UOUEssQ5uyDEmSsuBAr+WY5G/EpixDkqQsaAeeGDtErbAMbcqdZJKkrHCqrMoytKm9YweQJGmW7Bs7QK2wDFXlC6X5QEfsHJIkzZJFsQPUCstQVRujuzczPhY7hyRJs8QyVNUSO0CtuKejey+gJUkYBQYnCMMTNA2P0zQ8TvPIKM2jo7SMDdM6Opy0jg/RNjFA+8Rg0p4M0JH00xH6kk766GjqSzqb+uhs7ks6m/vpaOmns60v6WgdoKNtIGlvH6C9Y5D2jiHaOiCE2D+7JCmTFscOUCssQ5MWAoRAK9DaTEIz47QyDoxu+sxpqi9JQgIMJTCY0DQ0ThgZp3lkjOaRUVpGR2gZG05ax4ZoGx+ibWKQ9qQ/6ZgYoJ3+pIM+Opv66Az9SWdzHx0t65M5aflKOlr76Wjvp6NtcLJ8dY7R4unakqQNHBmqsgxNWjjbLxgCAegI0AETNAMwvtmTpu/1koRxYDAhDE0QhsZpGtlo1Gt0mNax6qjX+GB11KufdvrpTPqTjtBHZ1Nf0hn66GjuT0e+WvvobOlPOtoGaG/rp7N9IGlrH6K9Y5C2zoQmp2ElqXZZhqosQ5PmxQ4w00KgGZgXSOY1kdDCBLDZMqnpLV/DwNAEYWjDlONYtXiN0DI6nLSODdM6PkR7MkD7+EA65Rj6ko6kn87Qn045Nq+ns6k/6Wzpo6Oln462/qSztTrq1dFPe8cQ7R0jtLZPX3JJygTLUJVlaJJTSNMsBNqB9maS3JRTjtNbvCaAoYTw2HqvscemHJvHRmgdG0pax4ZpGx+krTrq1ZEM0EFfOuoV+tO1Xk19dLT0JXPS8pV0tPXT2TZAe9tA0t4+SHvnIO2dEzQ1T196SYoiFztArbAMTbIM1bEQaALmBJI5TSTABG0zO+q18UL7oXGaRjaUrxFaNpSv8SHaxwdpSwaSjsfK14Ypx/VJZ1M/HdWF9p2tfXS0bljvNZBOPXYM0eZCe0kzxQ5Q5W/EJH8vtN0iLLQfTGAoSacbN5SvjaccN1rr1TaxYYdjf9JBP52hL93t2Fzd5djST0drX3W6sTrq1TFAR/sQbS60l7LDdZ1VFoBJ/gWgmlRdaN8ZoDNdaD8xxZOm7/WShDEmF9oPV4vXhvVeY+lC+7axIVonBmmfGEg6kgHak/7JUa9QPV6ipa+607E/XWjf3k9H6wAdGxbadw7Q3umolxSN0/1VlqFJ/l5IQAi0ALsFkt02LLRvn8Epx9GEvvNy8285a0Fu/vqm4GXJ0uwZgq7YGWqCBWCSw4VSBK2BeW9dt+7Et65bx7UdHXf0LF5Y+Utry7GE4GitNLPsAFUWgEkDsQNIWXfC0NChv17+4LP/sHTFmpf09V/RlCQrY2eSGphXUFVZhib1xw4gKbXn+PieX3xk1Uk39y7d/Z9Xrb52/vj4bbEzSQ1odNtPyQbL0CTLkFRjWqDlzev6TrjmgeWH/2jFyrufPjx8FUniKK40PRwZqrIMTeqLHUDSlh02PPLUn6x46MSrH1g+8tp1fVe0JMnS2JmkOjcYO0CtsAxNcmRIqgO5iYkFxVWrT7q5d+m+n3lk1Q27j43fTJIksXNJdejB2AFqhWVokiNDUh1pgqZX9fUfe9nS5Uf9avnK3qMHh64kSdbFziXVkRWxA9QKy9Ckh2IHkLRzDhodPeB7Kx9+7nVLljV1V9Zd2TaR3B87k1QHLENVlqFJ/qGQ6tzcJJn34dVrn3vzkqUHfuWhR27ZZ3TsepJkPHYuqUb5916VZWiDYqUPcIhdahCnDAweeemyFcddvGzFyucODF4RkmRV7ExSjVkeO0CtsAxtyj8YUoPZb2x83/946JGTblyydO5716y9Zs7ExP/GziTVCEeGqixDm7IMSQ2qPaHj3WvXPfv6JcuedvaDD93x5JHR/yFJRmLnkiKyDFVZhjZlGZIy4Pih4UN/vfzBZ/1x6fLKS9NrP9xirKwZLneXnTqusgxt6r7YASTNnj3GJ/b4Qnrtxx7/vGr1tbnx8dtjZ5Jmif8A2IhlaFPl2AEkzb4N135c/cDyw7z2QxnhTMhGLEObsgxJGbfxtR+vX7f+ipYkeSB2JmkG3BU7QC2xDG3qr3gStSTSaz8+uWrNSTf3Lt3vM4+sutFrP9RgbowdoJZYhjZWrCTAnbFjSKod1Ws/jrls6fKj/mv5g0uOGRy6wms/1ABuih2glliGHs+pMklTOnB0LP/dlQ+fdN2SZU1vWbvuqvaJCTddqB4NAXfEDlFLLEOPZxmStFVzk2Teh9asPfGmJcsO+spDj9y6r9d+qL7cVu4uj8YOUUtaYgeoQZYhSdvtlIHBI04ZGGRZS/Pyzy9edN9VnR2HJiEsjp2rFj36u0dZc8UaSGDhSQvZ/cW7s/JnK1l/+3o69+9kv3fuB8Da/1nL2Poxdn/x7pETNyynyDbjyNDjWYYk7bAN137c1Lt03vtWr71mzsSEu3U2MrRsiDVXrOHATx7IQZ85iPW3rWfwgUEGlwxy8GcPJrQEhpYOMTEywZqr1rD4BfbJGeTi6c1YhjZXrDwKuJVW0k5pg/Z3VdY9+/olyw45+8GH7jxwZMRrP4DhFcN0PrmTpvYmQnNg7lPnsv5P60nGEpIkYWJkgtAcePTiR1l8ymJCS4gduZE5MrQZy9DULokdQFL9O35o+On/tXyl134A7fu1M3DPAGN9Y0wMT7D+9vWMrR9jt8N34/5P3k9LroWmOU0M/mWQ+UfNjx23kfUDXla8meCxGVMo5l4J/FfsGJIayxiM/XT+vJv+c0Gus9LcfHjsPLNt9RWrWf3H1TS1N9GxbwehJfCE05/w2OeXf3c5i56/iMElg/Td0UfHEzvY8xV7RkzckK4qd5efGztErXFkaGq/BzI/rC1perVAy+nr+o6/+oHlh/94+cp7Ds3YtR+LTlrEQZ86iCd/7Mk0zW2ibe+2xz43uGSQJElof0I7625cx/7v3Z+Rh0cYXjkcMXFDuiZ2gFpkGZpKsdIPXBk7hqTG9YyRkaecv+KhE69+YNloVq79GFs3BsDIqhHW3bSOBccveOxzD1/wMHu9Zq90DdFEdcYiwMTIRIyojezC2AFqkVvrt+y3wCmxQ0hqbLmJJPfJVWtO+sSqNRMXzZt741cX5sIjzc1HEULDrSB+4BsPMN43TmgO7PP3+9A8txmAdTevoyPfQevCVgA69u/g3k/cS8d+HXTu3xkzcqN5CLg+doha5JqhLSnmngr8OXYMSdlzf2tL7+cWL3rgxo72wwkhFzuPGsY55e7yGbFD1CKnybakWLkbuD92DEnZU73247nXLVnW7LUfmka/jh2gVlmGtu7i2AEkZdfG13589aFHbt1vdOw6r/3QTuon3RykKViGtq4UO4AkAbxgYPCIi5etOP6SZSseOmlg4IqQJI/GzqS6cmm5uzwUO0StcgH11l0ODABzIueQJAD2HRvf5xsPPbrPCAx/Lzf/mu8umL9woKnpkNi5VPOcItsKF1BvSzF3EdAVO4Ykbcn1He139ixeuPa+1tZjCKFt21+hjBkD9ip3l1fHDlKrnCbbtt/EDiBJW3Pc0PDTf7V85bMvW7q88rKMX/uhKV1tEdo6y9C2nU+68EySatru4xN7/Nsjq066uXfpHh99dPW1ufHx22JnUk3weqltcJpsexRz3wLeFTuGJO2oO9ra7v3c7gtX3tHWdhQhuP4xe4aB/crdZRfcb4VlaHsUc4cB/gtLUt2qNIXK1xYuuPWC3eblx0J4Uuw8mjU/KXeX3xQ7RK2zDG2vYu4q4DmxY0jSrpiAiYvmzb25ka/90CaeX+4uXxY7RK2zDG2vYu400vVDktQQ/tLasuRzixf13tDRfoTXfjSke8vd5afEDlEPXEC9/X5JesmdJDWEJ4+OPek7Kx8+6foly1rell77cW/sTJpWZ8UOUC8sQ9urWBkFzo4dQ5Km25wkmftP6bUfB6fXfox67Uf9GwC+M1PfPISQhBDO2+j9lhDCIyGEi6rvvyKEUJim1+qbju+zNZahHfNtwP9ASGpY6bUfDx5/ybIVD53cP3C5137UrR+Wu8trZvD79wOHhhA6q++/EFi+4ZNJklyYJEnPDL7+tLIM7YhiZRlwYewYkjTT9h0b3+frDz968k29S3f7x9Vrr5k7MXFX7EzabgnwtVl4nd8yeUPDm9hoXW0I4S0hhG9U3/51COHvq2+/K4Two+rbB4YQLgkh3BxCuCqE8DfVjx8QQrg2hFAOIXx2Fn4Oy9BO+I/YASRptrRB+xmVdc++bsmyQ77z4EN3HjQycg1JMhI7l7bqd+Xu8v/Owuv8BDgthNABHAZcv4XnvRP4ZAjhROBDwPurHz8LeH+SJEcBHwa+Wf34vwP/mSTJM4BZOU3dMrSjipU/AH+OHUOSZtuxG1370dXXf7nXftSsr87GiyRJcjuQJx0V+u1WnvcQ8EngMuBDSZKsDiHMA54F/DyEcCvpMpQnVL/k2UyOMp07M+k3ZRnaOd+IHUCSYtl9fGKPnkdWnXxz79I9Pvbo6uty4+O3xs6kx1xX7i5fMouvdyHwJbZ99MwzgFXAPtX3m4C1SZIcsdHjaRs9f1bP/bEM7Zyzgb/GDiFJMbVAy5vW9x1/9QPLjzh/+cqArMl6AAARzElEQVR7Dx0avook8S7HuKZlB9cO+C7wqSRJylt6QgjhWOClwJHAh0MIByRJsg74awjh9dXnhBDC4dUvuQY4rfr26TMXfZJlaGcUKyPAx2LHkKRacejIyMHnP/jQiVc/sGzsjevWX9GSJEtiZ8qgS8vd5Stm8wWTJFmWJMkWF2uHENpJBxDeliTJCtI1Q98N6cnnpwNvDyHcBtwJvLL6ZWcC7w0hlIF9Z/QH2JDTE6h3UjEXSBeLHRM7iiTVmgSSi+bNufmrCxfwsNd+zIYEOLrcXf5T7CD1yDK0K4q5k4DLY8eQpFrmtR+z4mfl7vIbY4eoV5ahXVXMXQicGjuGJNW6gRD6v71g/s0/mr/bE4abmg6OnaeBjAGHlLvLXqeykyxDu6qYexpQBppjR5GkenHZnM7bvrhoweDSlpajCaEldp46d1a5u/yu2CHqmWVoOhRz3wL8gyhJO2hFS/ODn1+08J4r5nQ+PQlh99h56tAgcFC5u7widpB6ZhmaDsXcXsB9wLzYUSSpHo3A8A9y82/6zoL5C/ubmg6JnaeOfLHcXf5I7BD1zjI0XYq5TwKfih1DkurdDR3td31+8cI197W2Hk26NVtTWwscWO4ur44dpN5ZhqZLMTcXuJfJ48QlSbvg0eamR760aOFdF8+d85SJEPxv6+OdUe4unxM7RCOwDE2nYu4dpIdLSZKmyTiM/3y3eTf+x8Jcx9rm5iNi56kRfyx3l18QO0Sj8ATq6fU94ObYISSpkTRD82nr+46/6oHlR/xk+cp7n+G1HwPAGbFDNBJHhqZbutX+T0BH7CiS1KgqTaHy9YULbv3lbvPyYyE8KXaeWfahcnf5y7FDNBLL0Ewo5s4Evho7hiQ1ugSS0tw5N39lUWau/bgeeFa5uzwRO0gjsQzNhPTesv8GnM+VpFnyl9aWJf+6eFHv9Y177ccI8Mxyd/nO2EEajWVophRzTwRuBxbEjiJJWTIQQv9ZC+b/6bz5u+3dYNd+fKrcXS7GDtGILEMzqZg7HTgvdgxJyqrL53Te+oVFC4Ya4NqPO4Cjyt3lkdhBGpFlaKYVcz8DXh87hiRl2YqW5gd7Fi28+/I5nYfW4bUfE8AJ5e7yDbGDNCrL0Ewr5haRNnoPDJOkyEZg+Ie5+Teds2D+gv6mpqfHzrOd/q3cXS7EDtHILEOzoZh7CXBx7BiSpEk3ptd+rL63tfWYGr724wrgBeXu8njsII3MMjRbirlvAu+JHUOStKlVTU2P/r9FC+/87bw5B4+HsE/sPBt5kHT32MrYQRqdZWi2FHNzgFuBRtrZIEkNo8au/RgDnlfuLl8dOUcmWIZmUzF3HHAV0Bo7iiRpy+5sa7v3c4sXriy3tz2TEOZGiOAp07PIMjTbirm3Ad+JHUOStG3r0ms/bvvlbvP2Hw0hP0sv+4tyd9ldyLPIMhRDMfdvwEdix5AkbZ+Nrv1IHm5uPnoGr/24Gzim3F1eP0PfX1OwDMWQXtfxS+DVsaNIknbMX1tblnxuZq796AeO87qN2WcZiiVdUH0V8MzYUSRJO24whIGzFsy/+dzpu/bjb8vd5fOn4ftoB1mGYirm9gFuAPaNHUWStPMu7+y87QuLFwzuwrUfXy93l/9x2oNpu1iGYivmjiQdIYqxW0GSNI2q137cc/mczkOSEPbYzi+7CHiVByvGYxmqBcXcK4ELgKbYUSRJu24ERs7Nzb/p7AXzc9u49uMG0vOEBmYrmx7PMlQrirkPA1+MHUOSNL2q136sube19ejNrv24D3hWubv8SKxsSlmGakkxdzbwjtgxJEnTb7NrP1pJi9B9sXPJMlRbirlW4BLg+bGjSJJmxjis/frC3PM/cOYDt8TOopRlqNYUcwuBq4FDYkeRJE27IeClFCuXxw6iSS7YrTXFyhrgBaSnkEqSGscY8HqLUO2xDNWiYmUl8DzgnthRVPvGJxKO/HYfL/9xuhnl9AsGeOo3+jj0m3287deDjI6no7+/vGuUp3+zjxO/18+qgQkA7l89wRt/4SYWaRZMAH9PsXJR7CB6PMtQrSpWHiQtRC6u01b9+/UjPG33yf8rn/6MVv783rmU3zOXwbGEc/40CsDXbxjhxjPm8q6jWvlxeQyAT1w2xGef1z7l95U0rd5LseLp0jXKMlTLipUVpIXo/thRVJuWrZugdO8Y73hm22Mfe9nBrYQQCCFw7D7NLFuXjgI1BRgeg4FRaG2Gq5aMsffcJg5e3BwrvpQVH6FY+VbsENoyy1CtK1aWkRaiv8aOotrzgUuG+MIpHTRNcX/26HjCubeP8pKD0psBPvqcdk45t5/f3DPGmw5t5TNXDvN/T3JUSJpBE8B7KFY8Q67GWYbqQbGyFHguLqrWRi66Z5Q95waO2mfqkZ1/KA3x3Ce1cOKT0jL0wgNbuPmd8/jNm+bw67tHednBLdyzapzX/WyAMy4cZGDUnaXSNBoBTnNEqD64tb6eFHN7Ar8HnhE7iuL76O+HOPf2UVqaYGgM1g0nvOZprZz3mk4+dfkwt6wc54I3dtIUNh02GhhNePmPB7j0zXN4+fkDXPCGOfzirlFGxuGMo9q28GqSdkAf8GqKld/HDqLt48hQPSlWHgZOBm6MnEQ14POndLDsg7vR+4Hd+MnrOnn+AS2c95pOzvnTCJfeP8b5r318EQL44jUj/ONxbbQ2BwZHIYR0PZEjQ9K0WAU83yJUXyxD9aZYWU16DtFVsaOoNr37oiEe6p/ghO/0c8S3+vj0FcOPfW7F+gluWDHOq/6mFYD3H9vGMWf3862bR/nbZ7TGiiw1iqXAcyhW/AdrnXGarF4Vc3OAXwEvih1FksSfgRdV13iqzjgyVK+KlQHg5cB3YkeRpIy7ETjRIlS/HBlqBMXcB4AvAR4YI0mz6/eki6X7YgfRzrMMNYpi7kXAT4EFsaNIUkb8AjidYmUkdhDtGqfJGkWx8jvgeLzPTJJmw38Cb7QINQZHhhpNMbcA+BnwwthRJKkBDQHvo1hxvWYDcWSo0RQra4GXAl+LHUWSGkwv6dZ5i1CDcWSokRVzZwD/AXiAjCTtmouBN1fPelODcWSokRUrZwOnAI/GjiJJdWoC+BegyyLUuBwZyoJiLg/8Bjg0chJJqiergb+lWLk0dhDNLEeGsqBY6QWeRXpitSRp224CnmkRygZHhrKmmHsn8GVgbuwoklSjzgbeT7EyvM1nqiFYhrKomDsYOBc4LnYUSaohg8A/UKx8P3YQzS6nybKoWLkXeA7wKWAschpJqgV/AZ5lEcomR4ayrpg7DjgPOCh2FEmK5PvAP1XPaVMGWYYExdxc0nVE74wdRZJm0VLgDBdJyzKkScXcqcA5wJ6xo0jSDEqAs4D/Q7GyPnYYxWcZ0qaKuT2B7wAvjx1FkmbAX4F3UKz8MXYQ1Q7LkKbmFnxJjSUBvg58jGKlP3YY1RbLkLYs3YL/fdIDGyWpXt0DvJ1i5erYQVSb3FqvLZvcgv9W4KHIaSRpR40DXwKOsAhpaxwZ0vYp5nKk5xK9F2iJnEaStuUu4K0UKzfEDqLaZxnSjinmDiWddz85chJJmsoo8AXg0xQrI7HDqD5YhrRzirk3kg4/7xc7iiRV/Qr45+oUv7TdLEPaeelhjZ8APgi0RU4jKbtuBD5EsXJV7CCqT5Yh7bp019nXgJfEjiIpUx4APgqcT7HiX2baaZYhTZ9i7pXAV4ADYkeR1NDWAZ8HvkqxMhQ7jOqfZUjTq5jrAD4CFIDOyGkkNZYx0ms0ihQrj8QOo8ZhGdLMKOb2Bv4ZeBeWIkm77iLSu8T+HDuIGo9lSDNrshS9G+iInEZS/bkF+LB3iWkmWYY0O4q5JzA5UmQpkrQtDwCfBM6lWJmIHUaNzTKk2ZWWogLwTixFkh7vLtJDE39MsTIaO4yywTKkOIq5fUhL0RlYiiTB9UAP8Gu3yWu2WYYUV1qKPkpaitojp5E0+34HfJ5i5fLYQZRdliHVhmJuXyZHiixFUmObAH4B9FCs3BI7jGQZUm1JS9H7gbcDu0dOI2l6jQA/AL5AsXJf7DDSBpYh1ab08MY3Au8Djo6cRtKuWQ98G/gKxcqK2GGkzVmGVPuKuWNJS9EbcApNqicPA18H/oNiZU3sMNKWWIZUP4q5PYB3kB7guH/kNJKmNgH8N3AO6c4wt8er5lmGVH+KuWbg5aSjRS8AQtxAkkgPSfwe8F2KlQdih5F2hGVI9a2YeyrwXqAbmB85jZQ1o8BvSEeBLvWkaNUry5AaQzE3D/g74D3AMyKnkRrdPcB3gB9QrDwUO4y0qyxDajzF3GHAm0h3ox0QOY3UKAZJzwY6h2LlythhpOlkGVJjK+aOJy1GbwD2jpxGqke3kk6D/YhiZW3sMNJMsAwpG4q5JuBk0mL0GmBR1DxSbSsDFwC/pFgpxw4jzTTLkLKnmGsFXgycBrwSmBc3kBRdQnpR6q+ACzwdWlljGVK2FXNzSLfpnwa8DA91VHaMAVeSjgD9ypOhlWWWIWmDYi4HvJp0tOj5uFVfjWeY9EDEC4ALKVZWRc4j1QTLkDSVYq4FOIF0Ou3FwFF4uKPq03rgt6RTYL+lWFkfOY9UcyxD0vZIrwJ5IWkxehHuTFNtuwf4A2kJ+m+KleHIeaSaZhmSdlQxF4DDSIvRS4BnA21RMynrVpKWnz8Av6dYWRo5j1RXLEPSrirm5gLPY3JK7eC4gZQBa4GrmCw/d0bOI9U1y5A03Yq5A0gXYJ9QfTwN1xtp1zxMWn6uBK4Ayt4DJk0fy5A004q5hcBxwLNIy9FxwG5RM6nW9QLXkJafKylW/hw3jtTYLEPSbEtPwz4UOBY4uvo4DGiNGUtRjAP/S3rlxS3Vx60UK2uippIyxjIk1YJirp20EB290ePpQHPMWJpWA8DtbFp8yhQrQ1FTSbIMSTWrmOsEnrrZ42+Ap+AVIrVuFZuWnluAeyhWxqOmkjQly5BUj4q5fZksRxuXpf2BpojJsqQC/AX462a/3unWdqm+WIakRpKOJh3MpiNJTwX2A/bEorQjRoElPL7spG8XK6sjZpM0jSxDUlYUc83AHqSnZz+h+uuW3m7kabgJYDXpVNaGX1cBy9i47MAyp7WkbLAMSXq89CDJqUrSQmAO0Fn9dUuPTqB9BhOOk47cjAJrmCw0G5ebLT3WUqz4Hz5Jj7EMSZoZ6RECWypLbcAYaZnZ1q+bf2zMMiNpOlmGJElSprmYUpIkZZplSJIkZZplSA0nhDAeQrh1o0d+Bl/rLSGEb8zU95ckzbyW2AGkGTCYJMkRsUNIkuqDI0PKhBBCcwjhiyGEG0MIt4cQ3lX9+MkhhCtCCL8OIfwlhNATQjg9hHBDCKEcQjiw+rxTQwjXhxBuCSH8PoSw1xSvsUcI4ZfV17gxhPDs2f45JUk7zjKkRtS50RTZr6ofeztQSZLkGOAY4IwQwgHVzx0OvBt4GvB3wFOSJDkWOAd4f/U5VwPHJ0lyJPAT4CNTvO6/A1+pvsZrq18vSapxTpOpEU01TfYi4LAQwuuq7+dIr60YAW5MkuRBgBDC/cDvqs8pA8+rvr0f8NMQwhNIz8j56xSvewpwSAhhw/vzQwjzkiTpm4afSZI0QyxDyooAvD9Jkks3+WAIJwPDG31oYqP3J5j8/8jXgS8nSXJh9WuKU7xGE+no0dD0xZYkzTSnyZQVlwLvCSG0AoQQnhJCmLsDX58Dllff7t7Cc37H5LQaIQQXcUtSHbAMKSvOAe4C/hRCuAP4Njs2MloEfh5CuBl4dAvP+Ufg6OoC7btI1yFJkmqc13FIkqRMc2RIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRlmmVIkiRl2v8H1HmoMyh6x5QAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 720x504 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib import pyplot as plt \n",
    "import numpy as np\n",
    "# Creating dataset \n",
    "sex = [\"Male\",\"Female\",\"Mixed\"] \n",
    "  \n",
    "data = [sex_male,sex_female,sex_mixed] \n",
    "  \n",
    "# Creating plot \n",
    "fig = plt.figure(figsize =(10, 7)) \n",
    "plt.pie(data, labels = sex,autopct='%1.0f%%') \n",
    "  \n",
    "# show plot \n",
    "plt.show() "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
